{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "Jacobo de la Cuesta-Zuluaga.\n",
    "\n",
    "According to the `sourmash` result, after metagenome assembly and binning, there were two bins classified as *Archaea*. One of them is classified as *Methanobrevibacter smithii*, while the second one is only classified up to kingdom. We suspect this bin to be **vadinC11**. Here, I will assess the quality of the potential **vadicC11** bin assembled using shotgun reads from the 81 and 93 sequencing runs of the TwinsUK samples."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:15:36.880604Z",
     "start_time": "2018-08-03T11:15:36.862738Z"
    }
   },
   "outputs": [],
   "source": [
    "# Work dir\n",
    "work_dir = \"/ebio/abt3_projects/vadinCA11/data/V11\"\n",
    "\n",
    "# vadinCA11 genome bin\n",
    "V11_bin = \"/ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/metabat2_low_PE.478.contigs.fa\"\n",
    "V11_folder = \"/ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/\"\n",
    "\n",
    "# Misc\n",
    "quality_env = \"py2_genome_quality\"\n",
    "anvio_env = \"anvio\"\n",
    "metacompass_env = \"metacompass\"\n",
    "\n",
    "# anvio coassemble\n",
    "contig_db_file = '/ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/anvio/coassemble.db'\n",
    "\n",
    "# twinsUK\n",
    "tax_file = '/ebio/abt3_projects/TwinsUK/Dataset_summary/TwinsUK_Batch3/TwinsUK_Batch3_taxonomy.txt'\n",
    "seqs_file = '/ebio/abt3_projects/TwinsUK/Dataset_summary/TwinsUK_Batch3/rep_set.fna'\n",
    "\n",
    "# SILVA 132 database\n",
    "silva_align = \"/ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/rep_set/rep_set_16S_only/97/silva_132_97_16S.fna\"\n",
    "silva_tax = \"/ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:15:43.996801Z",
     "start_time": "2018-08-03T11:15:36.882744Z"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import subprocess\n",
    "from Bio import SeqIO\n",
    "from Bio.Blast import NCBIWWW, NCBIXML"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Confirm 16S rRNA gene\n",
    "The Main objective of this section is to confirm that the bin classified as *unclassified archaeon* indeed corresponds to **vadinCA11** using the 16S rRNA gene"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting archaeal 16S genes from anvio db\n",
    "* Aim: To extract all 16S rRNA archaeal reads from the coassembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:16:36.644670Z",
     "start_time": "2018-08-03T11:15:43.999336Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate anvio; anvi-get-sequences-for-hmm-hits   -c /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/anvio/coassemble.db   --hmm-source Ribosomal_RNAs    --list-available-gene-names\"\n",
      "\u001b[0;33m* Ribosomal_RNAs [type: Ribosomal_RNAs]: Archaeal_16S_rRNA, Archaeal_23S_rRNA,\n",
      "Archaeal_5S_rRNA, Bacterial_16S_rRNA, Bacterial_23S_rRNA, Bacterial_5S_rRNA,\n",
      "Eukaryotic_28S_rRNA, Eukaryotic_5S_rRNA, Eukaryotic_5_8S_rRNA,\n",
      "Mitochondrial_12S_rRNA, Mitochondrial_16S_rRNA, Archaeal_5_8S_rRNA\n",
      "\u001b[0m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# listing available genes\n",
    "list_cmd = \"\"\"anvi-get-sequences-for-hmm-hits \\\n",
    "  -c {0} \\\n",
    "  --hmm-source Ribosomal_RNAs \\\n",
    "   --list-available-gene-names\"\"\"\n",
    "list_cmd = list_cmd.format(contig_db_file)\n",
    "list_job = 'bash -c \"source activate {0}; {1}\"'\n",
    "list_job = list_job.format(anvio_env, list_cmd)\n",
    "print(list_job)\n",
    "!$list_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:17:56.660623Z",
     "start_time": "2018-08-03T11:16:36.647172Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate anvio; anvi-get-sequences-for-hmm-hits   -c /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/anvio/coassemble.db   --hmm-source Ribosomal_RNAs   --gene Archaeal_16S_rRNA   -o /ebio/abt3_projects/vadinCA11/data/V11/16S/archaeal_16SrRNA.fna\"\n",
      "\u001b[0;36mContigs DB\u001b[0m ...................................: \u001b[0;33mInitialized: /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/anvio/coassemble.db (v. 10)\u001b[0m\n",
      "\u001b[0;36mHits\u001b[0m .........................................: \u001b[0;33m75 hits for 1 source(s)\u001b[0m\n",
      "\u001b[0;36mFiltered hits\u001b[0m ................................: \u001b[0;33m0 hits remain after filtering for 1 gene(s)\u001b[0m\n",
      "\n",
      "\n",
      "\u001b[0;31mConfig Error\u001b[0m: Your selections resulted in 0 hits. There is nothing to report. Are you sure you\n",
      "              have the right set of gene names and sources? If you select an HMM source, and  \n",
      "              then use gene names that belong to another source, the intersection of the two  \n",
      "              can be empty. Just saying.                                                      \n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# writing out genes\n",
    "archaeal_16S_file = os.path.join(work_dir, \"16S\", 'archaeal_16SrRNA.fna')\n",
    "w16S_cmd = \"\"\"anvi-get-sequences-for-hmm-hits \\\n",
    "  -c {0} \\\n",
    "  --hmm-source Ribosomal_RNAs \\\n",
    "  --gene Archaeal_16S_rRNA \\\n",
    "  -o {1}\"\"\"\n",
    "w16S_cmd = w16S_cmd.format(contig_db_file, archaeal_16S_file)\n",
    "w16S_job = 'bash -c \"source activate {}; {}\"'\n",
    "w16S_job = w16S_job.format(anvio_env, w16S_cmd)\n",
    "print(w16S_job)\n",
    "!$w16S_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:17:56.796656Z",
     "start_time": "2018-08-03T11:17:56.663016Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "grep: /ebio/abt3_projects/vadinCA11/data/V11/16S/archaeal_16SrRNA.fna: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!grep \">\" $archaeal_16S_file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For some reason the extraction of 16S rRNA sequences from the coassembly using anvio doesn't work. I will verify what is happening latter, but for now I will take a different approach. Instead of extracting the 16S sequences from the complete coassembly, I will do it from the **vadinCA11** bin, and will work from there."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting vadinCA11 16S genes\n",
    "* Aim: To extract the 16S amplicon reads classified as **vadinCA11** from the TwinsUK data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:17:57.312677Z",
     "start_time": "2018-08-03T11:17:56.799166Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 /ebio/abt3_projects/vadinCA11/data/V11/16S/vadinCA11_tax.txt\r\n"
     ]
    }
   ],
   "source": [
    "# querying tax file\n",
    "vadinCA11_tax_file = os.path.join(work_dir, \"16S\", 'vadinCA11_tax.txt')\n",
    "!grep -i \"vadinCA11\" $tax_file > $vadinCA11_tax_file\n",
    "!wc -l $vadinCA11_tax_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:17:57.684645Z",
     "start_time": "2018-08-03T11:17:57.315228Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 /ebio/abt3_projects/vadinCA11/data/V11/16S/tmp.txt\n",
      ">565570 \n",
      ">4442338 \n",
      ">New.0.CleanUp.ReferenceOTU19387 \n",
      ">New.0.CleanUp.ReferenceOTU25898 \n",
      ">152715 \n",
      ">4011307 \n",
      ">New.3.CleanUp.ReferenceOTU8270 \n",
      ">New.10.CleanUp.ReferenceOTU2657 \n",
      ">New.11.CleanUp.ReferenceOTU470 \n",
      ">New.17.CleanUp.ReferenceOTU26822 \n",
      ">New.26.CleanUp.ReferenceOTU28048 \n",
      ">New.32.CleanUp.ReferenceOTU2934 \n"
     ]
    }
   ],
   "source": [
    "# querying tax file\n",
    "tmp_file = os.path.join(work_dir, \"16S\", 'tmp.txt')\n",
    "!cut -f 1 $vadinCA11_tax_file | perl -pe 's/^/>/;s/$/ /'> $tmp_file\n",
    "!wc -l $tmp_file\n",
    "!cat $tmp_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:17:58.676660Z",
     "start_time": "2018-08-03T11:17:57.687083Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12\n",
      ">4442338 High1.23_142512\n",
      "TACCCGCAGCCCAAGTGGTGGTCGATTTTATTGAGTCTAAAACGTTCGTAGCCGGTTCATTAAATCCTTGGGTAAATCGGGAAGCTTAACTTTCCGACTTCCGAGGAGACTGGTGAACTTGGGACCGGGAGAGGCAAGAGGTACTTCTGGGGTAGGGGTAAAATCCTGTAATCCTAGAAGGACCACCGGTGGCGAAGGCGTCTTGCTAGAACGGATCCGACGGTGAGGGACGAAGCCCTGGGTCGCAAACGGG\n",
      ">565570 High1.67_26391\n",
      "TACCTGCAGCCCAAGTGGTGGTCGATTTTATTGAGTCTAAAACGTTCGTAGCCGGTCTGGTAAATCCTTGGGTAAATCGGGAAGCTTAACTTTCCGACTTCCGAGGAGACTGCCAGACTTGGGACCGGGAGAGGCAAGAGGTACTTCTGGGGTAGGGGTAAAATCCTGTAATCCTAGAAGGACCACCGGTGGCGAAGGCGTCTTGCTAGAACGGATCCGACGGTGAGGGACGAAGCCCTGGGTCGCAAACGGG\n",
      ">New.0.CleanUp.ReferenceOTU19387 High1.82_4039280\n",
      "TACCTGCAGCCCAAGTGGTGGTCGATTTTATTGAGTCTAAAACGTTCGTAGCCGGTCTGGTAAATCCTTGGGTAAATCGGGGAGCTCAACTTTCCGAATTCCGAGGAGACTGCCAGACTTGGGACCGGGAGAGGCTAGAGGTACTTCTGGGGTAGGGGTAAAATCCTGTAATCCTAGAAGGACCACCGGTGGCGAAGGCGTCTAGTTAGAACGGATCCGACGGTGAGGGACGAAGCCCTGGGTCGCAAACGGG\n",
      ">New.0.CleanUp.ReferenceOTU25898 High1.23_1032914\n",
      "TACCCGCAGCCCGAGTGGTGGTCGATTTTATTGAGTCTAAAACGTTCGTAGCCGGTCTGATAGATCCTTGGGTAAATCGGGGGGCTTAACCTTCCGAATTCCGAGGAGACCGTCAGGCTTGGGATCGGGAGAGGTAAGAGGTACTTCAGGGGTAGGGGTAAAATCCTGTAATCCTTGGAGGACCACCGGTGGCGAAGGCGTCTTACTAGAACGAATCCGACGGTGAGGGACGAAGCCCTAGGTCGCAAACGGG\n",
      ">152715 High2.44_116499\n",
      "TACCTGCAGCCCAAGTGGTGGTCGATTTTATTGAGTCTAAAGCGTTCGTAGCCGGTCCGGTAAATCCTTGGGTAAATCGGGAAGCTTAACTTTCCGACTTCCGAGGAGACTGCCGGACTTGGGACCGGGAGAGGCTAGAGGTACTTCTGGGGTAGGGGTAAAATCCTGTAATCCTAGAAGGACCACCGGTGGCGAAGGCGTCTAGCTAGAACGGATCCGACGGTGAGGGACGAAGCCCTGGGTCGCAAACGGG\n"
     ]
    }
   ],
   "source": [
    "vadinCA11_seqs_file = os.path.join(work_dir, \"16S\", 'vadinCA11_rep.fna')\n",
    "!egrep -A 1 -f $tmp_file $seqs_file | perl -ne 'print unless /^--$/' > $vadinCA11_seqs_file\n",
    "!grep -c \">\" $vadinCA11_seqs_file\n",
    "!head $vadinCA11_seqs_file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Blasting 16S amplicons to the vadinCA11 bin\n",
    "* Aim: to blast the 16S amplicons classified as **VadinCA11** to the candidate bin to determine their similarity. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:05.004615Z",
     "start_time": "2018-08-03T11:17:58.679129Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate anvio; blastn -query /ebio/abt3_projects/vadinCA11/data/V11/16S/vadinCA11_rep.fna -subject /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/metabat2_low_PE.478.contigs.fa -outfmt 6 > /ebio/abt3_projects/vadinCA11/data/V11/16S/V11_bin_16S.txt\"\n"
     ]
    }
   ],
   "source": [
    "# Run blast job\n",
    "blast_res_file = os.path.join(work_dir, \"16S\", 'V11_bin_16S.txt')\n",
    "blast_cmd = 'blastn -query {query} -subject {subject} -outfmt 6 > {output}'\n",
    "blast_cmd = blast_cmd.format(query = vadinCA11_seqs_file,\n",
    "                 subject = V11_bin,\n",
    "                 output = blast_res_file)\n",
    "blast_job = 'bash -c \"source activate {}; {}\"'\n",
    "blast_job = blast_job.format(anvio_env, blast_cmd)\n",
    "print(blast_job)\n",
    "!$blast_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:05.048605Z",
     "start_time": "2018-08-03T11:18:05.007113Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4442338</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>91.37</td>\n",
       "      <td>255</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>414</td>\n",
       "      <td>162</td>\n",
       "      <td>4.000000e-98</td>\n",
       "      <td>346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>565570</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>92.49</td>\n",
       "      <td>253</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>414</td>\n",
       "      <td>162</td>\n",
       "      <td>4.000000e-103</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>New.0.CleanUp.ReferenceOTU19387</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>91.70</td>\n",
       "      <td>253</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>414</td>\n",
       "      <td>162</td>\n",
       "      <td>8.000000e-100</td>\n",
       "      <td>351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>New.0.CleanUp.ReferenceOTU25898</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>414</td>\n",
       "      <td>162</td>\n",
       "      <td>8.000000e-135</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>152715</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>90.51</td>\n",
       "      <td>253</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>414</td>\n",
       "      <td>162</td>\n",
       "      <td>8.000000e-95</td>\n",
       "      <td>335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4011307</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>91.30</td>\n",
       "      <td>253</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>414</td>\n",
       "      <td>162</td>\n",
       "      <td>4.000000e-98</td>\n",
       "      <td>346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>New.3.CleanUp.ReferenceOTU8270</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>91.70</td>\n",
       "      <td>253</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>414</td>\n",
       "      <td>162</td>\n",
       "      <td>8.000000e-100</td>\n",
       "      <td>351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>New.10.CleanUp.ReferenceOTU2657</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>88.41</td>\n",
       "      <td>207</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>207</td>\n",
       "      <td>414</td>\n",
       "      <td>208</td>\n",
       "      <td>3.000000e-69</td>\n",
       "      <td>250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>New.11.CleanUp.ReferenceOTU470</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>92.27</td>\n",
       "      <td>207</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>207</td>\n",
       "      <td>414</td>\n",
       "      <td>208</td>\n",
       "      <td>1.000000e-82</td>\n",
       "      <td>294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>New.17.CleanUp.ReferenceOTU26822</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>90.91</td>\n",
       "      <td>253</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>414</td>\n",
       "      <td>162</td>\n",
       "      <td>2.000000e-96</td>\n",
       "      <td>340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>New.26.CleanUp.ReferenceOTU28048</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>92.00</td>\n",
       "      <td>200</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>200</td>\n",
       "      <td>414</td>\n",
       "      <td>215</td>\n",
       "      <td>1.000000e-78</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>New.32.CleanUp.ReferenceOTU2934</td>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>90.80</td>\n",
       "      <td>250</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>250</td>\n",
       "      <td>414</td>\n",
       "      <td>165</td>\n",
       "      <td>8.000000e-95</td>\n",
       "      <td>335</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              qseqid            sseqid  pident  length  \\\n",
       "0                            4442338  coassemble_31453   91.37     255   \n",
       "1                             565570  coassemble_31453   92.49     253   \n",
       "2    New.0.CleanUp.ReferenceOTU19387  coassemble_31453   91.70     253   \n",
       "3    New.0.CleanUp.ReferenceOTU25898  coassemble_31453  100.00     253   \n",
       "4                             152715  coassemble_31453   90.51     253   \n",
       "5                            4011307  coassemble_31453   91.30     253   \n",
       "6     New.3.CleanUp.ReferenceOTU8270  coassemble_31453   91.70     253   \n",
       "7    New.10.CleanUp.ReferenceOTU2657  coassemble_31453   88.41     207   \n",
       "8     New.11.CleanUp.ReferenceOTU470  coassemble_31453   92.27     207   \n",
       "9   New.17.CleanUp.ReferenceOTU26822  coassemble_31453   90.91     253   \n",
       "10  New.26.CleanUp.ReferenceOTU28048  coassemble_31453   92.00     200   \n",
       "11   New.32.CleanUp.ReferenceOTU2934  coassemble_31453   90.80     250   \n",
       "\n",
       "    mismatch  gapopen  qstart  qend  sstart  send         evalue  bitscore  \n",
       "0         18        4       1   253     414   162   4.000000e-98       346  \n",
       "1         19        0       1   253     414   162  4.000000e-103       363  \n",
       "2         21        0       1   253     414   162  8.000000e-100       351  \n",
       "3          0        0       1   253     414   162  8.000000e-135       468  \n",
       "4         24        0       1   253     414   162   8.000000e-95       335  \n",
       "5         22        0       1   253     414   162   4.000000e-98       346  \n",
       "6         21        0       1   253     414   162  8.000000e-100       351  \n",
       "7         24        0       1   207     414   208   3.000000e-69       250  \n",
       "8         16        0       1   207     414   208   1.000000e-82       294  \n",
       "9         23        0       1   253     414   162   2.000000e-96       340  \n",
       "10        16        0       1   200     414   215   1.000000e-78       281  \n",
       "11        23        0       1   250     414   165   8.000000e-95       335  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print table with blast results\n",
    "bin16S_df = pd.read_csv(blast_res_file, sep='\\t', header=None)\n",
    "bin16S_df.columns = ['qseqid','sseqid','pident','length','mismatch',\n",
    "              'gapopen','qstart','qend','sstart','send','evalue','bitscore']\n",
    "bin16S_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extracting 16S genes from V11 bin\n",
    "* Aim: To extract the 16S rRNA gene present in the **vadinCA11** bin and to determine the closest matches in SILVA.\n",
    "\n",
    "For this, I will extract it using `barrnap` and use blast to compare it to sequences in the SILVA 132 database (clustered at 97% identity)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:16.484612Z",
     "start_time": "2018-08-03T11:18:05.050689Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate metacompass; barrnap --kingdom arc --threads 8 --outseq /ebio/abt3_projects/vadinCA11/data/V11/16S/barrnap_rRNA.fa < /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/metabat2_low_PE.478.contigs.fa > /ebio/abt3_projects/vadinCA11/data/V11/16S/barrnap_rRNA.gff\"\n",
      "[barrnap] This is barrnap 0.9\n",
      "[barrnap] Written by Torsten Seemann\n",
      "[barrnap] Obtained from https://github.com/tseemann/barrnap\n",
      "[barrnap] Detected operating system: linux\n",
      "[barrnap] Adding /ebio/abt3_projects/software/miniconda3/envs/metacompass/lib/barrnap/bin/../binaries/linux to end of PATH\n",
      "[barrnap] Checking for dependencies:\n",
      "[barrnap] Found nhmmer - /ebio/abt3_projects/software/miniconda3/envs/metacompass/bin/nhmmer\n",
      "[barrnap] Found bedtools - /ebio/abt3_projects/software/miniconda3/envs/metacompass/bin/bedtools\n",
      "[barrnap] Will use 8 threads\n",
      "[barrnap] Setting evalue cutoff to 1e-06\n",
      "[barrnap] Will tag genes < 0.8 of expected length.\n",
      "[barrnap] Will reject genes < 0.25 of expected length.\n",
      "[barrnap] Using database: /ebio/abt3_projects/software/miniconda3/envs/metacompass/lib/barrnap/bin/../db/arc.hmm\n",
      "[barrnap] Copying STDIN to a temporary file: /tmp/QimSU7SJ6H\n",
      "[barrnap] Scanning /tmp/QimSU7SJ6H for arc rRNA genes... please wait\n",
      "[barrnap] Command: nhmmer --cpu 8 -E 1e-06 --w_length 3878 -o /dev/null --tblout /dev/stdout '/ebio/abt3_projects/software/miniconda3/envs/metacompass/lib/barrnap/bin/../db/arc.hmm' '/tmp/QimSU7SJ6H'\n",
      "[barrnap] Found: 16S_rRNA coassemble_31453 L=878/1585 1..878 - 16S ribosomal RNA (partial)\n",
      "[barrnap] Found: 16S_rRNA coassemble_80719 L=679/1585 2..680 + 16S ribosomal RNA (partial)\n",
      "[barrnap] Found: 23S_rRNA coassemble_80719 L=2887/3232 41136..44022 + 23S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA coassemble_55654 L=106/119 19..124 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA coassemble_59052 L=106/119 19..124 - 5S ribosomal RNA\n",
      "[barrnap] Found: 5S_rRNA coassemble_60057 L=106/119 8234..8339 - 5S ribosomal RNA\n",
      "[barrnap] Found 6 ribosomal RNA features.\n",
      "[barrnap] Sorting features and outputting GFF3...\n",
      "[barrnap] Writing hit sequences to: /ebio/abt3_projects/vadinCA11/data/V11/16S/barrnap_rRNA.fa\n",
      "[barrnap] Running: bedtools getfasta -s -name+ -fo '/ebio/abt3_projects/vadinCA11/data/V11/16S/barrnap_rRNA.fa' -fi '/tmp/QimSU7SJ6H' -bed '/tmp/6RgXAsGkFd'\n",
      "index file /tmp/QimSU7SJ6H.fai not found, generating...\n",
      "[barrnap] Done.\n"
     ]
    }
   ],
   "source": [
    "# Extract 16S sequences from the assembled contigs using barrnap\n",
    "barrnap_fasta = os.path.join(work_dir, \"16S\", 'barrnap_rRNA.fa')\n",
    "barrnap_gff = os.path.join(work_dir, \"16S\", 'barrnap_rRNA.gff')\n",
    "barrnap_cmd = \"barrnap --kingdom {0} --threads {1} --outseq {2} < {3} > {4}\"\n",
    "barrnap_cmd = barrnap_cmd.format(\"arc\", 8, barrnap_fasta, V11_bin, barrnap_gff)\n",
    "barrnap_job = 'bash -c \"source activate {}; {}\"'\n",
    "barrnap_job = barrnap_job.format(metacompass_env, barrnap_cmd)\n",
    "print(barrnap_job)\n",
    "!$barrnap_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:16.516553Z",
     "start_time": "2018-08-03T11:18:16.487362Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>source</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>attribute</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>coassemble_31453</td>\n",
       "      <td>barrnap:0.9</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>1</td>\n",
       "      <td>878</td>\n",
       "      <td>7.600000e-212</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>Name=16S_rRNA;product=16S ribosomal RNA (parti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>coassemble_55654</td>\n",
       "      <td>barrnap:0.9</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>19</td>\n",
       "      <td>124</td>\n",
       "      <td>4.400000e-11</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>Name=5S_rRNA;product=5S ribosomal RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>coassemble_59052</td>\n",
       "      <td>barrnap:0.9</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>19</td>\n",
       "      <td>124</td>\n",
       "      <td>4.400000e-11</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>Name=5S_rRNA;product=5S ribosomal RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>coassemble_60057</td>\n",
       "      <td>barrnap:0.9</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>8234</td>\n",
       "      <td>8339</td>\n",
       "      <td>4.400000e-11</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>Name=5S_rRNA;product=5S ribosomal RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>coassemble_80719</td>\n",
       "      <td>barrnap:0.9</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>2</td>\n",
       "      <td>680</td>\n",
       "      <td>6.300000e-177</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Name=16S_rRNA;product=16S ribosomal RNA (parti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>coassemble_80719</td>\n",
       "      <td>barrnap:0.9</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>41136</td>\n",
       "      <td>44022</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Name=23S_rRNA;product=23S ribosomal RNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            seqname       source feature  start    end          score strand  \\\n",
       "0  coassemble_31453  barrnap:0.9    rRNA      1    878  7.600000e-212      -   \n",
       "1  coassemble_55654  barrnap:0.9    rRNA     19    124   4.400000e-11      -   \n",
       "2  coassemble_59052  barrnap:0.9    rRNA     19    124   4.400000e-11      -   \n",
       "3  coassemble_60057  barrnap:0.9    rRNA   8234   8339   4.400000e-11      -   \n",
       "4  coassemble_80719  barrnap:0.9    rRNA      2    680  6.300000e-177      +   \n",
       "5  coassemble_80719  barrnap:0.9    rRNA  41136  44022   0.000000e+00      +   \n",
       "\n",
       "  frame                                          attribute  \n",
       "0     .  Name=16S_rRNA;product=16S ribosomal RNA (parti...  \n",
       "1     .              Name=5S_rRNA;product=5S ribosomal RNA  \n",
       "2     .              Name=5S_rRNA;product=5S ribosomal RNA  \n",
       "3     .              Name=5S_rRNA;product=5S ribosomal RNA  \n",
       "4     .  Name=16S_rRNA;product=16S ribosomal RNA (parti...  \n",
       "5     .            Name=23S_rRNA;product=23S ribosomal RNA  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print all rRNA genes found by `barrnap`\n",
    "V11_rRNA_df = pd.read_csv(barrnap_gff, sep='\\t', skiprows=1, header=None)\n",
    "V11_rRNA_df.columns = ['seqname','source','feature','start','end', 'score', \n",
    "                       'strand','frame','attribute']\n",
    "V11_rRNA_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:16.652671Z",
     "start_time": "2018-08-03T11:18:16.518610Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">16S_rRNA::coassemble_31453:0-878(-)\r\n",
      "CTCCGGTTGATCCTGCCGGCGGCCACCGCTATAGGAATTCGATTAAGACATGCGAGTCGAGAGCGCAAGCTCGGCGGACTGCTCAGTAACACGTGGACAACGTGCCCTAAAGTGGGGGATAATCAGGGGAAACTCTGGATAATACCCCATAGATCATGAGATCTGGAATGACTTATGGTTCAAAGTTCCGGCGCTTTAGGATCGGTCTGCGGCCTATCAGGTAGTAGTGGGTGTAATGTACCCCCTAGCCTATTACGGGTATGGGCCTTGAGAGAGGGAGCCCAGAGTTGGATTCTGAGACACGAATCCAGGCCCTACGGGGCGCAGCAGTCGCGAAAACTTCACAATGGGCGAAAGCCCGATGAGGGAATTCCTAGTGCTAGCACTTTTGTGTTAGCTTTTCTTTAGCGTAGATAACTAGAGGAATAAGGGCTGGGTAAGACGGGTGCCAGCCGCCGCGGTAATACCCGCAGCCCGAGTGGTGGTCGATTTTATTGAGTCTAAAACGTTCGTAGCCGGTCTGATAGATCCTTGGGTAAATCGGGGGGCTTAACCTTCCGAATTCCGAGGAGACCGTCAGGCTTGGGATCGGGAGAGGTAAGAGGTACTTCAGGGGTAGGGGTAAAATCCTGTAATCCTTGGAGGACCACCGGTGGCGAAGGCGTCTTACTAGAACGAATCCGACGGTGAGGGACGAAGCCCTAGGTCGCAAACGGGATTAGATACCCCGGTAGTCTAGGGTGTAAACGCTGCAGACTTGGTGTTGGAGGCCCTTCGGGGGCATTCAGTGCCGGAGAGAAGTTGTTAAGTCTGCTACTTGGGGAGTACGTCCGCAAGGATGAAACTTAAAGGAATTGGCGGGGGAGCACCGCAACGGG\r\n",
      ">16S_rRNA::coassemble_80719:1-680(+)\r\n",
      "AGTGCCGGAGAGAAGTTGTTAAGTCTGCTACTTGGGGAGTACGTCCGCAAGGATGAAACTTAAAGGAATTGGCGGGGGAGCACCGCAACGGGAGGAGCGTGCGGTTTAATTGGATTCAACACCGGAAAACTCACCAAGGGAGACCATCACATGAAAGCCAGGCTAATGACTTTGCTTGATTCTTGGAGAAGTGGTGCATGGCCATCGTCAGTTCGTACTGTAAAGCGTTCTCTTAAGTGAGATAACGAACAAGACCCTCACTTATAATTGCTAACTAGATCTCCGGATCTGGTGCACATTATCGGGACCGCTGGCGCTAAGTCAGAGGAAGGAGAGGTCAACGGTAGGTCAGTATGCCCTGAATCTCTTGGGCTACACGCGCGCTACAAAGGGCGGGACAATGAGTTCCGACACCGAAAGGTGAAGGTAATCTCCAAACCCGTCCGTAGTTCGGACTGAGGGTTGTAACTCACCCTCACGAAGCTGGATTCCGTAGTAATCGCGAATCAAAAACTCGCGGTGAATATGCCCCTGCTCCTTGCACACACCGCCCGTCAAACCACCCGAGTCGGGTTTCAGTGAGGCTGCCTTTTATTGGGGTATTCGAGCTGAGATTCAGCAAGGGAGGTTAAGTCGTAACAAGGTATCTGTAGGGGAACCTGCAGATGGATCACCTCCT\r\n"
     ]
    }
   ],
   "source": [
    "# Write fasta file with 16S sequences\n",
    "barrnap_16S = os.path.join(work_dir, \"16S\", 'barrnap_16S.fa')\n",
    "!grep -A 1 \"16S_rRNA\" $barrnap_fasta > $barrnap_16S; cat $barrnap_16S"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:52.300674Z",
     "start_time": "2018-08-03T11:18:16.655140Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate anvio; blastn -query /ebio/abt3_projects/vadinCA11/data/V11/16S/barrnap_16S.fa -subject /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/rep_set/rep_set_16S_only/97/silva_132_97_16S.fna -outfmt 6 > /ebio/abt3_projects/vadinCA11/data/V11/16S/SILVA_blast.txt\"\n"
     ]
    }
   ],
   "source": [
    "# Blast 16S sequences against SILVA 132 clustered at 97% identity\n",
    "silva_result = os.path.join(work_dir, \"16S\", 'SILVA_blast.txt')\n",
    "silva_cmd = 'blastn -query {query} -subject {subject} -outfmt 6 > {output}'\n",
    "silva_cmd = silva_cmd.format(query = barrnap_16S, subject = silva_align, output = silva_result)\n",
    "silva_job = 'bash -c \"source activate {}; {}\"'\n",
    "silva_job = silva_job.format(anvio_env, silva_cmd)\n",
    "print(silva_job)\n",
    "!$silva_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:52.341044Z",
     "start_time": "2018-08-03T11:18:52.303165Z"
    }
   },
   "outputs": [],
   "source": [
    "# Generate tables with blast results\n",
    "silva_df = pd.read_csv(silva_result, sep='\\t', header=None)\n",
    "silva_df.columns = ['qseqid','sseqid','pident','length','mismatch',\n",
    "              'gapopen','qstart','qend','sstart','send','evalue','bitscore']\n",
    "\n",
    "\n",
    "silva_seq1 = silva_df.loc[silva_df['qseqid'] == \"16S_rRNA::coassemble_31453:0-878(-)\"]\n",
    "silva_seq2 = silva_df.loc[silva_df['qseqid'] == \"16S_rRNA::coassemble_80719:1-680(+)\"]\n",
    "\n",
    "silva_seq1 = silva_seq1.sort_values(by=[\"pident\"], ascending=False)\n",
    "silva_seq2 = silva_seq2.sort_values(by=[\"pident\"], ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:52.876682Z",
     "start_time": "2018-08-03T11:18:52.343042Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"grep \"AB237752.1.1428\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"CBKQ010000059.83060.84525\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n"
     ]
    }
   ],
   "source": [
    "# Sequence 1\n",
    "# Select matches with identity >= 95% and find their tax classification\n",
    "seq1_tax = os.path.join(work_dir, \"16S\", '16S_seq1.tax')\n",
    "silva_grep = 'bash -c \"grep {0} {1}\"'\n",
    "for match in silva_seq1.loc[silva_df[\"pident\"] >= 97][[\"sseqid\"]].values:\n",
    "    tax_id = '\"{0}\"'.format(match[0]) \n",
    "    grep_job = silva_grep.format(tax_id, silva_tax)\n",
    "    print(grep_job)\n",
    "    !$grep_job | sed 's/D_[0-9]__//g' >> $seq1_tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:54.588691Z",
     "start_time": "2018-08-03T11:18:52.879206Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"grep \"CBKQ010000059.83060.84525\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"KJ522703.1.1246\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"JF807263.1.1257\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"KU754011.1.1256\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"JF807280.1.1256\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"JF807141.1.1256\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"JF807311.1.1257\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"KM213864.1.1403\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"JF807283.1.1257\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"JF807167.1.1259\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n",
      "bash -c \"grep \"HQ616009.1.1257\" /ebio/abt3_projects/databases/SILVA/SILVA_132_QIIME_release/taxonomy/16S_only/97/majority_taxonomy_7_levels.txt\"\n"
     ]
    }
   ],
   "source": [
    "# Sequence 2\n",
    "# Select matches with identity >= 97% and find their tax classification\n",
    "seq2_tax = os.path.join(work_dir, \"16S\", '16S_seq2.tax')\n",
    "silva_grep = 'bash -c \"grep {0} {1}\"'\n",
    "for match in silva_seq2.loc[silva_df[\"pident\"] >= 97][[\"sseqid\"]].values:\n",
    "    tax_id = '\"{0}\"'.format(match[0]) \n",
    "    grep_job = silva_grep.format(tax_id, silva_tax)\n",
    "    print(grep_job)\n",
    "    !$grep_job | sed 's/D_[0-9]__//g' >> $seq2_tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:18:54.648604Z",
     "start_time": "2018-08-03T11:18:54.591530Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/ebio/abt3_projects/software/miniconda3/envs/qiime2/lib/python3.5/site-packages/ipykernel/__main__.py:2: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AB237752.1.1428</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>uncultured</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBKQ010000059.83060.84525</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KJ522703.1.1246</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KU754009.1.1257</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JF807078.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JF807279.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JF807141.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KU754014.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CP014214.1550807.1552273</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>JF807281.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>JF807294.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>KM213864.1.1403</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>JF807145.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>uncultured</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>JF807263.1.1257</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>JF807130.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>uncultured</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AB237752.1.1428</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>uncultured</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CBKQ010000059.83060.84525</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      seqname  Kingdom         Phylum           Class  \\\n",
       "0             AB237752.1.1428  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "1   CBKQ010000059.83060.84525  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "2             KJ522703.1.1246  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "3             KU754009.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "4             JF807078.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "5             JF807279.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "6             JF807141.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "7             KU754014.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "8    CP014214.1550807.1552273  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "9             JF807281.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "10            JF807294.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "11            KM213864.1.1403  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "12            JF807145.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "13            JF807263.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "14            JF807130.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "15            AB237752.1.1428  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "16  CBKQ010000059.83060.84525  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "\n",
       "                      Order                   Family  \\\n",
       "0                uncultured      uncultured archaeon   \n",
       "1   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "2   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "3   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "4   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "5   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "6   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "7   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "8   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "9   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "10  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "11  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "12  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "13  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "14  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "15               uncultured      uncultured archaeon   \n",
       "16  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "\n",
       "                              Genus                                Species  \n",
       "0               uncultured archaeon                    uncultured archaeon  \n",
       "1   Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "2   Candidatus Methanomethylophilus  Candidatus Methanomethylophilus alvus  \n",
       "3   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "4   Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "5   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "6   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "7   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "8   Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "9   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "10  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "11  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "12                       uncultured                         Ambiguous_taxa  \n",
       "13  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "14                       uncultured                    uncultured archaeon  \n",
       "15              uncultured archaeon                    uncultured archaeon  \n",
       "16  Candidatus Methanomethylophilus                         Ambiguous_taxa  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the 7 level taxonomic classification of the top hits\n",
    "seq_1_df = pd.read_csv(seq1_tax, sep='\\t|;', skiprows=0, header=None)\n",
    "seq_1_df.columns = ['seqname','Kingdom','Phylum','Class','Order', 'Family', \n",
    "                       'Genus','Species']\n",
    "seq_1_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-08-03T11:22:12.188599Z",
     "start_time": "2018-08-03T11:22:12.142880Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/ebio/abt3_projects/software/miniconda3/envs/qiime2/lib/python3.5/site-packages/ipykernel/__main__.py:2: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CBKQ010000059.83060.84525</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KJ522703.1.1246</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JF807263.1.1257</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KU754011.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JF807280.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JF807141.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JF807311.1.1257</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KM213864.1.1403</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>JF807283.1.1257</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>JF807167.1.1259</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HQ616009.1.1257</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>EU413662.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AY995300.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CP002916.334433.335897</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CP014214.1550807.1552273</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AY995290.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>JX522702.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>uncultured</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HM038376.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GU797104.1.1254</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured methanogenic archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>JF807191.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>JF807274.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>JF807161.1.1257</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>uncultured</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>JF807281.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>JF807118.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>CBKQ010000059.83060.84525</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Ambiguous_taxa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>KJ522703.1.1246</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>JF807263.1.1257</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>KU754011.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>JF807280.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Thermoplasmata</td>\n",
       "      <td>Methanomassiliicoccales</td>\n",
       "      <td>Methanomethylophilaceae</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>uncultured archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>JF807141.1.1256</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Eurya"
      ],
      "text/plain": [
       "                      seqname  Kingdom         Phylum           Class  \\\n",
       "0   CBKQ010000059.83060.84525  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "1             KJ522703.1.1246  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "2             JF807263.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "3             KU754011.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "4             JF807280.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "5             JF807141.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "6             JF807311.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "7             KM213864.1.1403  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "8             JF807283.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "9             JF807167.1.1259  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "10            HQ616009.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "11            EU413662.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "12            AY995300.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "13     CP002916.334433.335897  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "14   CP014214.1550807.1552273  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "15            AY995290.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "16            JX522702.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "17            HM038376.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "18            GU797104.1.1254  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "19            JF807191.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "20            JF807274.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "21            JF807161.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "22            JF807281.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "23            JF807118.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "24  CBKQ010000059.83060.84525  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "25            KJ522703.1.1246  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "26            JF807263.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "27            KU754011.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "28            JF807280.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "29            JF807141.1.1256  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "30            JF807311.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "31            KM213864.1.1403  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "32            JF807283.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "33            JF807167.1.1259  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "34            HQ616009.1.1257  Archaea  Euryarchaeota  Thermoplasmata   \n",
       "\n",
       "                      Order                   Family  \\\n",
       "0   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "1   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "2   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "3   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "4   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "5   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "6   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "7   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "8   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "9   Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "10  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "11  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "12  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "13  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "14  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "15  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "16  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "17  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "18  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "19  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "20  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "21  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "22  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "23  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "24  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "25  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "26  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "27  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "28  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "29  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "30  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "31  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "32  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "33  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "34  Methanomassiliicoccales  Methanomethylophilaceae   \n",
       "\n",
       "                              Genus                                Species  \n",
       "0   Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "1   Candidatus Methanomethylophilus  Candidatus Methanomethylophilus alvus  \n",
       "2   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "3   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "4   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "5   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "6   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "7   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "8   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "9   Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "10  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "11  Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "12  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "13  Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "14  Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "15  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "16                       uncultured                    uncultured archaeon  \n",
       "17  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "18  Candidatus Methanomethylophilus       uncultured methanogenic archaeon  \n",
       "19  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "20  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "21                       uncultured                    uncultured archaeon  \n",
       "22  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "23  Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "24  Candidatus Methanomethylophilus                         Ambiguous_taxa  \n",
       "25  Candidatus Methanomethylophilus  Candidatus Methanomethylophilus alvus  \n",
       "26  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "27  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "28  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "29  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "30  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "31  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "32  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "33  Candidatus Methanomethylophilus                    uncultured archaeon  \n",
       "34  Candidatus Methanomethylophilus                    uncultured archaeon  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 11790 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Print the 7 level taxonomic classification of the top hits\n",
    "seq_2_df = pd.read_csv(seq2_tax, sep='\\t|;', skiprows=0, header=None)\n",
    "seq_2_df.columns = ['seqname','Kingdom','Phylum','Class','Order', 'Family', \n",
    "                       'Genus','Species']\n",
    "seq_2_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Assessment of bin assembly\n",
    "* Aim: to assess the quality of the assembly of the **vadinCA11** genome bin.\n",
    "\n",
    "For this I will use both CheckM and Quast."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CheckM"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Running CheckM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-05T14:38:27.560656Z",
     "start_time": "2018-06-05T14:30:27.806292Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate py2_genome_quality; checkm lineage_wf -t 8     -x 478.contigs.fa     -f /ebio/abt3_projects/vadinCA11/data/V11/checkm_output/CheckM_V11.txt     /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/ /ebio/abt3_projects/vadinCA11/data/V11/checkm_output \"\n",
      "\n",
      "*******************************************************************************\n",
      " [CheckM - tree] Placing bins in reference genome tree.\n",
      "*******************************************************************************\n",
      "\n",
      "  Identifying marker genes in 2 bins with 8 threads:\n",
      "    Finished processing 2 of 2 (100.00%) bins.\n",
      "  Saving HMM info to file.\n",
      "\n",
      "  Calculating genome statistics for 2 bins with 8 threads:\n",
      "    Finished processing 2 of 2 (100.00%) bins.\n",
      "\n",
      "  Extracting marker genes to align.\n",
      "  Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 2 of 2 (100.00%) bins.\n",
      "  Extracting 43 HMMs with 8 threads:\n",
      "    Finished extracting 43 of 43 (100.00%) HMMs.\n",
      "  Aligning 43 marker genes with 8 threads:\n",
      "    Finished aligning 43 of 43 (100.00%) marker genes.\n",
      "\n",
      "  Reading marker alignment files.\n",
      "  Concatenating alignments.\n",
      "  Placing 2 bins into the genome tree with pplacer (be patient).\n",
      "\n",
      "  { Current stage: 0:06:43.140 || Total: 0:06:43.140 }\n",
      "\n",
      "*******************************************************************************\n",
      " [CheckM - lineage_set] Inferring lineage-specific marker sets.\n",
      "*******************************************************************************\n",
      "\n",
      "  Reading HMM info from file.\n",
      "  Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 2 of 2 (100.00%) bins.\n",
      "\n",
      "  Determining marker sets for each genome bin.\n",
      "    Finished processing 2 of 2 (100.00%) bins (current: metabat2_low_PE.478.contigs). \n",
      "\n",
      "  Marker set written to: /ebio/abt3_projects/vadinCA11/data/V11/checkm_output/lineage.ms\n",
      "\n",
      "  { Current stage: 0:00:01.718 || Total: 0:06:44.858 }\n",
      "\n",
      "*******************************************************************************\n",
      " [CheckM - analyze] Identifying marker genes in bins.\n",
      "*******************************************************************************\n",
      "\n",
      "  Identifying marker genes in 2 bins with 8 threads:\n",
      "    Finished processing 2 of 2 (100.00%) bins.\n",
      "  Saving HMM info to file.\n",
      "\n",
      "  { Current stage: 0:00:59.252 || Total: 0:07:44.110 }\n",
      "\n",
      "  Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 2 of 2 (100.00%) bins.\n",
      "  Aligning marker genes with multiple hits in a single bin:\n",
      "    Finished processing 2 of 2 (100.00%) bins.\n",
      "\n",
      "  { Current stage: 0:00:02.873 || Total: 0:07:46.983 }\n",
      "\n",
      "  Calculating genome statistics for 2 bins with 8 threads:\n",
      "    Finished processing 2 of 2 (100.00%) bins.\n",
      "\n",
      "  { Current stage: 0:00:00.178 || Total: 0:07:47.162 }\n",
      "\n",
      "*******************************************************************************\n",
      " [CheckM - qa] Tabulating genome statistics.\n",
      "*******************************************************************************\n",
      "\n",
      "  Calculating AAI between multi-copy marker genes.\n",
      "\n",
      "  Reading HMM info from file.\n",
      "  Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 2 of 2 (100.00%) bins.\n",
      "\n",
      "  QA information written to: /ebio/abt3_projects/vadinCA11/data/V11/checkm_output/CheckM_V11.txt\n",
      "\n",
      "  { Current stage: 0:00:01.527 || Total: 0:07:48.690 }\n"
     ]
    }
   ],
   "source": [
    "# Running CheckM to verify the completness and redundancy of the V11 bin\n",
    "#\n",
    "# CheckM requires a folder with the bins and doesn't allow a single file as input\n",
    "# Therefore, I provided the folder with all the bins as input, but especified \n",
    "# that the extension of the file is \"478.contigs.fa\", limiting the analysis to two files\n",
    "# One of which is the vadinCA11 bin I'm interested in\n",
    "#\n",
    "checkm_file = os.path.join(work_dir, \"checkm_output\", 'CheckM_V11.txt')\n",
    "checkm_cmd = \"\"\"checkm lineage_wf -t {0} \\\n",
    "    -x {1} \\\n",
    "    -f {2} \\\n",
    "    {3} {4}/checkm_output \"\"\"\n",
    "checkm_cmd = checkm_cmd.format(8, \"478.contigs.fa\", checkm_file, V11_folder, work_dir)\n",
    "checkm_job = 'bash -c \"source activate {0}; {1}\"'\n",
    "checkm_job = checkm_job.format(quality_env, checkm_cmd)\n",
    "print(checkm_job)\n",
    "!$checkm_job"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CheckM results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-05T14:38:27.592529Z",
     "start_time": "2018-06-05T14:38:27.563135Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stat</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Completeness</th>\n",
       "      <td>98.3871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Contamination</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genome size</th>\n",
       "      <td>1590932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># ambiguous bases</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mean contig length</th>\n",
       "      <td>33144.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50 (contigs)</th>\n",
       "      <td>50438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longest contig</th>\n",
       "      <td>103078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># scaffolds</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mean scaffold length</th>\n",
       "      <td>33144.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50 (scaffolds)</th>\n",
       "      <td>50438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longest scaffold</th>\n",
       "      <td>103078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC</th>\n",
       "      <td>0.559415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC std</th>\n",
       "      <td>0.016983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes</th>\n",
       "      <td>1605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translation table</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coding density</th>\n",
       "      <td>0.892202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>marker lineage</th>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># genomes</th>\n",
       "      <td>148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># markers</th>\n",
       "      <td>187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># marker sets</th>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Value\n",
       "Stat                                  \n",
       "Completeness                   98.3871\n",
       "Contamination                        0\n",
       "Genome size                    1590932\n",
       "# ambiguous bases                    0\n",
       "# contigs                           48\n",
       "Mean contig length             33144.4\n",
       "N50 (contigs)                    50438\n",
       "Longest contig                  103078\n",
       "# scaffolds                         48\n",
       "Mean scaffold length           33144.4\n",
       "N50 (scaffolds)                  50438\n",
       "Longest scaffold                103078\n",
       "GC                            0.559415\n",
       "GC std                        0.016983\n",
       "# predicted genes                 1605\n",
       "Translation table                   11\n",
       "Coding density                0.892202\n",
       "marker lineage        p__Euryarchaeota\n",
       "# genomes                          148\n",
       "# markers                          187\n",
       "# marker sets                      124"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkm_stats = \"{0}/checkm_output/storage/bin_stats_ext.tsv\".format(work_dir)\n",
    "checkm_df = pd.read_csv(checkm_stats, sep='\\t', skiprows=0, header=None)\n",
    "checkm_df = eval(checkm_df.iat[0,1]) # Select only the stats from the V11 bin and turn into dict\n",
    "checkm_df = pd.DataFrame(list(checkm_df.items()))\n",
    "checkm_df.columns = [\"Stat\", \"Value\"]\n",
    "checkm_df = checkm_df.set_index('Stat')\n",
    "checkm_df.loc[[\"Completeness\", \"Contamination\", \"Genome size\", \"# ambiguous bases\", \n",
    "               \"# contigs\", \"Mean contig length\", \"N50 (contigs)\", \"Longest contig\",\n",
    "               \"# scaffolds\", \"Mean scaffold length\", \"N50 (scaffolds)\",\n",
    "               \"Longest scaffold\", \"GC\", \"GC std\", \"# predicted genes\", \n",
    "               \"Translation table\", \"Coding density\", \"marker lineage\", \"# genomes\",\n",
    "               \"# markers\", \"# marker sets\" ]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quast"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Running Quast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-06T10:42:46.452637Z",
     "start_time": "2018-06-06T10:42:20.808445Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate py2_genome_quality; quast /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/metabat2_low_PE.478.contigs.fa     -o /ebio/abt3_projects/vadinCA11/data/V11/quast_output     --gene-finding     --glimmer     --threads 8\"\n",
      "/ebio/abt3_projects/software/miniconda3/envs/py2_genome_quality/opt/quast-4.6.3/quast.py /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/metabat2_low_PE.478.contigs.fa -o /ebio/abt3_projects/vadinCA11/data/V11/quast_output --gene-finding --glimmer --threads 8\n",
      "\n",
      "Version: 4.6.3\n",
      "\n",
      "System information:\n",
      "  OS: Linux-4.4.67-x86_64-with-debian-stretch-sid (linux_64)\n",
      "  Python version: 2.7.14\n",
      "  CPUs number: 80\n",
      "\n",
      "Started: 2018-06-06 12:42:26\n",
      "\n",
      "Logging to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/quast.log\n",
      "NOTICE: Output directory already exists. Existing Nucmer alignments can be used\n",
      "\n",
      "CWD: /ebio/abt3_projects/vadinCA11/notebooks/metagenome/assembly/HiSeqRuns83-91\n",
      "Main parameters: \n",
      "  Threads: 8, minimum contig length: 500, ambiguity: one, threshold for extensive misassembly size: 1000\n",
      "\n",
      "Contigs:\n",
      "  Pre-processing...\n",
      "  /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/metabat2_low_PE.478.contigs.fa ==> metabat2_low_PE.478.contigs\n",
      "\n",
      "2018-06-06 12:42:29\n",
      "Running Basic statistics processor...\n",
      "  Contig files: \n",
      "    metabat2_low_PE.478.contigs\n",
      "  Calculating N50 and L50...\n",
      "    metabat2_low_PE.478.contigs, N50 = 50438, L50 = 13, Total length = 1590932, GC % = 55.94, # N's per 100 kbp =  0.00\n",
      "  Drawing Nx plot...\n",
      "    saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/basic_stats/Nx_plot.pdf\n",
      "  Drawing cumulative plot...\n",
      "    saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/basic_stats/cumulative_plot.pdf\n",
      "  Drawing GC content plot...\n",
      "    saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/basic_stats/GC_content_plot.pdf\n",
      "  Drawing metabat2_low_PE.478.contigs GC content plot...\n",
      "    saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/basic_stats/metabat2_low_PE.478.contigs_GC_content_plot.pdf\n",
      "Done.\n",
      "\n",
      "2018-06-06 12:42:31\n",
      "Running GlimmerHMM...\n",
      "  metabat2_low_PE.478.contigs\n",
      "    Genes = 1477 unique, 1477 total\n",
      "    Predicted genes (GFF): /ebio/abt3_projects/vadinCA11/data/V11/quast_output/predicted_genes/metabat2_low_PE-478-contigs_glimmer_genes.gff.gz\n",
      "Done.\n",
      "\n",
      "2018-06-06 12:42:45\n",
      "Creating large visual summaries...\n",
      "This may take a while: press Ctrl-C to skip this step..\n",
      "  1 of 2: Creating Icarus viewers...\n",
      "  2 of 2: Creating PDF with all tables and plots...\n",
      "Done\n",
      "\n",
      "2018-06-06 12:42:46\n",
      "RESULTS:\n",
      "  Text versions of total report are saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/report.txt, report.tsv, and report.tex\n",
      "  Text versions of transposed total report are saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
      "  HTML version (interactive tables and plots) saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/report.html\n",
      "  PDF version (tables and plots) is saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/report.pdf\n",
      "  Icarus (contig browser) is saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/icarus.html\n",
      "  Log saved to /ebio/abt3_projects/vadinCA11/data/V11/quast_output/quast.log\n",
      "\n",
      "Finished: 2018-06-06 12:42:46\n",
      "Elapsed time: 0:00:19.690517\n",
      "NOTICEs: 1; WARNINGs: 0; non-fatal ERRORs: 0\n",
      "\n",
      "Thank you for using QUAST!\n"
     ]
    }
   ],
   "source": [
    "# Running Quast on the bin\n",
    "quast_cmd = \"\"\"quast {0} \\\n",
    "    -o {1}/quast_output \\\n",
    "    --gene-finding \\\n",
    "    --glimmer \\\n",
    "    --threads {2}\"\"\"\n",
    "quast_cmd = quast_cmd.format(V11_bin, work_dir, 8)\n",
    "quast_job = 'bash -c \"source activate {0}; {1}\"'\n",
    "quast_job = quast_job.format(quality_env, quast_cmd)\n",
    "print(quast_job)\n",
    "!$quast_job"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Quast results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-06T10:47:42.736493Z",
     "start_time": "2018-06-06T10:47:42.711111Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>metabat2_low_PE.478.contigs</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembly</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th># contigs (&gt;= 0 bp)</th>\n",
       "      <td>48.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs (&gt;= 1000 bp)</th>\n",
       "      <td>48.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs (&gt;= 5000 bp)</th>\n",
       "      <td>47.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs (&gt;= 10000 bp)</th>\n",
       "      <td>40.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs (&gt;= 25000 bp)</th>\n",
       "      <td>25.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <td>14.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total length (&gt;= 0 bp)</th>\n",
       "      <td>1590932.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total length (&gt;= 1000 bp)</th>\n",
       "      <td>1590932.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total length (&gt;= 5000 bp)</th>\n",
       "      <td>1587803.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total length (&gt;= 10000 bp)</th>\n",
       "      <td>1532732.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total length (&gt;= 25000 bp)</th>\n",
       "      <td>1286989.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total length (&gt;= 50000 bp)</th>\n",
       "      <td>891649.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs</th>\n",
       "      <td>48.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Largest contig</th>\n",
       "      <td>103078.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total length</th>\n",
       "      <td>1590932.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC (%)</th>\n",
       "      <td>55.94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50</th>\n",
       "      <td>50438.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N75</th>\n",
       "      <td>31277.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L50</th>\n",
       "      <td>13.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L75</th>\n",
       "      <td>22.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># N's per 100 kbp</th>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes (unique)</th>\n",
       "      <td>1477.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes (&gt;= 0 bp)</th>\n",
       "      <td>1477.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes (&gt;= 300 bp)</th>\n",
       "      <td>1346.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes (&gt;= 1500 bp)</th>\n",
       "      <td>193.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes (&gt;= 3000 bp)</th>\n",
       "      <td>28.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                metabat2_low_PE.478.contigs\n",
       "Assembly                                                   \n",
       "# contigs (>= 0 bp)                                   48.00\n",
       "# contigs (>= 1000 bp)                                48.00\n",
       "# contigs (>= 5000 bp)                                47.00\n",
       "# contigs (>= 10000 bp)                               40.00\n",
       "# contigs (>= 25000 bp)                               25.00\n",
       "# contigs (>= 50000 bp)                               14.00\n",
       "Total length (>= 0 bp)                           1590932.00\n",
       "Total length (>= 1000 bp)                        1590932.00\n",
       "Total length (>= 5000 bp)                        1587803.00\n",
       "Total length (>= 10000 bp)                       1532732.00\n",
       "Total length (>= 25000 bp)                       1286989.00\n",
       "Total length (>= 50000 bp)                        891649.00\n",
       "# contigs                                             48.00\n",
       "Largest contig                                    103078.00\n",
       "Total length                                     1590932.00\n",
       "GC (%)                                                55.94\n",
       "N50                                                50438.00\n",
       "N75                                                31277.00\n",
       "L50                                                   13.00\n",
       "L75                                                   22.00\n",
       "# N's per 100 kbp                                      0.00\n",
       "# predicted genes (unique)                          1477.00\n",
       "# predicted genes (>= 0 bp)                         1477.00\n",
       "# predicted genes (>= 300 bp)                       1346.00\n",
       "# predicted genes (>= 1500 bp)                       193.00\n",
       "# predicted genes (>= 3000 bp)                        28.00"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "quast_stats = \"{0}/quast_output/report.tsv\".format(work_dir)\n",
    "quast_df = pd.read_csv(quast_stats, sep='\\t', skiprows=0)\n",
    "quast_df = quast_df.set_index('Assembly')\n",
    "quast_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# tRNA gene content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-06T10:55:29.624560Z",
     "start_time": "2018-06-06T10:55:24.205714Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate metacompass; aragorn -gc11 -fons -o /ebio/abt3_projects/vadinCA11/data/V11/tRNA_output/V11_tRNAs.fasta /ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/metabat2_low_PE.478.contigs.fa\"\n"
     ]
    }
   ],
   "source": [
    "# Running aragorn on the bin\n",
    "aragorn_file = os.path.join(work_dir, \"tRNA_output\", 'V11_tRNAs.fasta')\n",
    "aragorn_cmd = \"\"\"aragorn -gc11 -fons -o {0} {1}\"\"\"\n",
    "aragorn_cmd = aragorn_cmd.format(aragorn_file, V11_bin)\n",
    "aragorn_job = 'bash -c \"source activate {0}; {1}\"'\n",
    "aragorn_job = aragorn_job.format(metacompass_env, aragorn_cmd)\n",
    "print(aragorn_job)\n",
    "!$aragorn_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-06T10:56:54.072561Z",
     "start_time": "2018-06-06T10:56:53.822892Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "40\n",
      ">1-1tRNA-Thr(tgt)c[72,148]\n",
      ">1-2tRNA-Arg(cct)[2895,2972]\n",
      ">2-1tRNA-Leu(tag)[44244,44329]\n",
      ">3-1tRNA-Ser(tga)[46845,46930]\n",
      ">4-1tRNA-Ser(gga)c[15630,15716]\n",
      ">5-1tRNA-Ser(gct)[1523,1608]\n",
      ">5-2tRNA-Leu(taa)c[64902,64987]\n",
      ">6-1tRNA-Pyl(cta)c[29512,29582]\n",
      ">7-1tRNA-Gly(tcc)[33795,33870]\n",
      ">10-1tRNA-Asp(gtc)[29110,29185]\n",
      ">11-1tRNA-Met(cat)c[12359,12436]\n",
      ">11-2tRNA-Gln(ctg)c[50830,50905]\n",
      ">11-3tRNA-Ala(ggc)[72006,72079]\n",
      ">11-4tRNA-Glu(ctc)[102867,102940]\n",
      ">12-1tRNA-Pro(ggg)c[7914,7989]\n",
      ">14-1tRNA-Pro(tgg)c[3136,3209]\n",
      ">16-1tRNA-Thr(tgt)c[55396,6]\n",
      ">20-1tRNA-Arg(gcg)[16426,16502]\n",
      ">21-1tRNA-Gln(ttg)c[1842,1917]\n",
      ">21-2tRNA-Ala(tgc)[14706,14779]\n",
      ">21-3tRNA-Leu(caa)[46349,46434]\n",
      ">21-4tRNA-His(gtg)c[84599,84673]\n",
      ">22-1tRNA-Arg(ccg)c[8353,8427]\n",
      ">24-1tRNA-Gly(gcc)[3797,3871]\n",
      ">27-1tRNA-Leu(cag)[384,470]\n",
      ">27-2tRNA-Glu(ttc)[18094,18169]\n",
      ">28-1tRNA-Val(gac)[7911,7986]\n",
      ">28-2tRNA-Ser(cga)[11120,11205]\n",
      ">30-1tRNA-Thr(cgt)[31738,31812]\n",
      ">30-2tRNA-Phe(gaa)[38091,38166]\n",
      ">30-3tRNA-Met(cat)c[45494,45570]\n",
      ">30-4tRNA-Asn(gtt)c[45673,45748]\n",
      ">32-1tRNA-Lys(ttt)[4085,4159]\n",
      ">33-1tRNA-Pro(cgg)[186,259]\n",
      ">37-1tRNA-Cys(gca)[28519,28590]\n",
      ">41-1tRNA-Leu(gag)c[269,355]\n",
      ">43-1tRNA-Lys(ctt)[22253,22329]\n",
      ">44-1tRNA-Thr(ggt)[2359,2433]\n",
      ">44-2tRNA-Arg(tcg)[5082,5159]\n",
      ">47-1tRNA-Ala(cgc)[11985,12061]\n"
     ]
    }
   ],
   "source": [
    "# Count and print names of the tRNA genes detected\n",
    "!grep -c \">\" $aragorn_file\n",
    "!grep \">\" $aragorn_file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Of the 20 standard aminoacids, 17 were found; Isoleucine (Ile), Tyrosine (Tyr) and Tryptophan (Trp) were not detected. In addition, a tRNA for the non-standar amino acid pyrrolysine  was also detected\t"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-21T11:26:24.581462Z",
     "start_time": "2018-09-21T11:26:23.725366Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# packages in environment at /ebio/abt3_projects/software/miniconda3/envs/py2_genome_quality:\n",
      "#\n",
      "backports                 1.0                      py27_1    conda-forge\n",
      "backports.functools_lru_cache 1.5                      py27_0    conda-forge\n",
      "backports_abc             0.5                      py27_0    conda-forge\n",
      "bcftools                  1.6                           1    bioconda\n",
      "blast                     2.7.1               boost1.64_3    bioconda\n",
      "boost                     1.64.0                   py27_4    conda-forge\n",
      "boost-cpp                 1.64.0                        1    conda-forge\n",
      "bz2file                   0.98                     py27_0  \n",
      "bzip2                     1.0.6                         1    conda-forge\n",
      "ca-certificates           2018.1.18                     0    conda-forge\n",
      "centrifuge                1.0.3            py27pl5.22.0_2    bioconda\n",
      "certifi                   2018.1.18                py27_0    conda-forge\n",
      "checkm-genome             1.0.7                    py27_0    bioconda\n",
      "curl                      7.55.1                        0    conda-forge\n",
      "cycler                    0.10.0                   py27_0    conda-forge\n",
      "cython                    0.27.3                   py27_0    conda-forge\n",
      "dbus                      1.10.22                       0    conda-forge\n",
      "dendropy                  4.2.0                    py27_0    bioconda\n",
      "expat                     2.2.5                         0    conda-forge\n",
      "fontconfig                2.12.6                        0    conda-forge\n",
      "freetype                  2.8.1                         0    conda-forge\n",
      "functools32               3.2.3.2                  py27_2    conda-forge\n",
      "gettext                   0.19.8.1                      0    conda-forge\n",
      "glib                      2.55.0                        0    conda-forge\n",
      "gmp                       6.1.2                         0    conda-forge\n",
      "gnutls                    3.5.17                        0    conda-forge\n",
      "gst-plugins-base          1.8.0                         0    conda-forge\n",
      "gstreamer                 1.8.0                         1    conda-forge\n",
      "htslib                    1.7                           0    bioconda\n",
      "icu                       58.2                          0    conda-forge\n",
      "ijson                     2.3                      py27_0    conda-forge\n",
      "intel-openmp              2018.0.0             hc7b2577_8  \n",
      "joblib                    0.11                     py27_0    conda-forge\n",
      "jpeg                      9b                            2    conda-forge\n",
      "khmer                     2.1.2                    py27_0    bioconda\n",
      "krb5                      1.14.2                        0    conda-forge\n",
      "libffi                    3.2.1                         3    conda-forge\n",
      "libgcc                    7.2.0                h69d50b8_2  \n",
      "libgcc-ng                 7.2.0                h7cc24e2_2  \n",
      "libgfortran-ng            7.2.0                h9f7466a_2  \n",
      "libiconv                  1.15                          0    conda-forge\n",
      "libidn11                  1.33                          0    conda-forge\n",
      "libpng                    1.6.34                        0    conda-forge\n",
      "libssh2                   1.8.0                         2    conda-forge\n",
      "libstdcxx-ng              7.2.0                h7a57d05_2  \n",
      "libxcb                    1.12                          1    conda-forge\n",
      "libxml2                   2.9.7                         0    conda-forge\n",
      "matplotlib                2.1.2                    py27_0    conda-forge\n",
      "mkl                       2018.0.1             h19d6760_4  \n",
      "ncurses                   5.9                          10    conda-forge\n",
      "nettle                    3.3                           0    conda-forge\n",
      "numpy                     1.14.0           py27h3dfced4_1  \n",
      "openjdk                   8.0.121                       1  \n",
      "openssl                   1.0.2n                        0    conda-forge\n",
      "pcre                      8.39                          0    conda-forge\n",
      "perl                      5.22.0.1                      0    conda-forge\n",
      "perl-app-cpanminus        1.7043               pl5.22.0_0    bioconda\n",
      "perl-archive-tar          2.18                 pl5.22.0_2    bioconda\n",
      "perl-carp                 1.38                 pl5.22.0_0    bioconda\n",
      "perl-compress-raw-bzip2   2.069                         1    bioconda\n",
      "perl-compress-raw-zlib    2.069                         3    bioconda\n",
      "perl-data-dumper          2.161                pl5.22.0_0    bioconda\n",
      "perl-exporter             5.72                 pl5.22.0_0    bioconda\n",
      "perl-exporter-tiny        0.042                         1    bioconda\n",
      "perl-extutils-makemaker   7.24                 pl5.22.0_1    bioconda\n",
      "perl-io-compress          2.069                pl5.22.0_2    bioconda\n",
      "perl-io-zlib              1.10                          1    bioconda\n",
      "perl-list-moreutils       0.428                pl5.22.0_0    bioconda\n",
      "perl-pathtools            3.73                          0    bioconda\n",
      "perl-scalar-list-utils    1.45                          2    bioconda\n",
      "perl-test-more            1.001002             pl5.22.0_0    bioconda\n",
      "perl-threaded             5.22.0              pl5.22.0_12    bioconda\n",
      "pip                       9.0.1                    py27_1    conda-forge\n",
      "pplacer                   1.1.alpha17                   0    bioconda\n",
      "prodigal                  2.6.3                         0    bioconda\n",
      "pyparsing                 2.2.0                    py27_0    conda-forge\n",
      "pyqt                      5.6.0                    py27_4    conda-forge\n",
      "pysam                     0.14.0          py27_htslib1.7_1    bioconda\n",
      "python                    2.7.14                        4    conda-forge\n",
      "python-dateutil           2.3                      py27_0    bioconda\n",
      "pytz                      2018.3                     py_0    conda-forge\n",
      "pyyaml                    3.12                     py27_1    conda-forge\n",
      "qt                        5.6.2                         7    conda-forge\n",
      "quast                     4.6.3           py27_boost1.64_1    bioconda\n",
      "readline                  7.0                           0    conda-forge\n",
      "samtools                  1.7                           1    bioconda\n",
      "scipy                     1.0.0            py27hf5f0f52_0  \n",
      "screamingbackpack         0.2.333                  py27_0    bioconda\n",
      "screed                    1.0                      py27_0    bioconda\n",
      "seqtk                     1.2                           1    bioconda\n",
      "setuptools                38.5.1                   py27_0    conda-forge\n",
      "simplejson                3.8.1                    py27_0    bioconda\n",
      "singledispatch            3.4.0.3                  py27_0    conda-forge\n",
      "sip                       4.18                     py27_1    conda-forge\n",
      "six                       1.11.0                   py27_1    conda-forge\n",
      "sourmash                  2.0.0a3                  py27_0    bioconda\n",
      "sqlite                    3.20.1                        2    conda-forge\n",
      "ssl_match_hostname        3.5.0.1                  py27_1    conda-forge\n",
      "subprocess32              3.2.7                    py27_0    conda-forge\n",
      "tk                        8.6.7                         0    conda-forge\n",
      "tornado                   4.5.3                    py27_0    conda-forge\n",
      "wheel                     0.30.0                   py27_2    conda-forge\n",
      "xorg-libxau               1.0.8                         3    conda-forge\n",
      "xorg-libxdmcp             1.1.2                         3    conda-forge\n",
      "xz                        5.2.3                         0    conda-forge\n",
      "yaml                      0.1.6                         0    conda-forge\n",
      "zlib                      1.2.11                        0    conda-forge\n"
     ]
    }
   ],
   "source": [
    "!conda list -n py2_genome_quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-21T11:26:25.553271Z",
     "start_time": "2018-09-21T11:26:24.583820Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# packages in environment at /ebio/abt3_projects/software/miniconda3/envs/anvio:\n",
      "#\n",
      "anvio                     4.0.0                    py35_1    bioconda\n",
      "asn1crypto                0.22.0                   py35_0    conda-forge\n",
      "bcftools                  1.4.1                         0    bioconda\n",
      "blast                     2.2.31                        1    bioconda\n",
      "boost                     1.65.1                   py35_0    conda-forge\n",
      "boost-cpp                 1.65.1                        1    conda-forge\n",
      "bottle                    0.12.13                  py35_0    conda-forge\n",
      "bwa                       0.7.15                        1    bioconda\n",
      "bzip2                     1.0.6                         1    conda-forge\n",
      "ca-certificates           2017.11.5                     0    conda-forge\n",
      "cairo                     1.14.6                        5    conda-forge\n",
      "centrifuge                1.0.3            py35pl5.22.0_1    bioconda\n",
      "certifi                   2017.11.5                py35_0    conda-forge\n",
      "cffi                      1.10.0                   py35_0    conda-forge\n",
      "chardet                   3.0.4                    py35_0    conda-forge\n",
      "cherrypy                  8.0.0                    py35_0    conda-forge\n",
      "cryptography              2.0.3                    py35_0    conda-forge\n",
      "curl                      7.55.1                        0    conda-forge\n",
      "cycler                    0.10.0                   py35_0    conda-forge\n",
      "cython                    0.27.3                   py35_0    conda-forge\n",
      "diamond                   0.9.10                        0    bioconda\n",
      "django                    2.0.2                    py35_0    conda-forge\n",
      "ete3                      3.1.1                    py35_0    bioconda\n",
      "fasttree                  2.1.10                        0    bioconda\n",
      "fontconfig                2.12.1                        5    conda-forge\n",
      "freetype                  2.7                           2    conda-forge\n",
      "gettext                   0.19.7                        1    conda-forge\n",
      "glib                      2.51.4                        0    conda-forge\n",
      "graphite2                 1.3.9                         0    conda-forge\n",
      "gsl                       1.16                          0    conda-forge\n",
      "h5py                      2.7.1                    py35_2    conda-forge\n",
      "harfbuzz                  1.4.3                         0    conda-forge\n",
      "hdf5                      1.10.1                        1    conda-forge\n",
      "hmmer                     3.1b2                         3    bioconda\n",
      "htslib                    1.4.1                         0    bioconda\n",
      "icu                       58.2                          0    conda-forge\n",
      "idna                      2.6                      py35_1    conda-forge\n",
      "intel-openmp              2018.0.0             h15fc484_7  \n",
      "jpeg                      9b                            2    conda-forge\n",
      "krb5                      1.14.2                        0    conda-forge\n",
      "libffi                    3.2.1                         3    conda-forge\n",
      "libgcc                    7.2.0                h69d50b8_2  \n",
      "libgcc-ng                 7.2.0                h7cc24e2_2  \n",
      "libgfortran               3.0.0                         1  \n",
      "libgfortran-ng            7.2.0                h9f7466a_2  \n",
      "libiconv                  1.14                          4    conda-forge\n",
      "libpng                    1.6.28                        2    conda-forge\n",
      "libssh2                   1.8.0                         2    conda-forge\n",
      "libstdcxx-ng              7.2.0                h7a57d05_2  \n",
      "libtiff                   4.0.8                         0    conda-forge\n",
      "libuuid                   1.0.3                         1    conda-forge\n",
      "libxcb                    1.12                          1    conda-forge\n",
      "libxml2                   2.9.5                         1    conda-forge\n",
      "libxslt                   1.1.29                        5    conda-forge\n",
      "lxml                      4.1.1                    py35_0    conda-forge\n",
      "matplotlib                2.0.0               np112py35_2    conda-forge\n",
      "mcl                       14.137                        2    bioconda\n",
      "mistune                   0.7.4                    py35_0    conda-forge\n",
      "mkl                       2018.0.0             hb491cac_4  \n",
      "muscle                    3.8.1551                      2    bioconda\n",
      "ncurses                   5.9                          10    conda-forge\n",
      "numpy                     1.12.1           py35hca0bb5e_1  \n",
      "openblas                  0.2.20                        4    conda-forge\n",
      "openssl                   1.0.2l                        0    conda-forge\n",
      "pandas                    0.20.1              np112py35_0    conda-forge\n",
      "pango                     1.40.4                        0    conda-forge\n",
      "pcre                      8.39                          0    conda-forge\n",
      "perl                      5.22.0.1                      0    conda-forge\n",
      "perl-threaded             5.22.0              pl5.22.0_12    bioconda\n",
      "pip                       9.0.1                    py35_0    conda-forge\n",
      "pixman                    0.34.0                        0    conda-forge\n",
      "prodigal                  2.6.3                         0    bioconda\n",
      "psutil                    5.4.0                    py35_0    conda-forge\n",
      "pycparser                 2.18                     py35_0    conda-forge\n",
      "pyopenssl                 17.2.0                   py35_0    conda-forge\n",
      "pyparsing                 2.2.0                    py35_0    conda-forge\n",
      "pyqt                      4.11.4                   py35_2    conda-forge\n",
      "pysam                     0.11.1                   py35_0    bioconda\n",
      "pysocks                   1.6.7                    py35_0    conda-forge\n",
      "python                    3.5.4                         1    conda-forge\n",
      "python-dateutil           2.3                      py35_0    bioconda\n",
      "pytz                      2017.3                     py_2    conda-forge\n",
      "qt                        4.8.7                         7    conda-forge\n",
      "readline                  6.2                           0    conda-forge\n",
      "requests                  2.18.4                   py35_1    conda-forge\n",
      "samtools                  1.4.1                         0    bioconda\n",
      "scikit-learn              0.19.1           py35hbf1f462_0  \n",
      "scipy                     1.0.0            py35hcbbe4a2_0  \n",
      "setuptools                36.6.0                   py35_1    conda-forge\n",
      "sip                       4.18                     py35_1    conda-forge\n",
      "six                       1.10.0                   py35_1    conda-forge\n",
      "sqlite                    3.13.0                        1    conda-forge\n",
      "taxonkit                  0.2.0                         0    bioconda\n",
      "tk                        8.5.19                        2    conda-forge\n",
      "tornado                   5.0.1                    py35_1    conda-forge\n",
      "urllib3                   1.22                     py35_0    conda-forge\n",
      "wheel                     0.30.0                     py_1    conda-forge\n",
      "xorg-kbproto              1.0.7                         1    conda-forge\n",
      "xorg-libice               1.0.9                         2    conda-forge\n",
      "xorg-libsm                1.2.2                         2    conda-forge\n",
      "xorg-libx11               1.6.4                         6    conda-forge\n",
      "xorg-libxau               1.0.8                         3    conda-forge\n",
      "xorg-libxdmcp             1.1.2                         3    conda-forge\n",
      "xorg-libxext              1.3.3                         2    conda-forge\n",
      "xorg-libxrender           0.9.10                        0    conda-forge\n",
      "xorg-renderproto          0.11.1                        1    conda-forge\n",
      "xorg-xextproto            7.3.0                         1    conda-forge\n",
      "xorg-xproto               7.0.31                        6    conda-forge\n",
      "xz                        5.2.3                         0    conda-forge\n",
      "zlib                      1.2.11                        0    conda-forge\n"
     ]
    }
   ],
   "source": [
    "!conda list -n anvio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-21T11:26:27.121323Z",
     "start_time": "2018-09-21T11:26:25.555600Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# packages in environment at /ebio/abt3_projects/software/miniconda3/envs/metacompass:\n",
      "#\n",
      "anvio                     4.0.0                    py35_2    bioconda\n",
      "aragorn                   1.2.38                        1    bioconda\n",
      "asn1crypto                0.24.0                   py35_0  \n",
      "atomicwrites              1.1.5                    py35_0  \n",
      "attrs                     18.1.0                   py35_0  \n",
      "backcall                  0.1.0                    py35_0  \n",
      "barrnap                   0.9                           0    bioconda\n",
      "bcftools                  1.4.1                         0    bioconda\n",
      "bcrypt                    3.1.4            py35ha35c455_0  \n",
      "bedtools                  2.27.1                        1    bioconda\n",
      "biom-format               2.1.6                    py35_1    bioconda\n",
      "biopython                 1.68                     py35_0    bioconda\n",
      "blast                     2.7.1                h96bfa4b_5    bioconda\n",
      "blast-legacy              2.2.26                        0    bioconda\n",
      "blast-plus                2.2.31                        0    biocore\n",
      "bleach                    2.1.3                    py35_0  \n",
      "boost                     1.66.0                   py35_1    conda-forge\n",
      "boost-cpp                 1.66.0                        1    conda-forge\n",
      "bottle                    0.12.13                  py35_0    conda-forge\n",
      "bowtie2                   2.3.0                    py35_1    bioconda\n",
      "bracken                   1.0.0                         1    bioconda\n",
      "bwa                       0.7.17               ha92aebf_3    bioconda\n",
      "bz2file                   0.98                     py35_0  \n",
      "bzip2                     1.0.6                h9a117a8_4  \n",
      "ca-certificates           2018.03.07                    0  \n",
      "cairo                     1.14.12              h77bcde2_0  \n",
      "capnproto                 0.6.1                         0    conda-forge\n",
      "centrifuge                1.0.4_beta       py35pl5.22.0_0    bioconda\n",
      "certifi                   2018.4.16                py35_0  \n",
      "cffi                      1.11.5           py35h9745a5d_0  \n",
      "cherrypy                  8.0.0                    py35_0    conda-forge\n",
      "click                     6.7                        py_1    conda-forge\n",
      "cryptography              2.2.2            py35h14c3975_0  \n",
      "curl                      7.59.0               h84994c4_0  \n",
      "cycler                    0.10.0           py35hc4d5149_0  \n",
      "cython                    0.28.3           py35h14c3975_0  \n",
      "dbus                      1.13.2               hc3f9b76_0  \n",
      "decorator                 4.3.0                    py35_0  \n",
      "dendropy                  4.2.0                    py35_0    bioconda\n",
      "diamond                   0.9.22               h065a656_0    bioconda\n",
      "django                    2.0.2            py35hd476221_0  \n",
      "docutils                  0.14             py35hd11081d_0  \n",
      "drep                      2.0.5                    py35_0    bioconda\n",
      "dropbox                   5.2.1                    py35_0    bioconda\n",
      "entrypoints               0.2.3            py35h48174a2_2  \n",
      "ete3                      3.1.1                    py35_0    bioconda\n",
      "expat                     2.2.5                he0dffb1_0  \n",
      "fasttree                  2.1.10                        0    bioconda\n",
      "filechunkio               1.6                      py35_0    bioconda\n",
      "fontconfig                2.12.6               h49f89f6_0  \n",
      "freetype                  2.8                  hab7d2ae_1  \n",
      "ftputil                   3.2                      py35_0    bioconda\n",
      "future                    0.16.0                   py35_2    conda-forge\n",
      "gblocks                   0.91b                         0    bioconda\n",
      "glib                      2.53.6               h5d9569c_2  \n",
      "gmp                       6.1.2                h6c8ec71_1  \n",
      "gnutls                    3.5.17                        0    conda-forge\n",
      "graphite2                 1.3.11               h16798f4_2  \n",
      "gsl                       2.2.1                h0c605f7_3  \n",
      "gst-plugins-base          1.12.4               h33fb286_0  \n",
      "gstreamer                 1.12.4               hb53b477_0  \n",
      "h5py                      2.8.0            py35ha1f6525_0  \n",
      "harfbuzz                  1.7.6                hc5b324e_0  \n",
      "hdf5                      1.10.2               hba1933b_1  \n",
      "hmmer                     3.1b2                         0    biocore\n",
      "html5lib                  1.0.1            py35h2f9c1c0_0  \n",
      "htslib                    1.4.1                         0    bioconda\n",
      "icu                       58.2                 h9c2bf20_1  \n",
      "idna                      2.6              py35h8605a33_1  \n",
      "ijson                     2.3                      py35_0    conda-forge\n",
      "infernal                  1.1.2                         0    bioconda\n",
      "intel-openmp              2018.0.0                      8  \n",
      "ipykernel                 4.8.2                    py35_0  \n",
      "ipython                   6.4.0                    py35_0  \n",
      "ipython_genutils          0.2.0            py35hc9e07d0_0  \n",
      "jedi                      0.12.0                   py35_1  \n",
      "jellyfish                 1.1.12               h2d50403_0    bioconda\n",
      "jinja2                    2.10             py35h480ab6d_0  \n",
      "jpeg                      9b                   h024ee3a_2  \n",
      "jsonschema                2.6.0            py35h4395190_0  \n",
      "jupyter_client            5.2.3                    py35_0  \n",
      "jupyter_core              4.4.0            py35ha89e94b_0  \n",
      "khmer                     2.1.2                    py35_0    bioconda\n",
      "kiwisolver                1.0.1            py35hcb1117a_0  \n",
      "kraken                    1.1                  h470a237_2    bioconda\n",
      "kraken-biom               1.0.1                      py_2    bioconda\n",
      "libcurl                   7.59.0               h1ad7b7a_0  \n",
      "libdeflate                0.8                           0    bioconda\n",
      "libedit                   3.1.20170329                  0    conda-forge\n",
      "libffi                    3.2.1                hd88cf55_4  \n",
      "libgcc                    7.2.0                h69d50b8_2  \n",
      "libgcc-ng                 7.2.0                hdf63c60_3  \n",
      "libgfortran-ng            7.2.0                hdf63c60_3  \n",
      "libidn11                  1.33                          0    conda-forge\n",
      "libopenblas               0.2.20               h9ac9557_7  \n",
      "libpng                    1.6.34               hb9fc6fc_0  \n",
      "libsodium                 1.0.16               h1bed415_0  \n",
      "libssh2                   1.8.0                h9cfc8f7_4  \n",
      "libstdcxx-ng              7.2.0                hdf63c60_3  \n",
      "libtiff                   4.0.9                he85c1e1_1  \n",
      "libuuid                   1.0.3                h1bed415_2  \n",
      "libxcb                    1.13                 h1bed415_1  \n",
      "libxml2                   2.9.8                hf84eae3_0  \n",
      "libxslt                   1.1.32               h1312cb7_0  \n",
      "lxml                      4.2.2            py35hf71bdeb_0  \n",
      "markupsafe                1.0              py35h4f4fcf6_1  \n",
      "mash                      2.0                    gsl2.2_2    bioconda\n",
      "matplotlib                2.2.2            py35h0e671d2_1  \n",
      "mcl                       14.137               h470a237_3    bioconda\n",
      "megahit                   1.0.6                    py35_1    bioconda\n",
      "meryl                     2013                          0    bioconda\n",
      "minced                    0.3.0                         0    bioconda\n",
      "mistune                   0.7.4            py35hfd0f961_0  \n",
      "mkl                       2018.0.2                      1  \n",
      "mkl_fft                   1.0.1            py35h3010b51_0  \n",
      "mkl_random                1.0.1            py35h629b387_0  \n",
      "mmtf-python               1.0.2                    py35_0    bioconda\n",
      "more-itertools            4.2.0                    py35_0  \n",
      "msgpack-python            0.5.6            py35h6bb024c_0  \n",
      "mugsy                     1.2.3                         2    bioconda\n",
      "mummer                    3.23                 pl5.22.0_5    bioconda\n",
      "muscle                    3.8.1551                      2    bioconda\n",
      "nb_conda                  2.2.1                    py35_0    conda-forge\n",
      "nb_conda_kernels          2.1.1                    py35_1    conda-forge\n",
      "nbconvert                 5.3.1            py35hc5194e3_0  \n",
      "nbformat                  4.4.0            py35h12e6e07_0  \n",
      "ncbi-genome-download      0.2.6                    py35_0    bioconda\n",
      "ncbi-ngs-sdk              2.9.0                         0    bioconda\n",
      "ncbi-vdb                  2.9.0                         0    bioconda\n",
      "ncurses                   5.9                          10  \n",
      "nettle                    3.3                           0    conda-forge\n",
      "nonpareil                 3.2                    r3.4.1_0    bioconda\n",
      "notebook                  5.5.0                    py35_0  \n",
      "numpy                     1.12.1           py35hca0bb5e_1  \n",
      "numpy-base                1.14.3           py35hdbf6ddf_0  \n",
      "olefile                   0.45.1                   py35_0  \n",
      "openblas                  0.2.20                        4  \n",
      "openblas-devel            0.2.20                        7  \n",
      "openjdk                   8.0.121                       1  \n",
      "openssl                   1.0.2o               h20670df_0  \n",
      "pandas                    0.20.1              np112py35_0  \n",
      "pandaseq                  2.11                 he4cf2ce_3    bioconda\n",
      "pandoc                    1.19.2.1             hea2e7c5_1  \n",
      "pandocfilters             1.4.2            py35h1565a15_1  \n",
      "pango                     1.40.11              h8191d47_0  \n",
      "parallel                  20170422             pl5.22.0_0    bioconda\n",
      "paramiko                  2.4.1                    py35_0  \n",
      "parso                     0.2.1                    py35_0  \n",
      "patsy                     0.5.0                    py35_0  \n",
      "pcre                      8.39                          1  \n",
      "perl                      5.22.0.1                      0    conda-forge\n",
      "perl-app-cpanminus        1.7043               pl5.22.0_0    bioconda\n",
      "perl-archive-tar          2.18                 pl5.22.0_2    bioconda\n",
      "perl-bioperl              1.6.924                       4    bioconda\n",
      "perl-carp                 1.38                 pl5.22.0_0    bioconda\n",
      "perl-compress-raw-bzip2   2.069                         1    bioconda\n",
      "perl-compress-raw-zlib    2.069                         3    bioconda\n",
      "perl-data-dumper          2.161                pl5.22.0_0    bioconda\n",
      "perl-digest-hmac          1.03                 pl5.22.0_1    bioconda\n",
      "perl-encode-locale        1.05                 pl5.22.0_4    bioconda\n",
      "perl-exporter             5.72                 pl5.22.0_0    bioconda\n",
      "perl-exporter-tiny        0.042                         1    bioconda\n",
      "perl-extutils-makemaker   7.24                 pl5.22.0_1    bioconda\n",
      "perl-file-listing         6.04                          0    bioconda\n",
      "perl-html-parser          3.72                 pl5.22.0_1    bioconda\n",
      "perl-html-tagset          3.20                 pl5.22.0_1    bioconda\n",
      "perl-http-cookies         6.01                          0    bioconda\n",
      "perl-http-daemon          6.01                          0    bioconda\n",
      "perl-http-date            6.02                 pl5.22.0_1    bioconda\n",
      "perl-http-message         6.11                          0    bioconda\n",
      "perl-http-negotiate       6.01                          0    bioconda\n",
      "perl-io-compress          2.069                pl5.22.0_2    bioconda\n",
      "perl-io-html              1.001                pl5.22.0_1    bioconda\n",
      "perl-io-socket-ssl        2.024                         0    bioconda\n",
      "perl-io-zlib              1.10                          1    bioconda\n",
      "perl-libwww-perl          6.15                          0    bioconda\n",
      "perl-list-moreutils       0.428                pl5.22.0_0    bioconda\n",
      "perl-lwp-mediatypes       6.02                 pl5.22.0_1    bioconda\n",
      "perl-mime-base64          3.15                 pl5.22.0_0    bioconda\n",
      "perl-net-http             6.09                          0    bioconda\n",
      "perl-net-ssleay           1.84                 pl5.22.0_0    bioconda\n",
      "perl-ntlm                 1.09                          1    bioconda\n",
      "perl-parallel-forkmanager 1.17                          0    bioconda\n",
      "perl-pathtools            3.73                          0    bioconda\n",
      "perl-scalar-list-utils    1.45                          2    bioconda\n",
      "perl-test-more            1.001002             pl5.22.0_0    bioconda\n",
      "perl-threaded             5.22.0              pl5.22.0_12    bioconda\n",
      "perl-uri                  1.71                 pl5.22.0_1    bioconda\n",
      "perl-www-robotrules       6.02                          0    bioconda\n",
      "perl-xml-namespacesupport 1.11                          0    bioconda\n",
      "perl-xml-parser           2.44                          4    bioconda\n",
      "perl-xml-sax              0.99                          0    bioconda\n",
      "perl-xml-sax-base         1.08                          0    bioconda\n",
      "perl-xml-sax-expat        0.51                          0    bioconda\n",
      "perl-xml-simple           2.22                          0    bioconda\n",
      "perl-xsloader             0.22                 pl5.22.0_0    bioconda\n",
      "perl-yaml                 1.24                          0    bioconda\n",
      "pexpect                   4.6.0                    py35_0  \n",
      "pickleshare               0.7.4            py35hd57304d_0  \n",
      "pillow                    5.1.0            py35h3deb7b8_0  \n",
      "pilon                     1.22                     py35_0    bioconda\n",
      "pip                       10.0.1                   py35_0  \n",
      "pixman                    0.34.0               hceecf20_3  \n",
      "pluggy                    0.6.0            py35hb428803_0  \n",
      "prodigal                  2.6.3                         0    bioconda\n",
      "prokka                    1.12                          1    bioconda\n",
      "prompt_toolkit            1.0.15           py35hc09de7a_0  \n",
      "psutil                    5.4.6            py35h14c3975_0  \n",
      "ptyprocess                0.5.2            py35h38ce0a3_0  \n",
      "py                        1.5.3                    py35_0  \n",
      "pyani                     0.2.7                    py35_0    bioconda\n",
      "pyasn1                    0.4.2            py35h8f845f4_0  \n",
      "pycparser                 2.18             py35h61b3040_1  \n",
      "pygments                  2.2.0            py35h0f41973_0  \n",
      "pynacl                    1.2.1            py35h14c3975_0  \n",
      "pyparsing                 2.2.0            py35h041ed72_1  \n",
      "pyqt                      5.9.2            py35h751905a_0  \n",
      "pysam                     0.11.1                   py35_0    bioconda\n",
      "pysftp                    0.2.9                    py35_0    bioconda\n",
      "pytest                    3.6.1                    py35_0  \n",
      "python                    3.5.4               h56e0582_23  \n",
      "python-dateutil           2.7.3                    py35_0    anaconda\n",
      "pytz                      2018.4                   py35_0  \n",
      "pyyaml                    3.12             py35h46ef4ae_1  \n",
      "pyzmq                     17.0.0           py35h14c3975_0  \n",
      "qt                        5.9.4                h4e5bff0_0  \n",
      "r-assertthat              0.2.0                  r3.4.1_0    r\n",
      "r-backports               1.1.0                  r3.4.1_0    r\n",
      "r-base                    3.4.1                         4    conda-forge\n",
      "r-base64enc               0.1_3                  r3.4.1_0    r\n",
      "r-bh                      1.62.0_1               r3.4.1_0    r\n",
      "r-bindr                   0.1                    r3.4.1_0    r\n",
      "r-bindrcpp                0.2                    r3.4.1_0    r\n",
      "r-bit                     1.1_12                 r3.4.1_0    conda-forge\n",
      "r-bit64                   0.9_5                  r3.4.1_0    conda-forge\n",
      "r-bitops                  1.0_6                  r3.4.1_2    r\n",
      "r-blob                    1.1.1                  r3.4.1_0    conda-forge\n",
      "r-boot                    1.3_19                 r3.4.1_0    r\n",
      "r-broom                   0.4.2                  r3.4.1_0    r\n",
      "r-car                     2.1_4                  r3.4.1_0    r\n",
      "r-caret                   6.0_76                 r3.4.1_0    r\n",
      "r-catools                 1.17.1                 r3.4.1_2    r\n",
      "r-cellranger              1.1.0                  r3.4.1_0    r\n",
      "r-class                   7.3_14                 r3.4.1_0    r\n",
      "r-cluster                 2.0.6                  r3.4.1_0    r\n",
      "r-codetools               0.2_15                 r3.4.1_0    r\n",
      "r-colorspace              1.3_2                  r3.4.1_0    r\n",
      "r-crayon                  1.3.2                  r3.4.1_0    r\n",
      "r-curl                    2.6                    r3.4.1_0    r\n",
      "r-data.table              1.10.4                 r3.4.1_0    r\n",
      "r-dbi                     0.6_1                  r3.4.1_0    r\n",
      "r-dbplyr                  1.1.0                  r3.4.1_0    conda-forge\n",
      "r-dichromat               2.0_0                  r3.4.1_2    r\n",
      "r-digest                  0.6.12                 r3.4.1_0    r\n",
      "r-dplyr                   0.7.0                  r3.4.1_0    r\n",
      "r-essentials              1.6.0                  r3.4.1_0    r\n",
      "r-evaluate                0.10                   r3.4.1_0    r\n",
      "r-forcats                 0.2.0                  r3.4.1_0    r\n",
      "r-foreach                 1.4.3                  r3.4.1_0    r\n",
      "r-foreign                 0.8_68                 r3.4.1_0    r\n",
      "r-formatr                 1.5                    r3.4.1_0    r\n",
      "r-ggplot2                 2.2.1                  r3.4.1_0    r\n",
      "r-gistr                   0.4.0                  r3.4.1_0    r\n",
      "r-glmnet                  2.0_10                 r3.4.1_0    r\n",
      "r-glue                    1.1.1                  r3.4.1_0    r\n",
      "r-gridextra               2.3                    r3.4.1_0    conda-forge\n",
      "r-gtable                  0.2.0                  r3.4.1_0    r\n",
      "r-haven                   1.0.0                  r3.4.1_0    r\n",
      "r-hexbin                  1.27.1                 r3.4.1_0    r\n",
      "r-highr                   0.6                    r3.4.1_0    r\n",
      "r-hms                     0.3                    r3.4.1_0    r\n",
      "r-htmltools               0.3.6                  r3.4.1_0    r\n",
      "r-htmlwidgets             0.8                    r3.4.1_1    r\n",
      "r-httpuv                  1.3.3                  r3.4.1_0    r\n",
      "r-httr                    1.2.1                  r3.4.1_0    r\n",
      "r-irdisplay               0.4.4                  r3.4.1_0    r\n",
      "r-irkernel                0.7.1                  r3.4.1_0    r\n",
      "r-iterators               1.0.8                  r3.4.1_0    r\n",
      "r-jsonlite                1.5                    r3.4.1_0    r\n",
      "r-kernsmooth              2.23_15                r3.4.1_0    r\n",
      "r-knitr                   1.16                   r3.4.1_0    r\n",
      "r-labeling                0.3                    r3.4.1_2    r\n",
      "r-lattice                 0.20_35                r3.4.1_0    r\n",
      "r-lazyeval                0.2.0                  r3.4.1_0    r\n",
      "r-lme4                    1.1_13                 r3.4.1_0    r\n",
      "r-lubridate               1.6.0                  r3.4.1_0    r\n",
      "r-magrittr                1.5                    r3.4.1_2    r\n",
      "r-maps                    3.2.0                  r3.4.1_0    r\n",
      "r-markdown                0.8                    r3.4.1_0    r\n",
      "r-mass                    7.3_47                 r3.4.1_0    r\n",
      "r-matrix                  1.2_10                 r3.4.1_0    r\n",
      "r-matrixmodels            0.4_1                  r3.4.1_0    r\n",
      "r-memoise                 1.1.0                  r3.4.1_0    conda-forge\n",
      "r-mgcv                    1.8_17                 r3.4.1_0    r\n",
      "r-mime                    0.5                    r3.4.1_0    r\n",
      "r-minqa                   1.2.4                  r3.4.1_2    r\n",
      "r-mnormt                  1.5_5                  r3.4.1_0    r\n",
      "r-modelmetrics            1.1.0                  r3.4.1_0    r\n",
      "r-modelr                  0.1.0                  r3.4.1_0    r\n",
      "r-munsell                 0.4.3                  r3.4.1_0    r\n",
      "r-nlme                    3.1_131                r3.4.1_0    r\n",
      "r-nloptr                  1.0.4                  r3.4.1_2    r\n",
      "r-nnet                    7.3_12                 r3.4.1_0    r\n",
      "r-openssl                 0.9.6                  r3.4.1_0    r\n",
      "r-pbdzmq                  0.3_2                  r3.4.1_0    conda-forge\n",
      "r-pbkrtest                0.4_7                  r3.4.1_0    r\n",
      "r-pkgconfig               2.0.1                  r3.4.1_0    r\n",
      "r-plogr                   0.1_1                  r3.4.1_0    r\n",
      "r-plyr                    1.8.4                  r3.4.1_0    r\n",
      "r-prettyunits             1.0.2                  r3.4.1_0    conda-forge\n",
      "r-pryr                    0.1.2                  r3.4.1_0    r\n",
      "r-psych                   1.7.5                  r3.4.1_0    r\n",
      "r-purrr                   0.2.2.2                r3.4.1_0    r\n",
      "r-quantmod                0.4_9                  r3.4.1_0    r\n",
      "r-quantreg                5.33                   r3.4.1_0    r\n",
      "r-r6                      2.2.1                  r3.4.1_0    r\n",
      "r-randomforest            4.6_12                 r3.4.1_0    r\n",
      "r-rbokeh                  0.5.0                  r3.4.1_0    r\n",
      "r-rcolorbrewer            1.1_2                  r3.4.1_3    r\n",
      "r-rcpp                    0.12.11                r3.4.1_0    r\n",
      "r-rcppeigen               0.3.3.3.0              r3.4.1_0    r\n",
      "r-readr                   1.1.1                  r3.4.1_0    r\n",
      "r-readxl                  1.0.0                  r3.4.1_0    r\n",
      "r-recommended             3.4.1                  r3.4.1_0    r\n",
      "r-rematch                 1.0.1                  r3.4.1_0    r\n",
      "r-repr                    0.10                   r3.4.1_0    r\n",
      "r-reshape2                1.4.2                  r3.4.1_0    r\n",
      "r-rlang                   0.1.1                  r3.4.1_0    r\n",
      "r-rmarkdown               1.5                    r3.4.1_0    r\n",
      "r-rpart                   4.1_11                 r3.4.1_0    r\n",
      "r-rprojroot               1.2                    r3.4.1_0    r\n",
      "r-rsqlite                 2.0                    r3.4.1_0    conda-forge\n",
      "r-rvest                   0.3.2                  r3.4.1_0    r\n",
      "r-scales                  0.4.1                  r3.4.1_0    r\n",
      "r-selectr                 0.3_1                  r3.4.1_0    r\n",
      "r-shiny                   1.0.3                  r3.4.1_0    r\n",
      "r-sourcetools             0.1.6                  r3.4.1_0    r\n",
      "r-sparsem                 1.77                   r3.4.1_0    r\n",
      "r-spatial                 7.3_11                 r3.4.1_0    r\n",
      "r-stringi                 1.1.5                  r3.4.1_0    r\n",
      "r-stringr                 1.2.0                  r3.4.1_0    r\n",
      "r-survival                2.41_3                 r3.4.1_0    r\n",
      "r-tibble                  1.3.3                  r3.4.1_0    r\n",
      "r-tidyr                   0.6.3                  r3.4.1_0    r\n",
      "r-tidyverse               1.1.1                  r3.4.1_0    r\n",
      "r-ttr                     0.23_1                 r3.4.1_0    r\n",
      "r-upsetr                  1.3.3                  r3.4.1_0    conda-forge\n",
      "r-uuid                    0.1_2                  r3.4.1_0    r\n",
      "r-xml2                    1.1.1                  r3.4.1_0    r\n",
      "r-xtable                  1.8_2                  r3.4.1_0    r\n",
      "r-xts                     0.9_7                  r3.4.1_2    r\n",
      "r-yaml                    2.1.14                 r3.4.1_0    r\n",
      "r-zoo                     1.8_0                  r3.4.1_0    r\n",
      "readline                  7.0                  hb321a52_4  \n",
      "reportlab                 3.4.0                    py35_0  \n",
      "requests                  2.14.2                   py35_0  \n",
      "rpy2                      2.9.4           py35r341h24bf2e0_0    conda-forge\n",
      "samtools                  1.4.1                         0    bioconda\n",
      "scikit-learn              0.19.1           py35hbf1f462_0  \n",
      "scipy                     1.1.0            py35hfc37229_0  \n",
      "screamingbackpack         0.2.333                  py35_0    bioconda\n",
      "screed                    1.0                      py35_0    bioconda\n",
      "seaborn                   0.8.1            py35h04cba02_0  \n",
      "send2trash                1.5.0                    py35_0  \n",
      "setuptools                39.1.0                   py35_0  \n",
      "simplegeneric             0.8.1                    py35_2  \n",
      "sina                      1.3.0                         0    bioconda\n",
      "sip                       4.19.8           py35hf484d3e_0  \n",
      "six                       1.10.0           py35h5312c1b_1  \n",
      "snakemake                 3.7.1                    py35_0    bioconda\n",
      "sourmash                  2.0.0a7                  py35_0    bioconda\n",
      "sqlite                    3.23.1               he433501_0  \n",
      "sra-tools                 2.9.0                h470a237_1    bioconda\n",
      "statsmodels               0.9.0            py35h3010b51_0  \n",
      "taxonkit                  0.2.4                         1    bioconda\n",
      "tbb                       2018.0.2             hf484d3e_0  \n",
      "tbl2asn                   25.6                          2    bioconda\n",
      "terminado                 0.8.1                    py35_1  \n",
      "testpath                  0.3.1            py35had42eaf_0  \n",
      "tk                        8.6.7                hc745277_3  \n",
      "tornado                   5.0.2                    py35_0  \n",
      "traitlets                 4.3.2            py35ha522a97_0  \n",
      "tzlocal                   1.5.1                      py_0    conda-forge\n",
      "urllib3                   1.12                     py35_0    bioconda\n",
      "wcwidth                   0.1.7            py35hcd08066_0  \n",
      "webencodings              0.5.1            py35hb6cf162_1  \n",
      "wheel                     0.31.0                   py35_0  \n",
      "xorg-kbproto              1.0.7                         1    conda-forge\n",
      "xorg-libice               1.0.9                         2    conda-forge\n",
      "xorg-libsm                1.2.2                         2    conda-forge\n",
      "xorg-libx11               1.6.5                         0    conda-forge\n",
      "xorg-libxau               1.0.8                         3    conda-forge\n",
      "xorg-libxdmcp             1.1.2                         3    conda-forge\n",
      "xorg-libxext              1.3.3                         2    conda-forge\n",
      "xorg-libxrender           0.9.10                        0    conda-forge\n",
      "xorg-renderproto          0.11.1                        1    conda-forge\n",
      "xorg-xextproto            7.3.0                         1    conda-forge\n",
      "xorg-xproto               7.0.31                        6    conda-forge\n",
      "xz                        5.2.3                h5e939de_4  \n",
      "yaml                      0.1.7                had09818_2  \n",
      "zeromq                    4.2.5                h439df22_0  \n",
      "zlib                      1.2.11               ha838bed_2  \n"
     ]
    }
   ],
   "source": [
    "!conda list -n metacompass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:metacompass]",
   "language": "python",
   "name": "conda-env-metacompass-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": "block",
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
