{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "Jacobo de la Cuesta-Zuluaga.\n",
    "\n",
    "Now that I classified taxonomically the **vadinCA11** bin as member of the *Methanomassiliicoccales* order, and that I established that the bin can be considered as a near complete genome with low contamination (98.4% and 0%, respectively), I can download the genomes of closely related organisms to perform the comparative genomic analyses. In this notebook I will fetch the assemblies from NCBI and assess their quality. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T08:56:22.333222Z",
     "start_time": "2019-01-25T08:56:22.327343Z"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import subprocess\n",
    "import urllib.request\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T08:56:22.973248Z",
     "start_time": "2019-01-25T08:56:22.966729Z"
    }
   },
   "outputs": [],
   "source": [
    "# Work dir\n",
    "work_dir = \"/ebio/abt3_projects/vadinCA11/data/V11\"\n",
    "\n",
    "# Included genomes dir\n",
    "included_genomes_dir = os.path.join(work_dir, \"included_genomes\")\n",
    "\n",
    "# vadinCA11 genome bin\n",
    "V11_bin = \"/ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/metabat2_low_PE.478.contigs.fa\"\n",
    "V11_folder = \"/ebio/abt3_projects/vadinCA11/data/metagenome/LLMGA/HiSeqRun83-91/bin_refine/DAS_Tool//bins_DASTool_bins/\"\n",
    "\n",
    "# Misc\n",
    "quality_env = \"py2_genome_quality\"\n",
    "metacompass_env = \"metacompass\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get genomes\n",
    "I wasn't able to make `ncbi-genome-download` work to get the genomes of all available *Methanomassiliicoccales*. Therefore, I will adapt and use Guillermo's script, which I know works and have already used."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T08:56:34.389258Z",
     "start_time": "2019-01-25T08:56:25.301024Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>ftp_path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000001215.4</td>\n",
       "      <td>Release 6 plus ISO1 MT</td>\n",
       "      <td>Drosophila melanogaster</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000001405.27</td>\n",
       "      <td>GRCh38.p12</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000001515.5</td>\n",
       "      <td>Pan_tro 3.0</td>\n",
       "      <td>Pan troglodytes</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000001545.3</td>\n",
       "      <td>P_pygmaeus_2.0.2</td>\n",
       "      <td>Pongo abelii</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000001635.8</td>\n",
       "      <td>GRCm38.p6</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  # assembly_accession                asm_name            organism_name  \\\n",
       "0      GCA_000001215.4  Release 6 plus ISO1 MT  Drosophila melanogaster   \n",
       "1     GCA_000001405.27              GRCh38.p12             Homo sapiens   \n",
       "2      GCA_000001515.5             Pan_tro 3.0          Pan troglodytes   \n",
       "3      GCA_000001545.3        P_pygmaeus_2.0.2             Pongo abelii   \n",
       "4      GCA_000001635.8               GRCm38.p6             Mus musculus   \n",
       "\n",
       "                                            ftp_path  \n",
       "0  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  \n",
       "1  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  \n",
       "2  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  \n",
       "3  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  \n",
       "4  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get the list of assemblies into the file `assembly_summary.txt`\n",
    "# In case of needing only bacterial genes, change FTP adress for\n",
    "# ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/assembly_summary.txt\"\n",
    "assembly_summary = os.path.join(work_dir, \"genomes\", \"assembly_summary.txt\")\n",
    "ncbi_ftp = \"ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/assembly_summary_genbank.txt\"\n",
    "urllib.request.urlretrieve(ncbi_ftp, assembly_summary)\n",
    "\n",
    "# Read the file skipping the first line\n",
    "data_ncbi = pd.read_csv(assembly_summary, sep='\\t', skiprows=1);\n",
    "\n",
    "# Define a simple dataset of the complete genomes only, including the URLs where we can download them.\n",
    "# complete_genes = data_ncbi[data_ncbi[\"assembly_level\"] == \"Complete Genome\"][[\"# assembly_accession\", \n",
    "#                                                                              \"asm_name\", \"organism_name\", \n",
    "#                                                                              \"ftp_path\"]]\n",
    "\n",
    "\n",
    "# Define a simple dataset of all genomes and assemblies, including the URLs where we can download them.\n",
    "complete_genes = data_ncbi[[\"# assembly_accession\", \"asm_name\", \"organism_name\", \"ftp_path\"]]\n",
    "\n",
    "complete_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T08:56:34.401201Z",
     "start_time": "2019-01-25T08:56:34.392161Z"
    }
   },
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "\n",
    "\n",
    "def get_members(accession_list, directory, output_file):\n",
    "    os.chdir(directory)\n",
    "    assembly_lst = []\n",
    "    accession_lst = []\n",
    "    organism_lst = []\n",
    "    length_lst = []\n",
    "    \n",
    "    for accession in accession_list:\n",
    "        _df = complete_genes[complete_genes[\"# assembly_accession\"] == accession]\n",
    "        \n",
    "        if _df.shape[0] != 1:\n",
    "            continue\n",
    "        \n",
    "        url_dir = _df[\"ftp_path\"].iloc[0]\n",
    "        asm_name = _df[\"asm_name\"].iloc[0]\n",
    "        filename = \"{}_{}_genomic.fna.gz\".format(accession, asm_name)\n",
    "        \n",
    "        if not os.path.isfile(filename):\n",
    "            url_lnk = \"{}/{}\".format(url_dir, filename)\n",
    "            #urllib.request.urlretrieve(url_lnk, filename)\n",
    "            urllib.request.urlretrieve(\"{}/{}_genomic.fna.gz\".format(url_dir, url_dir.split(\"/\")[-1]), filename)\n",
    "        \n",
    "        assembly_lst.append(asm_name)\n",
    "        accession_lst.append(accession)\n",
    "        organism_lst.append(_df[\"organism_name\"].iloc[0])\n",
    "        \n",
    "    _df = pd.DataFrame({\"Assembly\": assembly_lst, \n",
    "                        \"Organism\": organism_lst, \n",
    "                        \"Accession\": accession_lst})\n",
    "    _df.to_csv(\"%s\" % output_file)\n",
    "    \n",
    "    return(_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-07T13:33:29.068470Z",
     "start_time": "2018-06-07T13:33:29.048613Z"
    }
   },
   "outputs": [],
   "source": [
    "# Looked for all assemblies available for Methanomassiliicoccales[ORGANISM] in NCBI assembly\n",
    "# on 07.06.2018\n",
    "download_folder = os.path.join(work_dir, \"genomes\")\n",
    "downloaded_list = os.path.join(download_folder, \"Methanomassiliicoccales.csv\")\n",
    "Methanomassiliicoccales_members = [\"GCA_000308215.1\", \"GCA_000300255.2\", \"GCA_000404225.1\", \n",
    "                                   \"GCA_000800805.1\", \"GCA_001481295.1\", \"GCA_001421175.1\", \n",
    "                                   \"GCA_001421185.1\", \"GCA_900313075.1\", \"GCA_900314325.1\", \n",
    "                                   \"GCA_002494585.1\", \"GCA_002494705.1\", \"GCA_002494805.1\", \n",
    "                                   \"GCA_002494945.1\", \"GCA_002495125.1\", \"GCA_002495325.1\", \n",
    "                                   \"GCA_002495495.1\", \"GCA_002495585.1\", \"GCA_002495665.1\", \n",
    "                                   \"GCA_002496345.1\", \"GCA_002496385.1\", \"GCA_002496785.1\", \n",
    "                                   \"GCA_002496945.1\", \"GCA_002497075.1\", \"GCA_002497155.1\", \n",
    "                                   \"GCA_002497195.1\", \"GCA_002497475.1\", \"GCA_002497815.1\", \n",
    "                                   \"GCA_002497995.1\", \"GCA_002498285.1\", \"GCA_002498365.1\", \n",
    "                                   \"GCA_002498425.1\", \"GCA_002498545.1\", \"GCA_002498605.1\", \n",
    "                                   \"GCA_002498765.1\", \"GCA_002498785.1\", \"GCA_002498805.1\", \n",
    "                                   \"GCA_002499085.1\", \"GCA_002502005.1\", \"GCA_002502165.1\", \n",
    "                                   \"GCA_002502465.1\", \"GCA_002502545.1\", \"GCA_002502765.1\", \n",
    "                                   \"GCA_002502925.1\", \"GCA_002502965.1\", \"GCA_002503495.1\", \n",
    "                                   \"GCA_002503545.1\", \"GCA_002503645.1\", \"GCA_002503785.1\", \n",
    "                                   \"GCA_002503925.1\", \"GCA_002504405.1\", \"GCA_002504495.1\", \n",
    "                                   \"GCA_002504525.1\", \"GCA_002504645.1\", \"GCA_002505225.1\", \n",
    "                                   \"GCA_002505245.1\", \"GCA_002505275.1\", \"GCA_002505345.1\", \n",
    "                                   \"GCA_002505735.1\", \"GCA_002506175.1\", \"GCA_002506255.1\", \n",
    "                                   \"GCA_002506325.1\", \"GCA_002506425.1\", \"GCA_002506565.1\", \n",
    "                                   \"GCA_002506865.1\", \"GCA_002506905.1\", \"GCA_002506985.1\", \n",
    "                                   \"GCA_002506995.1\", \"GCA_002508545.1\", \"GCA_002508555.1\", \n",
    "                                   \"GCA_002508585.1\", \"GCA_002508595.1\", \"GCA_002508625.1\", \n",
    "                                   \"GCA_002509405.1\", \"GCA_002509415.1\", \"GCA_002509425.1\", \n",
    "                                   \"GCA_002509465.1\", \"GCA_003135935.1\", \"GCA_003153895.1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-07T13:35:51.944527Z",
     "start_time": "2018-06-07T13:33:33.919345Z"
    }
   },
   "outputs": [],
   "source": [
    "# Download genome assemblies in fasta format from NCBI\n",
    "Methanomassiliicoccales = get_members(Methanomassiliicoccales_members, \n",
    "                                      download_folder, \n",
    "                                      downloaded_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-07T13:35:54.120577Z",
     "start_time": "2018-06-07T13:35:51.946713Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gunzip /ebio/abt3_projects/vadinCA11/data/V11/genomes/*.fna.gz\n"
     ]
    }
   ],
   "source": [
    "# Uncompress genome files\n",
    "gunzip_job = \"gunzip {0}/*.fna.gz\".format(genomes_folder)\n",
    "print(gunzip_job)\n",
    "!$gunzip_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-07T13:35:54.332633Z",
     "start_time": "2018-06-07T13:35:54.122749Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cd /ebio/abt3_projects/vadinCA11/data/V11/genomes; rename 's/_[a-zA-Z].*_genomic//g' GCA*\n"
     ]
    }
   ],
   "source": [
    "# Rename all files just to have the accession in the name\n",
    "rename_job = \"cd {0}; rename 's/_[a-zA-Z].*_genomic//g' GCA*\".format(download_folder)\n",
    "print(rename_job)\n",
    "!$rename_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-29T08:43:24.176609Z",
     "start_time": "2018-06-29T08:43:24.042651Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/ebio/abt3_projects/vadinCA11/data/V11/genomes\r\n"
     ]
    }
   ],
   "source": [
    "# Fix a couple of errors in the name of the files\n",
    "!pwd\n",
    "!mv GCA_003135935.1_20110800.fna GCA_003135935.1.fna\n",
    "!mv GCA_003153895.1_20120700.fna GCA_003153895.1.fna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-29T08:40:35.020476Z",
     "start_time": "2018-06-29T08:40:34.970147Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Organism</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000308215.1</td>\n",
       "      <td>ASM30821v1</td>\n",
       "      <td>Methanomassiliicoccus luminyensis B10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000300255.2</td>\n",
       "      <td>ASM30025v2</td>\n",
       "      <td>Candidatus Methanomethylophilus alvus Mx1201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000404225.1</td>\n",
       "      <td>ASM40422v1</td>\n",
       "      <td>Candidatus Methanomassiliicoccus intestinalis ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000800805.1</td>\n",
       "      <td>ASM80080v1</td>\n",
       "      <td>Candidatus Methanoplasma termitum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001481295.1</td>\n",
       "      <td>ASM148129v1</td>\n",
       "      <td>Candidatus Methanomethylophilus sp. 1R26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_001421175.1</td>\n",
       "      <td>ASM142117v1</td>\n",
       "      <td>Methanomassiliicoccales archaeon RumEn M2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_001421185.1</td>\n",
       "      <td>ASM142118v1</td>\n",
       "      <td>Methanomassiliicoccales archaeon RumEn M1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_900313075.1</td>\n",
       "      <td>Rumen uncultured genome RUG779</td>\n",
       "      <td>uncultured Candidatus Methanomethylophilus sp.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_900314325.1</td>\n",
       "      <td>Rumen uncultured genome hRUG898</td>\n",
       "      <td>uncultured Candidatus Methanomethylophilus sp.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_002494585.1</td>\n",
       "      <td>ASM249458v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCA_002494705.1</td>\n",
       "      <td>ASM249470v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_002494805.1</td>\n",
       "      <td>ASM249480v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCA_002494945.1</td>\n",
       "      <td>ASM249494v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_002495125.1</td>\n",
       "      <td>ASM249512v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_002495325.1</td>\n",
       "      <td>ASM249532v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCA_002495495.1</td>\n",
       "      <td>ASM249549v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCA_002495585.1</td>\n",
       "      <td>ASM249558v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCA_002495665.1</td>\n",
       "      <td>ASM249566v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCA_002496345.1</td>\n",
       "      <td>ASM249634v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_002496385.1</td>\n",
       "      <td>ASM249638v1</td>\n",
       "      <td>Methanomassiliicoccales archaeon UBA147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCA_002496785.1</td>\n",
       "      <td>ASM249678v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_002496945.1</td>\n",
       "      <td>ASM249694v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_002497075.1</td>\n",
       "      <td>ASM249707v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_002497155.1</td>\n",
       "      <td>ASM249715v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_002497195.1</td>\n",
       "      <td>ASM249719v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_002497475.1</td>\n",
       "      <td>ASM249747v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GCA_002497815.1</td>\n",
       "      <td>ASM249781v1</td>\n",
       "      <td>Methanomassiliicoccales archaeon UBA149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GCA_002497995.1</td>\n",
       "      <td>ASM249799v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_002498285.1</td>\n",
       "      <td>ASM249828v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GCA_002498365.1</td>\n",
       "      <td>ASM249836v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>GCA_002503925.1</td>\n",
       "      <td>ASM250392v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>GCA_002504405.1</td>\n",
       "      <td>ASM250440v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>GCA_002504495.1</td>\n",
       "      <td>ASM250449v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>GCA_002504525.1</td>\n",
       "      <td>ASM250452v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>GCA_002504645.1</td>\n",
       "      <td>ASM250464v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>GCA_002505225.1</td>\n",
       "      <td>ASM250522v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>GCA_002505245.1</td>\n",
       "      <td>ASM250524v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>GCA_002505275.1</td>\n",
       "      <td>ASM250527v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>GCA_002505345.1</td>\n",
       "      <td>ASM250534v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>GCA_002505735.1</td>\n",
       "      <td>ASM250573v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>GCA_002506175.1</td>\n",
       "      <td>ASM250617v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>GCA_002506255.1</td>\n",
       "      <td>ASM250625v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>GCA_002506325.1</td>\n",
       "      <td>ASM250632v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>GCA_002506425.1</td>\n",
       "      <td>ASM250642v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>GCA_002506565.1</td>\n",
       "      <td>ASM250656v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>GCA_002506865.1</td>\n",
       "      <td>ASM250686v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>GCA_002506905.1</td>\n",
       "      <td>ASM250690v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>GCA_002506985.1</td>\n",
       "      <td>ASM250698v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>GCA_002506995.1</td>\n",
       "      <td>ASM250699v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>GCA_002508545.1</td>\n",
       "      <td>ASM250854v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>GCA_002508555.1</td>\n",
       "      <td>ASM250855v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>GCA_002508585.1</td>\n",
       "      <td>ASM250858v1</td>\n",
       "      <td>Methanomassiliicoccus sp. UBA360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>GCA_002508595.1</td>\n",
       "      <td>ASM250859v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>GCA_002508625.1</td>\n",
       "      <td>ASM250862v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>GCA_002509405.1</td>\n",
       "      <td>ASM250940v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>GCA_002509415.1</td>\n",
       "      <td>ASM250941v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>GCA_002509425.1</td>\n",
       "      <td>ASM250942v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>GCA_002509465.1</td>\n",
       "      <td>ASM250946v1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon UBA308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>GCA_003135935.1</td>\n",
       "      <td>20110800_E3D</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>GCA_003153895.1</td>\n",
       "      <td>20120700_E3M</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Accession                         Assembly  \\\n",
       "0   GCA_000308215.1                       ASM30821v1   \n",
       "1   GCA_000300255.2                       ASM30025v2   \n",
       "2   GCA_000404225.1                       ASM40422v1   \n",
       "3   GCA_000800805.1                       ASM80080v1   \n",
       "4   GCA_001481295.1                      ASM148129v1   \n",
       "5   GCA_001421175.1                      ASM142117v1   \n",
       "6   GCA_001421185.1                      ASM142118v1   \n",
       "7   GCA_900313075.1   Rumen uncultured genome RUG779   \n",
       "8   GCA_900314325.1  Rumen uncultured genome hRUG898   \n",
       "9   GCA_002494585.1                      ASM249458v1   \n",
       "10  GCA_002494705.1                      ASM249470v1   \n",
       "11  GCA_002494805.1                      ASM249480v1   \n",
       "12  GCA_002494945.1                      ASM249494v1   \n",
       "13  GCA_002495125.1                      ASM249512v1   \n",
       "14  GCA_002495325.1                      ASM249532v1   \n",
       "15  GCA_002495495.1                      ASM249549v1   \n",
       "16  GCA_002495585.1                      ASM249558v1   \n",
       "17  GCA_002495665.1                      ASM249566v1   \n",
       "18  GCA_002496345.1                      ASM249634v1   \n",
       "19  GCA_002496385.1                      ASM249638v1   \n",
       "20  GCA_002496785.1                      ASM249678v1   \n",
       "21  GCA_002496945.1                      ASM249694v1   \n",
       "22  GCA_002497075.1                      ASM249707v1   \n",
       "23  GCA_002497155.1                      ASM249715v1   \n",
       "24  GCA_002497195.1                      ASM249719v1   \n",
       "25  GCA_002497475.1                      ASM249747v1   \n",
       "26  GCA_002497815.1                      ASM249781v1   \n",
       "27  GCA_002497995.1                      ASM249799v1   \n",
       "28  GCA_002498285.1                      ASM249828v1   \n",
       "29  GCA_002498365.1                      ASM249836v1   \n",
       "..              ...                              ...   \n",
       "48  GCA_002503925.1                      ASM250392v1   \n",
       "49  GCA_002504405.1                      ASM250440v1   \n",
       "50  GCA_002504495.1                      ASM250449v1   \n",
       "51  GCA_002504525.1                      ASM250452v1   \n",
       "52  GCA_002504645.1                      ASM250464v1   \n",
       "53  GCA_002505225.1                      ASM250522v1   \n",
       "54  GCA_002505245.1                      ASM250524v1   \n",
       "55  GCA_002505275.1                      ASM250527v1   \n",
       "56  GCA_002505345.1                      ASM250534v1   \n",
       "57  GCA_002505735.1                      ASM250573v1   \n",
       "58  GCA_002506175.1                      ASM250617v1   \n",
       "59  GCA_002506255.1                      ASM250625v1   \n",
       "60  GCA_002506325.1                      ASM250632v1   \n",
       "61  GCA_002506425.1                      ASM250642v1   \n",
       "62  GCA_002506565.1                      ASM250656v1   \n",
       "63  GCA_002506865.1                      ASM250686v1   \n",
       "64  GCA_002506905.1                      ASM250690v1   \n",
       "65  GCA_002506985.1                      ASM250698v1   \n",
       "66  GCA_002506995.1                      ASM250699v1   \n",
       "67  GCA_002508545.1                      ASM250854v1   \n",
       "68  GCA_002508555.1                      ASM250855v1   \n",
       "69  GCA_002508585.1                      ASM250858v1   \n",
       "70  GCA_002508595.1                      ASM250859v1   \n",
       "71  GCA_002508625.1                      ASM250862v1   \n",
       "72  GCA_002509405.1                      ASM250940v1   \n",
       "73  GCA_002509415.1                      ASM250941v1   \n",
       "74  GCA_002509425.1                      ASM250942v1   \n",
       "75  GCA_002509465.1                      ASM250946v1   \n",
       "76  GCA_003135935.1                     20110800_E3D   \n",
       "77  GCA_003153895.1                     20120700_E3M   \n",
       "\n",
       "                                             Organism  \n",
       "0               Methanomassiliicoccus luminyensis B10  \n",
       "1        Candidatus Methanomethylophilus alvus Mx1201  \n",
       "2   Candidatus Methanomassiliicoccus intestinalis ...  \n",
       "3                   Candidatus Methanoplasma termitum  \n",
       "4            Candidatus Methanomethylophilus sp. 1R26  \n",
       "5           Methanomassiliicoccales archaeon RumEn M2  \n",
       "6           Methanomassiliicoccales archaeon RumEn M1  \n",
       "7      uncultured Candidatus Methanomethylophilus sp.  \n",
       "8      uncultured Candidatus Methanomethylophilus sp.  \n",
       "9            Methanomassiliicoccaceae archaeon UBA409  \n",
       "10                   Methanomassiliicoccus sp. UBA238  \n",
       "11           Methanomassiliicoccaceae archaeon UBA482  \n",
       "12           Methanomassiliicoccaceae archaeon UBA396  \n",
       "13           Methanomassiliicoccaceae archaeon UBA387  \n",
       "14           Methanomassiliicoccaceae archaeon UBA593  \n",
       "15           Methanomassiliicoccaceae archaeon UBA394  \n",
       "16                   Methanomassiliicoccus sp. UBA292  \n",
       "17           Methanomassiliicoccaceae archaeon UBA378  \n",
       "18                   Methanomassiliicoccus sp. UBA382  \n",
       "19            Methanomassiliicoccales archaeon UBA147  \n",
       "20           Methanomassiliicoccaceae archaeon UBA407  \n",
       "21           Methanomassiliicoccaceae archaeon UBA408  \n",
       "22           Methanomassiliicoccaceae archaeon UBA472  \n",
       "23           Methanomassiliicoccaceae archaeon UBA309  \n",
       "24           Methanomassiliicoccaceae archaeon UBA354  \n",
       "25           Methanomassiliicoccaceae archaeon UBA483  \n",
       "26            Methanomassiliicoccales archaeon UBA149  \n",
       "27                   Methanomassiliicoccus sp. UBA393  \n",
       "28                   Methanomassiliicoccus sp. UBA386  \n",
       "29           Methanomassiliicoccaceae archaeon UBA421  \n",
       "..                                                ...  \n",
       "48           Methanomassiliicoccaceae archaeon UBA481  \n",
       "49            Methanomassiliicoccaceae archaeon UBA71  \n",
       "50           Methanomassiliicoccaceae archaeon UBA537  \n",
       "51                   Methanomassiliicoccus sp. UBA345  \n",
       "52           Methanomassiliicoccaceae archaeon UBA306  \n",
       "53                   Methanomassiliicoccus sp. UBA429  \n",
       "54           Methanomassiliicoccaceae archaeon UBA422  \n",
       "55           Methanomassiliicoccaceae archaeon UBA371  \n",
       "56           Methanomassiliicoccaceae archaeon UBA370  \n",
       "57           Methanomassiliicoccaceae archaeon UBA298  \n",
       "58           Methanomassiliicoccaceae archaeon UBA113  \n",
       "59           Methanomassiliicoccaceae archaeon UBA367  \n",
       "60           Methanomassiliicoccaceae archaeon UBA314  \n",
       "61           Methanomassiliicoccaceae archaeon UBA480  \n",
       "62           Methanomassiliicoccaceae archaeon UBA333  \n",
       "63           Methanomassiliicoccaceae archaeon UBA329  \n",
       "64           Methanomassiliicoccaceae archaeon UBA328  \n",
       "65           Methanomassiliicoccaceae archaeon UBA248  \n",
       "66           Methanomassiliicoccaceae archaeon UBA427  \n",
       "67                     Methanomassiliicoccus sp. UBA6  \n",
       "68                   Methanomassiliicoccus sp. UBA364  \n",
       "69                   Methanomassiliicoccus sp. UBA360  \n",
       "70            Methanomassiliicoccaceae archaeon UBA48  \n",
       "71           Methanomassiliicoccaceae archaeon UBA344  \n",
       "72            Methanomassiliicoccaceae archaeon UBA72  \n",
       "73           Methanomassiliicoccaceae archaeon UBA319  \n",
       "74           Methanomassiliicoccaceae archaeon UBA359  \n",
       "75           Methanomassiliicoccaceae archaeon UBA308  \n",
       "76                  Methanomassiliicoccaceae archaeon  \n",
       "77                  Methanomassiliicoccaceae archaeon  \n",
       "\n",
       "[78 rows x 3 columns]"
      ]
     },
     "execution_count": 266,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print table with accessions\n",
    "Methanomassiliicoccales_df = pd.read_csv(downloaded_list, sep=',', header=0)\n",
    "Methanomassiliicoccales_df = Methanomassiliicoccales_df.drop([\"Unnamed: 0\"], axis=1)\n",
    "Methanomassiliicoccales_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-28T17:25:21.456698Z",
     "start_time": "2018-06-28T17:25:21.438146Z"
    }
   },
   "outputs": [],
   "source": [
    "# Write file downloaded genomes names and accessions\n",
    "include_table = os.path.join(download_folder, \"included_table.txt\")\n",
    "Methanomassiliicoccales_df.to_csv(include_table, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Assess quality of downloaded genomes\n",
    "To guarantee that the downstream analyses are correct, I will first perform a quality assessment of the downloaded genomes using CheckM. This way, I can identify and exclude incomplete or contaminated genomes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run CheckM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-07-05T07:46:19.036624Z",
     "start_time": "2018-07-05T07:08:06.247410Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate py2_genome_quality; checkm lineage_wf -t 8     -x fna     -f /ebio/abt3_projects/vadinCA11/data/V11/checkm_downloaded_genomes/CheckM_genomes.txt     /ebio/abt3_projects/vadinCA11/data/V11/genomes /ebio/abt3_projects/vadinCA11/data/V11/checkm_downloaded_genomes \"\n",
      "\n",
      "*******************************************************************************\n",
      " [CheckM - tree] Placing bins in reference genome tree.\n",
      "*******************************************************************************\n",
      "\n",
      "  Identifying marker genes in 78 bins with 8 threads:\n",
      "    Finished processing 78 of 78 (100.00%) bins.\n",
      "  Saving HMM info to file.\n",
      "\n",
      "  Calculating genome statistics for 78 bins with 8 threads:\n",
      "    Finished processing 78 of 78 (100.00%) bins.\n",
      "\n",
      "  Extracting marker genes to align.\n",
      "  Parsing HMM hits to marker genes:\n",
      "    Finished parsing hits for 32 of 78 (41.03%) bins."
     ]
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 10053 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Running CheckM to verify the completness and redundancy of the downloaded genomes\n",
    "#\n",
    "\n",
    "checkm_file = os.path.join(work_dir, \"checkm_downloaded_genomes\", 'CheckM_genomes.txt')\n",
    "checkm_cmd = \"\"\"checkm lineage_wf -t {0} \\\n",
    "    -x {1} \\\n",
    "    -f {2} \\\n",
    "    {3} {4}/checkm_downloaded_genomes \"\"\"\n",
    "checkm_cmd = checkm_cmd.format(8, \"fna\", checkm_file, download_folder, work_dir)\n",
    "checkm_job = 'bash -c \"source activate {0}; {1}\"'\n",
    "checkm_job = checkm_job.format(quality_env, checkm_cmd)\n",
    "print(checkm_job)\n",
    "!$checkm_job"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-22T13:24:21.069248Z",
     "start_time": "2018-10-22T13:24:21.008086Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>V11_RL001</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stat</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Completeness</th>\n",
       "      <td>98.3871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Contamination</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genome size</th>\n",
       "      <td>1590932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># ambiguous bases</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mean contig length</th>\n",
       "      <td>33144.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50 (contigs)</th>\n",
       "      <td>50438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longest contig</th>\n",
       "      <td>103078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># scaffolds</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mean scaffold length</th>\n",
       "      <td>33144.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50 (scaffolds)</th>\n",
       "      <td>50438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longest scaffold</th>\n",
       "      <td>103078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC</th>\n",
       "      <td>0.559415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC std</th>\n",
       "      <td>0.016983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes</th>\n",
       "      <td>1605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translation table</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coding density</th>\n",
       "      <td>0.892202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>marker lineage</th>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># genomes</th>\n",
       "      <td>148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># markers</th>\n",
       "      <td>187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># marker sets</th>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             V11_RL001\n",
       "Stat                                  \n",
       "Completeness                   98.3871\n",
       "Contamination                        0\n",
       "Genome size                    1590932\n",
       "# ambiguous bases                    0\n",
       "# contigs                           48\n",
       "Mean contig length             33144.4\n",
       "N50 (contigs)                    50438\n",
       "Longest contig                  103078\n",
       "# scaffolds                         48\n",
       "Mean scaffold length           33144.4\n",
       "N50 (scaffolds)                  50438\n",
       "Longest scaffold                103078\n",
       "GC                            0.559415\n",
       "GC std                        0.016983\n",
       "# predicted genes                 1605\n",
       "Translation table                   11\n",
       "Coding density                0.892202\n",
       "marker lineage        p__Euryarchaeota\n",
       "# genomes                          148\n",
       "# markers                          187\n",
       "# marker sets                      124"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkm_stats = \"{0}/checkm_output/storage/bin_stats_ext.tsv\".format(work_dir)\n",
    "chepwdckm_df = pd.read_csv(checkm_stats, sep='\\t', skiprows=0, header=None)\n",
    "checkm_df = eval(chepwdckm_df.iat[0,1]) # Select only the stats from the V11 bin and turn into dict\n",
    "checkm_df = pd.DataFrame(list(checkm_df.items()))\n",
    "checkm_df.columns = [\"Stat\", \"Value\"]\n",
    "checkm_df = checkm_df.set_index('Stat')\n",
    "checkm_df = checkm_df.loc[[\"Completeness\", \"Contamination\", \"Genome size\", \"# ambiguous bases\", \n",
    "               \"# contigs\", \"Mean contig length\", \"N50 (contigs)\", \"Longest contig\",\n",
    "               \"# scaffolds\", \"Mean scaffold length\", \"N50 (scaffolds)\",\n",
    "               \"Longest scaffold\", \"GC\", \"GC std\", \"# predicted genes\", \n",
    "               \"Translation table\", \"Coding density\", \"marker lineage\", \"# genomes\",\n",
    "               \"# markers\", \"# marker sets\" ]]\n",
    "checkm_df.rename(columns = {\"Value\": \"V11_RL001\"}, inplace=True)\n",
    "checkm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-22T13:24:36.853210Z",
     "start_time": "2018-10-22T13:24:36.461446Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GCA_000300255.2</th>\n",
       "      <th>GCA_002503545.1</th>\n",
       "      <th>GCA_002495495.1</th>\n",
       "      <th>GCA_002506865.1</th>\n",
       "      <th>GCA_002499085.1</th>\n",
       "      <th>GCA_002494805.1</th>\n",
       "      <th>GCA_002502545.1</th>\n",
       "      <th>GCA_000404225.1</th>\n",
       "      <th>GCA_002495125.1</th>\n",
       "      <th>GCA_002498765.1</th>\n",
       "      <th>...</th>\n",
       "      <th>GCA_002503785.1</th>\n",
       "      <th>GCA_900313075.1</th>\n",
       "      <th>GCA_002505245.1</th>\n",
       "      <th>GCA_002508585.1</th>\n",
       "      <th>GCA_002508545.1</th>\n",
       "      <th>GCA_002509415.1</th>\n",
       "      <th>GCA_002494945.1</th>\n",
       "      <th>GCA_002506255.1</th>\n",
       "      <th>GCA_002498425.1</th>\n",
       "      <th>GCA_002496945.1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stat</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Completeness</th>\n",
       "      <td>98.3871</td>\n",
       "      <td>88.3788</td>\n",
       "      <td>86.2068</td>\n",
       "      <td>83.3633</td>\n",
       "      <td>76.8</td>\n",
       "      <td>92.3387</td>\n",
       "      <td>88.4904</td>\n",
       "      <td>98.7903</td>\n",
       "      <td>65.0538</td>\n",
       "      <td>95.0775</td>\n",
       "      <td>...</td>\n",
       "      <td>96.9086</td>\n",
       "      <td>82.2581</td>\n",
       "      <td>95.5645</td>\n",
       "      <td>88.3002</td>\n",
       "      <td>97.3118</td>\n",
       "      <td>88.936</td>\n",
       "      <td>64.0521</td>\n",
       "      <td>61.1559</td>\n",
       "      <td>90.7258</td>\n",
       "      <td>97.135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Contamination</th>\n",
       "      <td>0.806452</td>\n",
       "      <td>0.94086</td>\n",
       "      <td>0.403226</td>\n",
       "      <td>4.03226</td>\n",
       "      <td>2.4</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.41935</td>\n",
       "      <td>0</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genome size</th>\n",
       "      <td>1666795</td>\n",
       "      <td>1249378</td>\n",
       "      <td>1063745</td>\n",
       "      <td>1104259</td>\n",
       "      <td>1749312</td>\n",
       "      <td>1648656</td>\n",
       "      <td>1115929</td>\n",
       "      <td>1931651</td>\n",
       "      <td>882347</td>\n",
       "      <td>1174624</td>\n",
       "      <td>...</td>\n",
       "      <td>1308604</td>\n",
       "      <td>1262884</td>\n",
       "      <td>1175704</td>\n",
       "      <td>1883692</td>\n",
       "      <td>2360299</td>\n",
       "      <td>1123872</td>\n",
       "      <td>869682</td>\n",
       "      <td>823014</td>\n",
       "      <td>1224361</td>\n",
       "      <td>1401758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># ambiguous bases</th>\n",
       "      <td>0</td>\n",
       "      <td>1306</td>\n",
       "      <td>6646</td>\n",
       "      <td>20303</td>\n",
       "      <td>479</td>\n",
       "      <td>790</td>\n",
       "      <td>17388</td>\n",
       "      <td>0</td>\n",
       "      <td>21968</td>\n",
       "      <td>10307</td>\n",
       "      <td>...</td>\n",
       "      <td>383</td>\n",
       "      <td>0</td>\n",
       "      <td>5963</td>\n",
       "      <td>74196</td>\n",
       "      <td>2392</td>\n",
       "      <td>23022</td>\n",
       "      <td>22328</td>\n",
       "      <td>26401</td>\n",
       "      <td>5065</td>\n",
       "      <td>6936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs</th>\n",
       "      <td>1</td>\n",
       "      <td>56</td>\n",
       "      <td>107</td>\n",
       "      <td>175</td>\n",
       "      <td>46</td>\n",
       "      <td>47</td>\n",
       "      <td>135</td>\n",
       "      <td>1</td>\n",
       "      <td>158</td>\n",
       "      <td>66</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>192</td>\n",
       "      <td>49</td>\n",
       "      <td>464</td>\n",
       "      <td>111</td>\n",
       "      <td>138</td>\n",
       "      <td>216</td>\n",
       "      <td>223</td>\n",
       "      <td>38</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mean contig length</th>\n",
       "      <td>1.6668e+06</td>\n",
       "      <td>22287</td>\n",
       "      <td>9879.43</td>\n",
       "      <td>6194.03</td>\n",
       "      <td>38018.1</td>\n",
       "      <td>35061</td>\n",
       "      <td>8137.34</td>\n",
       "      <td>1.93165e+06</td>\n",
       "      <td>5445.44</td>\n",
       "      <td>17641.2</td>\n",
       "      <td>...</td>\n",
       "      <td>59464.6</td>\n",
       "      <td>6577.52</td>\n",
       "      <td>23872.3</td>\n",
       "      <td>3899.78</td>\n",
       "      <td>21242.4</td>\n",
       "      <td>7977.17</td>\n",
       "      <td>3922.94</td>\n",
       "      <td>3572.26</td>\n",
       "      <td>32086.7</td>\n",
       "      <td>23247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50 (contigs)</th>\n",
       "      <td>1666795</td>\n",
       "      <td>45710</td>\n",
       "      <td>16791</td>\n",
       "      <td>13507</td>\n",
       "      <td>56845</td>\n",
       "      <td>64245</td>\n",
       "      <td>25662</td>\n",
       "      <td>1931651</td>\n",
       "      <td>37789</td>\n",
       "      <td>135661</td>\n",
       "      <td>...</td>\n",
       "      <td>102142</td>\n",
       "      <td>8388</td>\n",
       "      <td>46981</td>\n",
       "      <td>7813</td>\n",
       "      <td>46502</td>\n",
       "      <td>21846</td>\n",
       "      <td>7750</td>\n",
       "      <td>6129</td>\n",
       "      <td>55747</td>\n",
       "      <td>49812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longest contig</th>\n",
       "      <td>1666795</td>\n",
       "      <td>99237</td>\n",
       "      <td>55665</td>\n",
       "      <td>44660</td>\n",
       "      <td>154585</td>\n",
       "      <td>269292</td>\n",
       "      <td>103911</td>\n",
       "      <td>1931651</td>\n",
       "      <td>86336</td>\n",
       "      <td>270640</td>\n",
       "      <td>...</td>\n",
       "      <td>181776</td>\n",
       "      <td>35746</td>\n",
       "      <td>122573</td>\n",
       "      <td>40576</td>\n",
       "      <td>119060</td>\n",
       "      <td>49127</td>\n",
       "      <td>31952</td>\n",
       "      <td>23521</td>\n",
       "      <td>153909</td>\n",
       "      <td>136345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># scaffolds</th>\n",
       "      <td>1</td>\n",
       "      <td>34</td>\n",
       "      <td>80</td>\n",
       "      <td>58</td>\n",
       "      <td>37</td>\n",
       "      <td>36</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>29</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>192</td>\n",
       "      <td>16</td>\n",
       "      <td>144</td>\n",
       "      <td>68</td>\n",
       "      <td>42</td>\n",
       "      <td>95</td>\n",
       "      <td>93</td>\n",
       "      <td>15</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mean scaffold length</th>\n",
       "      <td>1.6668e+06</td>\n",
       "      <td>36746.4</td>\n",
       "      <td>13296.8</td>\n",
       "      <td>19038.9</td>\n",
       "      <td>47278.7</td>\n",
       "      <td>45796</td>\n",
       "      <td>29366.6</td>\n",
       "      <td>1.93165e+06</td>\n",
       "      <td>44117.3</td>\n",
       "      <td>40504.3</td>\n",
       "      <td>...</td>\n",
       "      <td>109050</td>\n",
       "      <td>6577.52</td>\n",
       "      <td>73481.5</td>\n",
       "      <td>13081.2</td>\n",
       "      <td>34710.3</td>\n",
       "      <td>26758.9</td>\n",
       "      <td>9154.55</td>\n",
       "      <td>8849.61</td>\n",
       "      <td>81624.1</td>\n",
       "      <td>58406.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50 (scaffolds)</th>\n",
       "      <td>1666795</td>\n",
       "      <td>54472</td>\n",
       "      <td>18039</td>\n",
       "      <td>41159</td>\n",
       "      <td>73068</td>\n",
       "      <td>66041</td>\n",
       "      <td>69703</td>\n",
       "      <td>1931651</td>\n",
       "      <td>63391</td>\n",
       "      <td>163880</td>\n",
       "      <td>...</td>\n",
       "      <td>128983</td>\n",
       "      <td>8388</td>\n",
       "      <td>117614</td>\n",
       "      <td>18257</td>\n",
       "      <td>59274</td>\n",
       "      <td>44966</td>\n",
       "      <td>12321</td>\n",
       "      <td>12087</td>\n",
       "      <td>116637</td>\n",
       "      <td>100729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longest scaffold</th>\n",
       "      <td>1666795</td>\n",
       "      <td>120572</td>\n",
       "      <td>76417</td>\n",
       "      <td>119079</td>\n",
       "      <td>154591</td>\n",
       "      <td>291792</td>\n",
       "      <td>140813</td>\n",
       "      <td>1931651</td>\n",
       "      <td>183706</td>\n",
       "      <td>270640</td>\n",
       "      <td>...</td>\n",
       "      <td>181778</td>\n",
       "      <td>35746</td>\n",
       "      <td>414343</td>\n",
       "      <td>109207</td>\n",
       "      <td>119078</td>\n",
       "      <td>104203</td>\n",
       "      <td>36651</td>\n",
       "      <td>36878</td>\n",
       "      <td>234605</td>\n",
       "      <td>293218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC</th>\n",
       "      <td>0.555948</td>\n",
       "      <td>0.601188</td>\n",
       "      <td>0.564028</td>\n",
       "      <td>0.555444</td>\n",
       "      <td>0.547508</td>\n",
       "      <td>0.574851</td>\n",
       "      <td>0.551633</td>\n",
       "      <td>0.412554</td>\n",
       "      <td>0.555003</td>\n",
       "      <td>0.554079</td>\n",
       "      <td>...</td>\n",
       "      <td>0.543219</td>\n",
       "      <td>0.616806</td>\n",
       "      <td>0.551546</td>\n",
       "      <td>0.613088</td>\n",
       "      <td>0.627406</td>\n",
       "      <td>0.553595</td>\n",
       "      <td>0.556602</td>\n",
       "      <td>0.53087</td>\n",
       "      <td>0.560651</td>\n",
       "      <td>0.516718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC std</th>\n",
       "      <td>0</td>\n",
       "      <td>0.0170428</td>\n",
       "      <td>0.0200564</td>\n",
       "      <td>0.0181341</td>\n",
       "      <td>0.013905</td>\n",
       "      <td>0.0176275</td>\n",
       "      <td>0.019404</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0118407</td>\n",
       "      <td>0.0175766</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0162278</td>\n",
       "      <td>0.0220273</td>\n",
       "      <td>0.0109465</td>\n",
       "      <td>0.0215446</td>\n",
       "      <td>0.0215979</td>\n",
       "      <td>0.0177516</td>\n",
       "      <td>0.0204313</td>\n",
       "      <td>0.021296</td>\n",
       "      <td>0.0146937</td>\n",
       "      <td>0.0184419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes</th>\n",
       "      <td>1636</td>\n",
       "      <td>1279</td>\n",
       "      <td>1152</td>\n",
       "      <td>1206</td>\n",
       "      <td>1503</td>\n",
       "      <td>1564</td>\n",
       "      <td>1187</td>\n",
       "      <td>1855</td>\n",
       "      <td>913</td>\n",
       "      <td>1240</td>\n",
       "      <td>...</td>\n",
       "      <td>1348</td>\n",
       "      <td>1374</td>\n",
       "      <td>1212</td>\n",
       "      <td>2144</td>\n",
       "      <td>2497</td>\n",
       "      <td>1192</td>\n",
       "      <td>960</td>\n",
       "      <td>917</td>\n",
       "      <td>1278</td>\n",
       "      <td>1521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translation table</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coding density</th>\n",
       "      <td>0.893948</td>\n",
       "      <td>0.899556</td>\n",
       "      <td>0.914932</td>\n",
       "      <td>0.870157</td>\n",
       "      <td>0.890325</td>\n",
       "      <td>0.900677</td>\n",
       "      <td>0.883084</td>\n",
       "      <td>0.882898</td>\n",
       "      <td>0.865416</td>\n",
       "      <td>0.900474</td>\n",
       "      <td>...</td>\n",
       "      <td>0.925829</td>\n",
       "      <td>0.909772</td>\n",
       "      <td>0.91466</td>\n",
       "      <td>0.821946</td>\n",
       "      <td>0.881767</td>\n",
       "      <td>0.886052</td>\n",
       "      <td>0.869988</td>\n",
       "      <td>0.84989</td>\n",
       "      <td>0.912572</td>\n",
       "      <td>0.893666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>marker lineage</th>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>...</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># genomes</th>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>...</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># markers</th>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>188</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>...</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "      <td>187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># marker sets</th>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>125</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>...</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21 rows × 78 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       GCA_000300255.2   GCA_002503545.1   GCA_002495495.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   98.3871           88.3788           86.2068   \n",
       "Contamination                 0.806452           0.94086          0.403226   \n",
       "Genome size                    1666795           1249378           1063745   \n",
       "# ambiguous bases                    0              1306              6646   \n",
       "# contigs                            1                56               107   \n",
       "Mean contig length          1.6668e+06             22287           9879.43   \n",
       "N50 (contigs)                  1666795             45710             16791   \n",
       "Longest contig                 1666795             99237             55665   \n",
       "# scaffolds                          1                34                80   \n",
       "Mean scaffold length        1.6668e+06           36746.4           13296.8   \n",
       "N50 (scaffolds)                1666795             54472             18039   \n",
       "Longest scaffold               1666795            120572             76417   \n",
       "GC                            0.555948          0.601188          0.564028   \n",
       "GC std                               0         0.0170428         0.0200564   \n",
       "# predicted genes                 1636              1279              1152   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.893948          0.899556          0.914932   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               187               187   \n",
       "# marker sets                      124               124               124   \n",
       "\n",
       "                       GCA_002506865.1   GCA_002499085.1   GCA_002494805.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   83.3633              76.8           92.3387   \n",
       "Contamination                  4.03226               2.4          0.806452   \n",
       "Genome size                    1104259           1749312           1648656   \n",
       "# ambiguous bases                20303               479               790   \n",
       "# contigs                          175                46                47   \n",
       "Mean contig length             6194.03           38018.1             35061   \n",
       "N50 (contigs)                    13507             56845             64245   \n",
       "Longest contig                   44660            154585            269292   \n",
       "# scaffolds                         58                37                36   \n",
       "Mean scaffold length           19038.9           47278.7             45796   \n",
       "N50 (scaffolds)                  41159             73068             66041   \n",
       "Longest scaffold                119079            154591            291792   \n",
       "GC                            0.555444          0.547508          0.574851   \n",
       "GC std                       0.0181341          0.013905         0.0176275   \n",
       "# predicted genes                 1206              1503              1564   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.870157          0.890325          0.900677   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               188               187   \n",
       "# marker sets                      124               125               124   \n",
       "\n",
       "                       GCA_002502545.1   GCA_000404225.1   GCA_002495125.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   88.4904           98.7903           65.0538   \n",
       "Contamination                        0          0.806452                 0   \n",
       "Genome size                    1115929           1931651            882347   \n",
       "# ambiguous bases                17388                 0             21968   \n",
       "# contigs                          135                 1               158   \n",
       "Mean contig length             8137.34       1.93165e+06           5445.44   \n",
       "N50 (contigs)                    25662           1931651             37789   \n",
       "Longest contig                  103911           1931651             86336   \n",
       "# scaffolds                         38                 1                20   \n",
       "Mean scaffold length           29366.6       1.93165e+06           44117.3   \n",
       "N50 (scaffolds)                  69703           1931651             63391   \n",
       "Longest scaffold                140813           1931651            183706   \n",
       "GC                            0.551633          0.412554          0.555003   \n",
       "GC std                        0.019404                 0         0.0118407   \n",
       "# predicted genes                 1187              1855               913   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.883084          0.882898          0.865416   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               187               187   \n",
       "# marker sets                      124               124               124   \n",
       "\n",
       "                       GCA_002498765.1        ...          GCA_002503785.1  \\\n",
       "Stat                                          ...                            \n",
       "Completeness                   95.0775        ...                  96.9086   \n",
       "Contamination                        0        ...                        0   \n",
       "Genome size                    1174624        ...                  1308604   \n",
       "# ambiguous bases                10307        ...                      383   \n",
       "# contigs                           66        ...                       22   \n",
       "Mean contig length             17641.2        ...                  59464.6   \n",
       "N50 (contigs)                   135661        ...                   102142   \n",
       "Longest contig                  270640        ...                   181776   \n",
       "# scaffolds                         29        ...                       12   \n",
       "Mean scaffold length           40504.3        ...                   109050   \n",
       "N50 (scaffolds)                 163880        ...                   128983   \n",
       "Longest scaffold                270640        ...                   181778   \n",
       "GC                            0.554079        ...                 0.543219   \n",
       "GC std                       0.0175766        ...                0.0162278   \n",
       "# predicted genes                 1240        ...                     1348   \n",
       "Translation table                   11        ...                       11   \n",
       "Coding density                0.900474        ...                 0.925829   \n",
       "marker lineage        p__Euryarchaeota        ...         p__Euryarchaeota   \n",
       "# genomes                          148        ...                      148   \n",
       "# markers                          187        ...                      187   \n",
       "# marker sets                      124        ...                      124   \n",
       "\n",
       "                       GCA_900313075.1   GCA_002505245.1   GCA_002508585.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   82.2581           95.5645           88.3002   \n",
       "Contamination                  2.41935                 0          0.806452   \n",
       "Genome size                    1262884           1175704           1883692   \n",
       "# ambiguous bases                    0              5963             74196   \n",
       "# contigs                          192                49               464   \n",
       "Mean contig length             6577.52           23872.3           3899.78   \n",
       "N50 (contigs)                     8388             46981              7813   \n",
       "Longest contig                   35746            122573             40576   \n",
       "# scaffolds                        192                16               144   \n",
       "Mean scaffold length           6577.52           73481.5           13081.2   \n",
       "N50 (scaffolds)                   8388            117614             18257   \n",
       "Longest scaffold                 35746            414343            109207   \n",
       "GC                            0.616806          0.551546          0.613088   \n",
       "GC std                       0.0220273         0.0109465         0.0215446   \n",
       "# predicted genes                 1374              1212              2144   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.909772           0.91466          0.821946   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               187               187   \n",
       "# marker sets                      124               124               124   \n",
       "\n",
       "                       GCA_002508545.1   GCA_002509415.1   GCA_002494945.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   97.3118            88.936           64.0521   \n",
       "Contamination                 0.806452                 0                 0   \n",
       "Genome size                    2360299           1123872            869682   \n",
       "# ambiguous bases                 2392             23022             22328   \n",
       "# contigs                          111               138               216   \n",
       "Mean contig length             21242.4           7977.17           3922.94   \n",
       "N50 (contigs)                    46502             21846              7750   \n",
       "Longest contig                  119060             49127             31952   \n",
       "# scaffolds                         68                42                95   \n",
       "Mean scaffold length           34710.3           26758.9           9154.55   \n",
       "N50 (scaffolds)                  59274             44966             12321   \n",
       "Longest scaffold                119078            104203             36651   \n",
       "GC                            0.627406          0.553595          0.556602   \n",
       "GC std                       0.0215979         0.0177516         0.0204313   \n",
       "# predicted genes                 2497              1192               960   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.881767          0.886052          0.869988   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               187               187   \n",
       "# marker sets                      124               124               124   \n",
       "\n",
       "                       GCA_002506255.1   GCA_002498425.1   GCA_002496945.1  \n",
       "Stat                                                                        \n",
       "Completeness                   61.1559           90.7258            97.135  \n",
       "Contamination                 0.806452                 0                 0  \n",
       "Genome size                     823014           1224361           1401758  \n",
       "# ambiguous bases                26401              5065              6936  \n",
       "# contigs                          223                38                60  \n",
       "Mean contig length             3572.26           32086.7             23247  \n",
       "N50 (contigs)                     6129             55747             49812  \n",
       "Longest contig                   23521            153909            136345  \n",
       "# scaffolds                         93                15                24  \n",
       "Mean scaffold length           8849.61           81624.1           58406.6  \n",
       "N50 (scaffolds)                  12087            116637            100729  \n",
       "Longest scaffold                 36878            234605            293218  \n",
       "GC                             0.53087          0.560651          0.516718  \n",
       "GC std                        0.021296         0.0146937         0.0184419  \n",
       "# predicted genes                  917              1278              1521  \n",
       "Translation table                   11                11                11  \n",
       "Coding density                 0.84989          0.912572          0.893666  \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota  \n",
       "# genomes                          148               148               148  \n",
       "# markers                          187               187               187  \n",
       "# marker sets                      124               124               124  \n",
       "\n",
       "[21 rows x 78 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Summarize the CheckM results on downloaded genomes\n",
    "checkm_stats = \"{0}/checkm_downloaded_genomes/storage/bin_stats_ext.tsv\".format(work_dir)\n",
    "checkm_df = pd.read_csv(checkm_stats, sep='\\t', skiprows=0, header=None)\n",
    "\n",
    "# Names of assemblies\n",
    "assemblies = list(checkm_df.iloc[:,0])\n",
    "assemblies\n",
    "\n",
    "# Create a single df with all results\n",
    "summary_df = pd.DataFrame()\n",
    "for i in checkm_df.iloc[:, 1]:\n",
    "    tmp_df = eval(i)\n",
    "    tmp_df = pd.DataFrame(list(tmp_df.items()))\n",
    "    tmp_df.columns = [\"Stat\", \"Value\"]\n",
    "    tmp_df = tmp_df.set_index('Stat')\n",
    "    tmp_df = tmp_df.loc[[\"Completeness\", \"Contamination\", \"Genome size\", \"# ambiguous bases\", \n",
    "               \"# contigs\", \"Mean contig length\", \"N50 (contigs)\", \"Longest contig\",\n",
    "               \"# scaffolds\", \"Mean scaffold length\", \"N50 (scaffolds)\",\n",
    "               \"Longest scaffold\", \"GC\", \"GC std\", \"# predicted genes\", \n",
    "               \"Translation table\", \"Coding density\", \"marker lineage\", \"# genomes\",\n",
    "               \"# markers\", \"# marker sets\" ]]\n",
    "    summary_df = pd.concat([summary_df, tmp_df], axis=1)\n",
    "\n",
    "# Rename columns\n",
    "summary_df.columns = assemblies\n",
    "summary_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-22T13:24:40.769245Z",
     "start_time": "2018-10-22T13:24:40.756020Z"
    }
   },
   "outputs": [],
   "source": [
    "# Select only substantially complete genomes (>= 70%) with low contamination (<= 5%)\n",
    "complete = summary_df.loc['Completeness'] >= 70\n",
    "non_contaminated = summary_df.loc['Contamination'] <= 5\n",
    "# Summary of included\n",
    "include = complete & non_contaminated\n",
    "include_df = summary_df.loc[: , include]\n",
    "# Summary of excluded\n",
    "exclude = ~(summary_df.columns.isin(include_df.columns))\n",
    "exclude_df = summary_df.loc[: , exclude]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-07-05T09:21:53.760500Z",
     "start_time": "2018-07-05T09:21:53.739684Z"
    }
   },
   "outputs": [],
   "source": [
    "# Table with genome statistics of included\n",
    "include_df = pd.concat([checkm_df, include_df], axis=1)\n",
    "included_stats = os.path.join(download_folder, \"included_stats.txt\")\n",
    "include_df.to_csv(included_stats, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-07-05T09:21:29.448517Z",
     "start_time": "2018-07-05T09:21:29.393643Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>V11_RL001</th>\n",
       "      <th>GCA_000300255.2</th>\n",
       "      <th>GCA_002503545.1</th>\n",
       "      <th>GCA_002495495.1</th>\n",
       "      <th>GCA_002506865.1</th>\n",
       "      <th>GCA_002499085.1</th>\n",
       "      <th>GCA_002494805.1</th>\n",
       "      <th>GCA_002502545.1</th>\n",
       "      <th>GCA_000404225.1</th>\n",
       "      <th>GCA_002498765.1</th>\n",
       "      <th>...</th>\n",
       "      <th>GCA_002498285.1</th>\n",
       "      <th>GCA_002509465.1</th>\n",
       "      <th>GCA_002503785.1</th>\n",
       "      <th>GCA_900313075.1</th>\n",
       "      <th>GCA_002505245.1</th>\n",
       "      <th>GCA_002508585.1</th>\n",
       "      <th>GCA_002508545.1</th>\n",
       "      <th>GCA_002509415.1</th>\n",
       "      <th>GCA_002498425.1</th>\n",
       "      <th>GCA_002496945.1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stat</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Completeness</th>\n",
       "      <td>98.3871</td>\n",
       "      <td>98.3871</td>\n",
       "      <td>88.3788</td>\n",
       "      <td>86.2068</td>\n",
       "      <td>83.3633</td>\n",
       "      <td>76.8</td>\n",
       "      <td>92.3387</td>\n",
       "      <td>88.4904</td>\n",
       "      <td>98.7903</td>\n",
       "      <td>95.0775</td>\n",
       "      <td>...</td>\n",
       "      <td>84.7177</td>\n",
       "      <td>84.7455</td>\n",
       "      <td>96.9086</td>\n",
       "      <td>82.2581</td>\n",
       "      <td>95.5645</td>\n",
       "      <td>88.3002</td>\n",
       "      <td>97.3118</td>\n",
       "      <td>88.936</td>\n",
       "      <td>90.7258</td>\n",
       "      <td>97.135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Contamination</th>\n",
       "      <td>0</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0.94086</td>\n",
       "      <td>0.403226</td>\n",
       "      <td>4.03226</td>\n",
       "      <td>2.4</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.41935</td>\n",
       "      <td>0</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0.806452</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genome size</th>\n",
       "      <td>1590932</td>\n",
       "      <td>1666795</td>\n",
       "      <td>1249378</td>\n",
       "      <td>1063745</td>\n",
       "      <td>1104259</td>\n",
       "      <td>1749312</td>\n",
       "      <td>1648656</td>\n",
       "      <td>1115929</td>\n",
       "      <td>1931651</td>\n",
       "      <td>1174624</td>\n",
       "      <td>...</td>\n",
       "      <td>1827384</td>\n",
       "      <td>969311</td>\n",
       "      <td>1308604</td>\n",
       "      <td>1262884</td>\n",
       "      <td>1175704</td>\n",
       "      <td>1883692</td>\n",
       "      <td>2360299</td>\n",
       "      <td>1123872</td>\n",
       "      <td>1224361</td>\n",
       "      <td>1401758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># ambiguous bases</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1306</td>\n",
       "      <td>6646</td>\n",
       "      <td>20303</td>\n",
       "      <td>479</td>\n",
       "      <td>790</td>\n",
       "      <td>17388</td>\n",
       "      <td>0</td>\n",
       "      <td>10307</td>\n",
       "      <td>...</td>\n",
       "      <td>98446</td>\n",
       "      <td>5047</td>\n",
       "      <td>383</td>\n",
       "      <td>0</td>\n",
       "      <td>5963</td>\n",
       "      <td>74196</td>\n",
       "      <td>2392</td>\n",
       "      <td>23022</td>\n",
       "      <td>5065</td>\n",
       "      <td>6936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># contigs</th>\n",
       "      <td>48</td>\n",
       "      <td>1</td>\n",
       "      <td>56</td>\n",
       "      <td>107</td>\n",
       "      <td>175</td>\n",
       "      <td>46</td>\n",
       "      <td>47</td>\n",
       "      <td>135</td>\n",
       "      <td>1</td>\n",
       "      <td>66</td>\n",
       "      <td>...</td>\n",
       "      <td>572</td>\n",
       "      <td>104</td>\n",
       "      <td>22</td>\n",
       "      <td>192</td>\n",
       "      <td>49</td>\n",
       "      <td>464</td>\n",
       "      <td>111</td>\n",
       "      <td>138</td>\n",
       "      <td>38</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mean contig length</th>\n",
       "      <td>33144.4</td>\n",
       "      <td>1.6668e+06</td>\n",
       "      <td>22287</td>\n",
       "      <td>9879.43</td>\n",
       "      <td>6194.03</td>\n",
       "      <td>38018.1</td>\n",
       "      <td>35061</td>\n",
       "      <td>8137.34</td>\n",
       "      <td>1.93165e+06</td>\n",
       "      <td>17641.2</td>\n",
       "      <td>...</td>\n",
       "      <td>3022.62</td>\n",
       "      <td>9271.77</td>\n",
       "      <td>59464.6</td>\n",
       "      <td>6577.52</td>\n",
       "      <td>23872.3</td>\n",
       "      <td>3899.78</td>\n",
       "      <td>21242.4</td>\n",
       "      <td>7977.17</td>\n",
       "      <td>32086.7</td>\n",
       "      <td>23247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50 (contigs)</th>\n",
       "      <td>50438</td>\n",
       "      <td>1666795</td>\n",
       "      <td>45710</td>\n",
       "      <td>16791</td>\n",
       "      <td>13507</td>\n",
       "      <td>56845</td>\n",
       "      <td>64245</td>\n",
       "      <td>25662</td>\n",
       "      <td>1931651</td>\n",
       "      <td>135661</td>\n",
       "      <td>...</td>\n",
       "      <td>7077</td>\n",
       "      <td>15368</td>\n",
       "      <td>102142</td>\n",
       "      <td>8388</td>\n",
       "      <td>46981</td>\n",
       "      <td>7813</td>\n",
       "      <td>46502</td>\n",
       "      <td>21846</td>\n",
       "      <td>55747</td>\n",
       "      <td>49812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longest contig</th>\n",
       "      <td>103078</td>\n",
       "      <td>1666795</td>\n",
       "      <td>99237</td>\n",
       "      <td>55665</td>\n",
       "      <td>44660</td>\n",
       "      <td>154585</td>\n",
       "      <td>269292</td>\n",
       "      <td>103911</td>\n",
       "      <td>1931651</td>\n",
       "      <td>270640</td>\n",
       "      <td>...</td>\n",
       "      <td>71752</td>\n",
       "      <td>40716</td>\n",
       "      <td>181776</td>\n",
       "      <td>35746</td>\n",
       "      <td>122573</td>\n",
       "      <td>40576</td>\n",
       "      <td>119060</td>\n",
       "      <td>49127</td>\n",
       "      <td>153909</td>\n",
       "      <td>136345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># scaffolds</th>\n",
       "      <td>48</td>\n",
       "      <td>1</td>\n",
       "      <td>34</td>\n",
       "      <td>80</td>\n",
       "      <td>58</td>\n",
       "      <td>37</td>\n",
       "      <td>36</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>...</td>\n",
       "      <td>110</td>\n",
       "      <td>69</td>\n",
       "      <td>12</td>\n",
       "      <td>192</td>\n",
       "      <td>16</td>\n",
       "      <td>144</td>\n",
       "      <td>68</td>\n",
       "      <td>42</td>\n",
       "      <td>15</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mean scaffold length</th>\n",
       "      <td>33144.4</td>\n",
       "      <td>1.6668e+06</td>\n",
       "      <td>36746.4</td>\n",
       "      <td>13296.8</td>\n",
       "      <td>19038.9</td>\n",
       "      <td>47278.7</td>\n",
       "      <td>45796</td>\n",
       "      <td>29366.6</td>\n",
       "      <td>1.93165e+06</td>\n",
       "      <td>40504.3</td>\n",
       "      <td>...</td>\n",
       "      <td>16612.6</td>\n",
       "      <td>14048</td>\n",
       "      <td>109050</td>\n",
       "      <td>6577.52</td>\n",
       "      <td>73481.5</td>\n",
       "      <td>13081.2</td>\n",
       "      <td>34710.3</td>\n",
       "      <td>26758.9</td>\n",
       "      <td>81624.1</td>\n",
       "      <td>58406.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N50 (scaffolds)</th>\n",
       "      <td>50438</td>\n",
       "      <td>1666795</td>\n",
       "      <td>54472</td>\n",
       "      <td>18039</td>\n",
       "      <td>41159</td>\n",
       "      <td>73068</td>\n",
       "      <td>66041</td>\n",
       "      <td>69703</td>\n",
       "      <td>1931651</td>\n",
       "      <td>163880</td>\n",
       "      <td>...</td>\n",
       "      <td>30094</td>\n",
       "      <td>17909</td>\n",
       "      <td>128983</td>\n",
       "      <td>8388</td>\n",
       "      <td>117614</td>\n",
       "      <td>18257</td>\n",
       "      <td>59274</td>\n",
       "      <td>44966</td>\n",
       "      <td>116637</td>\n",
       "      <td>100729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longest scaffold</th>\n",
       "      <td>103078</td>\n",
       "      <td>1666795</td>\n",
       "      <td>120572</td>\n",
       "      <td>76417</td>\n",
       "      <td>119079</td>\n",
       "      <td>154591</td>\n",
       "      <td>291792</td>\n",
       "      <td>140813</td>\n",
       "      <td>1931651</td>\n",
       "      <td>270640</td>\n",
       "      <td>...</td>\n",
       "      <td>98428</td>\n",
       "      <td>40716</td>\n",
       "      <td>181778</td>\n",
       "      <td>35746</td>\n",
       "      <td>414343</td>\n",
       "      <td>109207</td>\n",
       "      <td>119078</td>\n",
       "      <td>104203</td>\n",
       "      <td>234605</td>\n",
       "      <td>293218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC</th>\n",
       "      <td>0.559415</td>\n",
       "      <td>0.555948</td>\n",
       "      <td>0.601188</td>\n",
       "      <td>0.564028</td>\n",
       "      <td>0.555444</td>\n",
       "      <td>0.547508</td>\n",
       "      <td>0.574851</td>\n",
       "      <td>0.551633</td>\n",
       "      <td>0.412554</td>\n",
       "      <td>0.554079</td>\n",
       "      <td>...</td>\n",
       "      <td>0.620404</td>\n",
       "      <td>0.564198</td>\n",
       "      <td>0.543219</td>\n",
       "      <td>0.616806</td>\n",
       "      <td>0.551546</td>\n",
       "      <td>0.613088</td>\n",
       "      <td>0.627406</td>\n",
       "      <td>0.553595</td>\n",
       "      <td>0.560651</td>\n",
       "      <td>0.516718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC std</th>\n",
       "      <td>0.016983</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0170428</td>\n",
       "      <td>0.0200564</td>\n",
       "      <td>0.0181341</td>\n",
       "      <td>0.013905</td>\n",
       "      <td>0.0176275</td>\n",
       "      <td>0.019404</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0175766</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0187164</td>\n",
       "      <td>0.0189734</td>\n",
       "      <td>0.0162278</td>\n",
       "      <td>0.0220273</td>\n",
       "      <td>0.0109465</td>\n",
       "      <td>0.0215446</td>\n",
       "      <td>0.0215979</td>\n",
       "      <td>0.0177516</td>\n",
       "      <td>0.0146937</td>\n",
       "      <td>0.0184419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># predicted genes</th>\n",
       "      <td>1605</td>\n",
       "      <td>1636</td>\n",
       "      <td>1279</td>\n",
       "      <td>1152</td>\n",
       "      <td>1206</td>\n",
       "      <td>1503</td>\n",
       "      <td>1564</td>\n",
       "      <td>1187</td>\n",
       "      <td>1855</td>\n",
       "      <td>1240</td>\n",
       "      <td>...</td>\n",
       "      <td>2237</td>\n",
       "      <td>1057</td>\n",
       "      <td>1348</td>\n",
       "      <td>1374</td>\n",
       "      <td>1212</td>\n",
       "      <td>2144</td>\n",
       "      <td>2497</td>\n",
       "      <td>1192</td>\n",
       "      <td>1278</td>\n",
       "      <td>1521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translation table</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coding density</th>\n",
       "      <td>0.892202</td>\n",
       "      <td>0.893948</td>\n",
       "      <td>0.899556</td>\n",
       "      <td>0.914932</td>\n",
       "      <td>0.870157</td>\n",
       "      <td>0.890325</td>\n",
       "      <td>0.900677</td>\n",
       "      <td>0.883084</td>\n",
       "      <td>0.882898</td>\n",
       "      <td>0.900474</td>\n",
       "      <td>...</td>\n",
       "      <td>0.797884</td>\n",
       "      <td>0.918026</td>\n",
       "      <td>0.925829</td>\n",
       "      <td>0.909772</td>\n",
       "      <td>0.91466</td>\n",
       "      <td>0.821946</td>\n",
       "      <td>0.881767</td>\n",
       "      <td>0.886052</td>\n",
       "      <td>0.912572</td>\n",
       "      <td>0.893666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>marker lineage</th>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryarchaeota</td>\n",
       "      <td>p__Euryar"
      ],
      "text/plain": [
       "                             V11_RL001   GCA_000300255.2   GCA_002503545.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   98.3871           98.3871           88.3788   \n",
       "Contamination                        0          0.806452           0.94086   \n",
       "Genome size                    1590932           1666795           1249378   \n",
       "# ambiguous bases                    0                 0              1306   \n",
       "# contigs                           48                 1                56   \n",
       "Mean contig length             33144.4        1.6668e+06             22287   \n",
       "N50 (contigs)                    50438           1666795             45710   \n",
       "Longest contig                  103078           1666795             99237   \n",
       "# scaffolds                         48                 1                34   \n",
       "Mean scaffold length           33144.4        1.6668e+06           36746.4   \n",
       "N50 (scaffolds)                  50438           1666795             54472   \n",
       "Longest scaffold                103078           1666795            120572   \n",
       "GC                            0.559415          0.555948          0.601188   \n",
       "GC std                        0.016983                 0         0.0170428   \n",
       "# predicted genes                 1605              1636              1279   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.892202          0.893948          0.899556   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               187               187   \n",
       "# marker sets                      124               124               124   \n",
       "\n",
       "                       GCA_002495495.1   GCA_002506865.1   GCA_002499085.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   86.2068           83.3633              76.8   \n",
       "Contamination                 0.403226           4.03226               2.4   \n",
       "Genome size                    1063745           1104259           1749312   \n",
       "# ambiguous bases                 6646             20303               479   \n",
       "# contigs                          107               175                46   \n",
       "Mean contig length             9879.43           6194.03           38018.1   \n",
       "N50 (contigs)                    16791             13507             56845   \n",
       "Longest contig                   55665             44660            154585   \n",
       "# scaffolds                         80                58                37   \n",
       "Mean scaffold length           13296.8           19038.9           47278.7   \n",
       "N50 (scaffolds)                  18039             41159             73068   \n",
       "Longest scaffold                 76417            119079            154591   \n",
       "GC                            0.564028          0.555444          0.547508   \n",
       "GC std                       0.0200564         0.0181341          0.013905   \n",
       "# predicted genes                 1152              1206              1503   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.914932          0.870157          0.890325   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               187               188   \n",
       "# marker sets                      124               124               125   \n",
       "\n",
       "                       GCA_002494805.1   GCA_002502545.1   GCA_000404225.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   92.3387           88.4904           98.7903   \n",
       "Contamination                 0.806452                 0          0.806452   \n",
       "Genome size                    1648656           1115929           1931651   \n",
       "# ambiguous bases                  790             17388                 0   \n",
       "# contigs                           47               135                 1   \n",
       "Mean contig length               35061           8137.34       1.93165e+06   \n",
       "N50 (contigs)                    64245             25662           1931651   \n",
       "Longest contig                  269292            103911           1931651   \n",
       "# scaffolds                         36                38                 1   \n",
       "Mean scaffold length             45796           29366.6       1.93165e+06   \n",
       "N50 (scaffolds)                  66041             69703           1931651   \n",
       "Longest scaffold                291792            140813           1931651   \n",
       "GC                            0.574851          0.551633          0.412554   \n",
       "GC std                       0.0176275          0.019404                 0   \n",
       "# predicted genes                 1564              1187              1855   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.900677          0.883084          0.882898   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               187               187   \n",
       "# marker sets                      124               124               124   \n",
       "\n",
       "                       GCA_002498765.1        ...          GCA_002498285.1  \\\n",
       "Stat                                          ...                            \n",
       "Completeness                   95.0775        ...                  84.7177   \n",
       "Contamination                        0        ...                        0   \n",
       "Genome size                    1174624        ...                  1827384   \n",
       "# ambiguous bases                10307        ...                    98446   \n",
       "# contigs                           66        ...                      572   \n",
       "Mean contig length             17641.2        ...                  3022.62   \n",
       "N50 (contigs)                   135661        ...                     7077   \n",
       "Longest contig                  270640        ...                    71752   \n",
       "# scaffolds                         29        ...                      110   \n",
       "Mean scaffold length           40504.3        ...                  16612.6   \n",
       "N50 (scaffolds)                 163880        ...                    30094   \n",
       "Longest scaffold                270640        ...                    98428   \n",
       "GC                            0.554079        ...                 0.620404   \n",
       "GC std                       0.0175766        ...                0.0187164   \n",
       "# predicted genes                 1240        ...                     2237   \n",
       "Translation table                   11        ...                       11   \n",
       "Coding density                0.900474        ...                 0.797884   \n",
       "marker lineage        p__Euryarchaeota        ...         p__Euryarchaeota   \n",
       "# genomes                          148        ...                      148   \n",
       "# markers                          187        ...                      187   \n",
       "# marker sets                      124        ...                      124   \n",
       "\n",
       "                       GCA_002509465.1   GCA_002503785.1   GCA_900313075.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   84.7455           96.9086           82.2581   \n",
       "Contamination                        0                 0           2.41935   \n",
       "Genome size                     969311           1308604           1262884   \n",
       "# ambiguous bases                 5047               383                 0   \n",
       "# contigs                          104                22               192   \n",
       "Mean contig length             9271.77           59464.6           6577.52   \n",
       "N50 (contigs)                    15368            102142              8388   \n",
       "Longest contig                   40716            181776             35746   \n",
       "# scaffolds                         69                12               192   \n",
       "Mean scaffold length             14048            109050           6577.52   \n",
       "N50 (scaffolds)                  17909            128983              8388   \n",
       "Longest scaffold                 40716            181778             35746   \n",
       "GC                            0.564198          0.543219          0.616806   \n",
       "GC std                       0.0189734         0.0162278         0.0220273   \n",
       "# predicted genes                 1057              1348              1374   \n",
       "Translation table                   11                11                11   \n",
       "Coding density                0.918026          0.925829          0.909772   \n",
       "marker lineage        p__Euryarchaeota  p__Euryarchaeota  p__Euryarchaeota   \n",
       "# genomes                          148               148               148   \n",
       "# markers                          187               187               187   \n",
       "# marker sets                      124               124               124   \n",
       "\n",
       "                       GCA_002505245.1   GCA_002508585.1   GCA_002508545.1  \\\n",
       "Stat                                                                         \n",
       "Completeness                   95.5645           88.3002           97.3118   \n",
       "Contamination                        0          0.806452          0.806452   \n",
       "Genome size                    1175704           1883692           2360299   \n",
       "# ambiguous bases                 5963             74196              2392   \n",
       "# contigs                           49               464               111   \n",
       "Mean contig length             23872.3           3899.78           21242.4   \n",
       "N50 (contigs)                    46981              7813             46502   \n",
       "Longest contig                  122573             40576            119060   \n",
       "# scaffolds                         16               144                68   \n",
       "Mean scaffold length           73481.5           13081.2           34710.3   \n",
       "N50 (scaffolds)                 117614             18257             59274   \n",
       "Longest sca"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 12564 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "include_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-10T13:41:26.464515Z",
     "start_time": "2018-06-10T13:41:26.457172Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GCA_002495125.1',\n",
       " 'GCA_002497815.1',\n",
       " 'GCA_002505735.1',\n",
       " 'GCA_002502165.1',\n",
       " 'GCA_002497195.1',\n",
       " 'GCA_002494945.1',\n",
       " 'GCA_002506255.1']"
      ]
     },
     "execution_count": 246,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exclude_df.columns.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-11T06:43:26.160477Z",
     "start_time": "2018-06-11T06:43:26.145374Z"
    }
   },
   "outputs": [],
   "source": [
    "# Write file with the paths of the V11 bin and the included genomes\n",
    "include_file = os.path.join(download_folder, \"included_genomes.txt\")\n",
    "with open(include_file, \"a\") as myfile:\n",
    "    myfile.write(V11_bin + \"\\n\") # V11\n",
    "    for inclusion in include_df.columns.tolist():\n",
    "        filename = os.path.join (download_folder, (inclusion + \".fna\"))\n",
    "        myfile.write(filename + \"\\n\") # Each included genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# *Thermoplasmata* genomes\n",
    "For the phylogenomic analyses, I need an outgroup. For this I will use several genomes from organisms of the class Thermoplasmata. I will download their genome sequences and proteomes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T08:56:42.877294Z",
     "start_time": "2019-01-25T08:56:42.870890Z"
    }
   },
   "outputs": [],
   "source": [
    "# Create folder and save accessions\n",
    "Thermoplasmata_folder = os.path.join(work_dir, \"genomes\", \"Thermoplasmata\")\n",
    "if not os.path.exists(Thermoplasmata_folder):\n",
    "    os.makedirs(Thermoplasmata_folder)\n",
    "Thermoplasmata_list = os.path.join(Thermoplasmata_folder, \"Thermoplasmata.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T09:02:58.081360Z",
     "start_time": "2019-01-25T09:02:58.074182Z"
    }
   },
   "outputs": [],
   "source": [
    "# I will include RefSeq representative genomes\n",
    "Thermoplasmata_members = [\"GCA_000195915.1\", \"GCA_900176435.1\", \n",
    "                          \"GCA_000152265.2\", \"GCA_001402945.1\", \n",
    "                          \"GCA_900090055.1\"]\n",
    "\n",
    "Thermoplasmata_newname = [\"GCA_000195915\", \"GCA_900176435\", \n",
    "                          \"GCA_000152265\", \"GCA_001402945\", \n",
    "                          \"GCA_900090055\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T09:03:06.409315Z",
     "start_time": "2019-01-25T09:02:58.976965Z"
    }
   },
   "outputs": [],
   "source": [
    "# Download genome assemblies in fasta format from NCBI\n",
    "Thermoplasmata = get_members(Thermoplasmata_members, \n",
    "                                Thermoplasmata_folder, \n",
    "                                Thermoplasmata_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T09:03:06.649476Z",
     "start_time": "2019-01-25T09:03:06.412005Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gunzip /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/*.fna.gz\n"
     ]
    }
   ],
   "source": [
    "# Uncompress genome files\n",
    "gunzip_job = \"gunzip {0}/*.fna.gz\".format(Thermoplasmata_folder)\n",
    "print(gunzip_job)\n",
    "!$gunzip_job"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T09:03:06.661260Z",
     "start_time": "2019-01-25T09:03:06.652581Z"
    }
   },
   "outputs": [],
   "source": [
    "# Fix file names\n",
    "for f in os.listdir(Thermoplasmata_folder):\n",
    "    f_new = re.sub('_[a-zA-Z].*_genomic', '', f)\n",
    "    os.rename(f, f.replace(f, f_new))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T09:03:06.685308Z",
     "start_time": "2019-01-25T09:03:06.664001Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASM19591v1</td>\n",
       "      <td>Thermoplasma acidophilum DSM 1728</td>\n",
       "      <td>GCA_000195915.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IMG-taxon 2579779151 annotated assembly</td>\n",
       "      <td>Picrophilus oshimae DSM 9789</td>\n",
       "      <td>GCA_900176435.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASM15226v2</td>\n",
       "      <td>Ferroplasma acidarmanus fer1</td>\n",
       "      <td>GCA_000152265.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASM140294v1</td>\n",
       "      <td>Acidiplasma aeolicum</td>\n",
       "      <td>GCA_001402945.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C.divulgatum PM4</td>\n",
       "      <td>Cuniculiplasma divulgatum</td>\n",
       "      <td>GCA_900090055.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Assembly                           Organism  \\\n",
       "0                               ASM19591v1  Thermoplasma acidophilum DSM 1728   \n",
       "1  IMG-taxon 2579779151 annotated assembly       Picrophilus oshimae DSM 9789   \n",
       "2                               ASM15226v2       Ferroplasma acidarmanus fer1   \n",
       "3                              ASM140294v1               Acidiplasma aeolicum   \n",
       "4                         C.divulgatum PM4          Cuniculiplasma divulgatum   \n",
       "\n",
       "         Accession  \n",
       "0  GCA_000195915.1  \n",
       "1  GCA_900176435.1  \n",
       "2  GCA_000152265.2  \n",
       "3  GCA_001402945.1  \n",
       "4  GCA_900090055.1  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print table with accessions\n",
    "Thermoplasmata_df = pd.read_csv(Thermoplasmata_list, sep=',', header=0)\n",
    "Thermoplasmata_df = Thermoplasmata_df.drop([\"Unnamed: 0\"], axis=1)\n",
    "Thermoplasmata_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T09:11:29.309477Z",
     "start_time": "2019-01-25T09:11:20.186161Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/195/915/GCF_000195915.1_ASM19591v1/GCF_000195915.1_ASM19591v1_protein.faa.gz --output-document /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_000195915.faa.gz; gunzip /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_000195915.faa.gz\n",
      "--2019-01-25 10:11:20--  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/195/915/GCF_000195915.1_ASM19591v1/GCF_000195915.1_ASM19591v1_protein.faa.gz\n",
      "           => ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_000195915.faa.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 2607:f220:41e:250::11\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.7|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /genomes/all/GCF/000/195/915/GCF_000195915.1_ASM19591v1 ... done.\n",
      "==> SIZE GCF_000195915.1_ASM19591v1_protein.faa.gz ... 298504\n",
      "==> PASV ... done.    ==> RETR GCF_000195915.1_ASM19591v1_protein.faa.gz ... done.\n",
      "Length: 298504 (292K) (unauthoritative)\n",
      "\n",
      "GCF_000195915.1_ASM 100%[===================>] 291.51K   750KB/s    in 0.4s    \n",
      "\n",
      "2019-01-25 10:11:21 (750 KB/s) - ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_000195915.faa.gz’ saved [298504]\n",
      "\n",
      "wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900/176/435/GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly/GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly_protein.faa.gz --output-document /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_900176435.faa.gz; gunzip /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_900176435.faa.gz\n",
      "--2019-01-25 10:11:22--  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900/176/435/GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly/GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly_protein.faa.gz\n",
      "           => ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_900176435.faa.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 2607:f220:41e:250::11\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.7|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /genomes/all/GCF/900/176/435/GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly ... done.\n",
      "==> SIZE GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly_protein.faa.gz ... 300255\n",
      "==> PASV ... done.    ==> RETR GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly_protein.faa.gz ... done.\n",
      "Length: 300255 (293K) (unauthoritative)\n",
      "\n",
      "GCF_900176435.1_IMG 100%[===================>] 293.22K   755KB/s    in 0.4s    \n",
      "\n",
      "2019-01-25 10:11:23 (755 KB/s) - ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_900176435.faa.gz’ saved [300255]\n",
      "\n",
      "wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/152/265/GCF_000152265.2_ASM15226v2/GCF_000152265.2_ASM15226v2_protein.faa.gz --output-document /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_000152265.faa.gz; gunzip /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_000152265.faa.gz\n",
      "--2019-01-25 10:11:23--  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/152/265/GCF_000152265.2_ASM15226v2/GCF_000152265.2_ASM15226v2_protein.faa.gz\n",
      "           => ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_000152265.faa.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 2607:f220:41e:250::11\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.7|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /genomes/all/GCF/000/152/265/GCF_000152265.2_ASM15226v2 ... done.\n",
      "==> SIZE GCF_000152265.2_ASM15226v2_protein.faa.gz ... 362596\n",
      "==> PASV ... done.    ==> RETR GCF_000152265.2_ASM15226v2_protein.faa.gz ... done.\n",
      "Length: 362596 (354K) (unauthoritative)\n",
      "\n",
      "GCF_000152265.2_ASM 100%[===================>] 354.10K   912KB/s    in 0.4s    \n",
      "\n",
      "2019-01-25 10:11:25 (912 KB/s) - ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_000152265.faa.gz’ saved [362596]\n",
      "\n",
      "wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/402/945/GCF_001402945.1_ASM140294v1/GCF_001402945.1_ASM140294v1_protein.faa.gz --output-document /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_001402945.faa.gz; gunzip /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_001402945.faa.gz\n",
      "--2019-01-25 10:11:25--  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/402/945/GCF_001402945.1_ASM140294v1/GCF_001402945.1_ASM140294v1_protein.faa.gz\n",
      "           => ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_001402945.faa.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 2607:f220:41e:250::11\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.7|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /genomes/all/GCF/001/402/945/GCF_001402945.1_ASM140294v1 ... done.\n",
      "==> SIZE GCF_001402945.1_ASM140294v1_protein.faa.gz ... 322914\n",
      "==> PASV ... done.    ==> RETR GCF_001402945.1_ASM140294v1_protein.faa.gz ... done.\n",
      "Length: 322914 (315K) (unauthoritative)\n",
      "\n",
      "GCF_001402945.1_ASM 100%[===================>] 315.35K   814KB/s    in 0.4s    \n",
      "\n",
      "2019-01-25 10:11:27 (814 KB/s) - ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_001402945.faa.gz’ saved [322914]\n",
      "\n",
      "wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900/090/055/GCF_900090055.1_C.divulgatum_PM4/GCF_900090055.1_C.divulgatum_PM4_protein.faa.gz --output-document /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_900090055.faa.gz; gunzip /ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_900090055.faa.gz\n",
      "--2019-01-25 10:11:27--  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900/090/055/GCF_900090055.1_C.divulgatum_PM4/GCF_900090055.1_C.divulgatum_PM4_protein.faa.gz\n",
      "           => ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_900090055.faa.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 2607:f220:41e:250::11\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.7|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /genomes/all/GCF/900/090/055/GCF_900090055.1_C.divulgatum_PM4 ... done.\n",
      "==> SIZE GCF_900090055.1_C.divulgatum_PM4_protein.faa.gz ... 354026\n",
      "==> PASV ... done.    ==> RETR GCF_900090055.1_C.divulgatum_PM4_protein.faa.gz ... done.\n",
      "Length: 354026 (346K) (unauthoritative)\n",
      "\n",
      "GCF_900090055.1_C.d 100%[===================>] 345.73K   879KB/s    in 0.4s    \n",
      "\n",
      "2019-01-25 10:11:29 (879 KB/s) - ‘/ebio/abt3_projects/vadinCA11/data/V11/genomes/Thermoplasmata/GCA_900090055.faa.gz’ saved [354026]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Download proteomes\n",
    "# Download Thermoplasmata prots\n",
    "Thermoplasma = \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/195/915/GCF_000195915.1_ASM19591v1/GCF_000195915.1_ASM19591v1_protein.faa.gz\"\n",
    "Picrophilus = \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900/176/435/GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly/GCF_900176435.1_IMG-taxon_2579779151_annotated_assembly_protein.faa.gz\"\n",
    "Ferroplasma = \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/152/265/GCF_000152265.2_ASM15226v2/GCF_000152265.2_ASM15226v2_protein.faa.gz\"\n",
    "Acidiplasma = \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/402/945/GCF_001402945.1_ASM140294v1/GCF_001402945.1_ASM140294v1_protein.faa.gz\"\n",
    "Cuniculiplasma = \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900/090/055/GCF_900090055.1_C.divulgatum_PM4/GCF_900090055.1_C.divulgatum_PM4_protein.faa.gz\"\n",
    "\n",
    "Thermoplasmata_prots = [Thermoplasma, Picrophilus, Ferroplasma, Acidiplasma, Cuniculiplasma]\n",
    "\n",
    "# download command\n",
    "download_cmd = \"wget {0} --output-document {1}/{3}.faa.gz; gunzip {1}/{3}.faa.gz\"\n",
    "for T in range(0, len(Thermoplasmata_prots)):\n",
    "    download_job = download_cmd.format(Thermoplasmata_prots[T], \n",
    "                                       Thermoplasmata_folder, \n",
    "                                       Thermoplasmata_members[T], \n",
    "                                       Thermoplasmata_newname[T])\n",
    "    print(download_job)\n",
    "    !$download_job"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Session info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T09:12:02.065362Z",
     "start_time": "2019-01-25T09:11:55.300180Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# packages in environment at /ebio/abt3_projects/software/miniconda3_gt4.4/envs/metacompass:\r\n",
      "#\r\n",
      "# Name                    Version                   Build  Channel\r\n",
      "aioeasywebdav             2.2.0                    py36_0    conda-forge\r\n",
      "aiohttp                   3.4.4            py36h470a237_0    conda-forge\r\n",
      "anvio                     5.1.0            py36hcb787e7_1    bioconda\r\n",
      "appdirs                   1.4.3                      py_1    conda-forge\r\n",
      "aragorn                   1.2.38               h470a237_2    bioconda\r\n",
      "asn1crypto                0.24.0                     py_1    conda-forge\r\n",
      "async-timeout             3.0.1                   py_1000    conda-forge\r\n",
      "attrs                     18.1.0                     py_1    conda-forge\r\n",
      "automat                   0.7.0                    py36_0    conda-forge\r\n",
      "backcall                  0.1.0                      py_0    conda-forge\r\n",
      "barrnap                   0.9                           2    bioconda\r\n",
      "bcftools                  1.9                  h4da6232_0    bioconda\r\n",
      "bcrypt                    3.1.4            py36h470a237_0    conda-forge\r\n",
      "bedtools                  2.27.1               he941832_2    bioconda\r\n",
      "biopython                 1.72                     py36_0    conda-forge\r\n",
      "blas                      1.1                    openblas    conda-forge\r\n",
      "blast                     2.7.1                h4422958_6    bioconda\r\n",
      "blast-legacy              2.2.26                        1    bioconda\r\n",
      "bleach                    2.1.3                      py_0    conda-forge\r\n",
      "boost                     1.67.0           py36h3e44d54_0    conda-forge\r\n",
      "boost-cpp                 1.67.0               h3a22d5f_0    conda-forge\r\n",
      "boto3                     1.7.84                     py_0    conda-forge\r\n",
      "botocore                  1.10.84                    py_0    conda-forge\r\n",
      "bottle                    0.12.13                    py_1    conda-forge\r\n",
      "bzip2                     1.0.6                h470a237_2    conda-forge\r\n",
      "ca-certificates           2018.4.16                     0    conda-forge\r\n",
      "cachetools                2.1.0                      py_0    conda-forge\r\n",
      "cairo                     1.14.12              he56eebe_3    conda-forge\r\n",
      "certifi                   2018.4.16                py36_0    conda-forge\r\n",
      "cffi                      1.11.5           py36h5e8e0c9_1    conda-forge\r\n",
      "chardet                   3.0.4                 py36_1003    conda-forge\r\n",
      "cherrypy                  8.0.0                    py36_0    conda-forge\r\n",
      "configargparse            0.13.0                     py_1    conda-forge\r\n",
      "constantly                15.1.0                     py_0    conda-forge\r\n",
      "cryptography              2.3              py36hdffb7b8_0    conda-forge\r\n",
      "cryptography-vectors      2.3                      py36_1    conda-forge\r\n",
      "curl                      7.61.0               h93b3f91_1    conda-forge\r\n",
      "cycler                    0.10.0                     py_1    conda-forge\r\n",
      "datrie                    0.7.1                    py36_0  \r\n",
      "dbus                      1.13.0               h3a4f0e9_0    conda-forge\r\n",
      "decorator                 4.3.0                      py_0    conda-forge\r\n",
      "django                    2.0.2                    py36_0    conda-forge\r\n",
      "docutils                  0.14                  py36_1001    conda-forge\r\n",
      "dropbox                   7.3.1                    py36_0    conda-forge\r\n",
      "entrypoints               0.2.3                    py36_1    conda-forge\r\n",
      "ete3                      3.1.1                    py36_0    bioconda\r\n",
      "expat                     2.2.5                hfc679d8_2    conda-forge\r\n",
      "filechunkio               1.6                      py36_0    bioconda\r\n",
      "fontconfig                2.13.0               h0c61d53_4    conda-forge\r\n",
      "freetype                  2.8.1                hfa320df_1    conda-forge\r\n",
      "ftputil                   3.2                      py36_0    bioconda\r\n",
      "gettext                   0.19.8.1                      0    conda-forge\r\n",
      "glib                      2.55.0               h464dc38_2    conda-forge\r\n",
      "gmp                       6.1.2                hfc679d8_0    conda-forge\r\n",
      "gnutls                    3.5.19               h2a4e5f8_1    conda-forge\r\n",
      "google-auth               1.6.1                      py_0    conda-forge\r\n",
      "google-auth-httplib2      0.0.3                      py_2    conda-forge\r\n",
      "google-cloud-core         0.24.1                   py36_0    conda-forge\r\n",
      "google-cloud-storage      1.1.1                    py36_0    conda-forge\r\n",
      "google-resumable-media    0.0.2                    py36_0    conda-forge\r\n",
      "googleapis-common-protos  1.5.5                      py_0    conda-forge\r\n",
      "graphite2                 1.3.11               hfc679d8_0    conda-forge\r\n",
      "gsl                       2.2.1           blas_openblas_4  [blas_openblas]  conda-forge\r\n",
      "gst-plugins-base          1.12.5               hde13a9d_0    conda-forge\r\n",
      "gstreamer                 1.12.5               h61a6719_0    conda-forge\r\n",
      "h5py                      2.8.0            py36he5c79e1_5    conda-forge\r\n",
      "harfbuzz                  1.8.5                h2bb21d5_0    conda-forge\r\n",
      "hdf5                      1.10.4          nompi_h5598ddc_1003    conda-forge\r\n",
      "hmmer                     3.2.1                hfc679d8_0    bioconda\r\n",
      "html5lib                  1.0.1                      py_0    conda-forge\r\n",
      "htslib                    1.7                           0    bioconda\r\n",
      "httplib2                  0.12.0                py36_1000    conda-forge\r\n",
      "hyperlink                 17.3.1                     py_0    conda-forge\r\n",
      "icu                       58.2                 hfc679d8_0    conda-forge\r\n",
      "idna                      2.7                      py36_2    conda-forge\r\n",
      "idna_ssl                  1.0.0                         0    conda-forge\r\n",
      "incremental               17.5.0                     py_0    conda-forge\r\n",
      "infernal                  1.1.2                h470a237_1    bioconda\r\n",
      "ipykernel                 4.8.2                    py36_0    conda-forge\r\n",
      "ipython                   6.5.0                    py36_0    conda-forge\r\n",
      "ipython_genutils          0.2.0                      py_1    conda-forge\r\n",
      "java-jdk                  8.0.92                        1    bioconda\r\n",
      "jedi                      0.12.1                   py36_0    conda-forge\r\n",
      "jellyfish                 1.1.12               h2d50403_0    bioconda\r\n",
      "jinja2                    2.10                       py_1    conda-forge\r\n",
      "jmespath                  0.9.3                      py_1    conda-forge\r\n",
      "jpeg                      9c                   h470a237_0    conda-forge\r\n",
      "jsonschema                2.6.0                    py36_1    conda-forge\r\n",
      "jupyter_client            5.2.3                      py_1    conda-forge\r\n",
      "jupyter_core              4.4.0                      py_0    conda-forge\r\n",
      "jupyterlab                0.35.4                   py36_0    conda-forge\r\n",
      "jupyterlab_server         0.2.0                      py_0    conda-forge\r\n",
      "kiwisolver                1.0.1            py36h2d50403_2    conda-forge\r\n",
      "kraken2                   2.0.7_beta      pl526h2d50403_0    bioconda\r\n",
      "krb5                      1.14.6                        0    conda-forge\r\n",
      "libdeflate                1.0                  h470a237_0    bioconda\r\n",
      "libffi                    3.2.1                         3    conda-forge\r\n",
      "libgcc                    7.2.0                h69d50b8_2  \r\n",
      "libgcc-ng                 7.2.0                hdf63c60_3  \r\n",
      "libgfortran               3.0.0                         1  \r\n",
      "libgfortran-ng            7.2.0                hdf63c60_3  \r\n",
      "libiconv                  1.15                 h470a237_1    conda-forge\r\n",
      "libidn11                  1.33                          0    conda-forge\r\n",
      "libopenblas               0.2.20               h9ac9557_7  \r\n",
      "libpng                    1.6.35               ha92aebf_0    conda-forge\r\n",
      "libprotobuf               3.6.1                hd28b015_0    conda-forge\r\n",
      "libsodium                 1.0.16                        0    conda-forge\r\n",
      "libssh2                   1.8.0                h5b517e9_2    conda-forge\r\n",
      "libstdcxx-ng              7.2.0                hdf63c60_3  \r\n",
      "libtiff                   4.0.9                he6b73bb_1    conda-forge\r\n",
      "libuuid                   1.0.3                         1    conda-forge\r\n",
      "libxcb                    1.13                 h470a237_2    conda-forge\r\n",
      "libxml2                   2.9.8                h422b904_2    conda-forge\r\n",
      "libxslt                   1.1.32               h88dbc4e_2    conda-forge\r\n",
      "lxml                      4.2.5            py36hc9114bc_0    conda-forge\r\n",
      "markupsafe                1.0                      py36_0    conda-forge\r\n",
      "matplotlib                2.2.2            py36h0e671d2_0  \r\n",
      "mcl                       14.137          pl526h470a237_4    bioconda\r\n",
      "minced                    0.2.0                         0    bioconda\r\n",
      "mistune                   0.7.4                    py36_0    conda-forge\r\n",
      "multidict                 4.5.1            py36h470a237_0    conda-forge\r\n",
      "mummer                    3.23                    pl526_6    bioconda\r\n",
      "muscle                    3.8.1551             h2d50403_3    bioconda\r\n",
      "nb_conda                  2.2.1                    py36_0    conda-forge\r\n",
      "nb_conda_kernels          2.1.1                    py36_1    conda-forge\r\n",
      "nbconvert                 5.3.1                      py_1    conda-forge\r\n",
      "nbformat                  4.4.0                      py_1    conda-forge\r\n",
      "ncurses                   6.1                  hfc679d8_1    conda-forge\r\n",
      "nettle                    3.3                           0    conda-forge\r\n",
      "notebook                  5.6.0                    py36_0    conda-forge\r\n",
      "numpy                     1.15.0          py36_blas_openblashd3ea46f_200  [blas_openblas]  conda-forge\r\n",
      "numpy-base                1.14.3           py36h0ea5e3f_1  \r\n",
      "openblas                  0.2.20                        8    conda-forge\r\n",
      "openssl                   1.0.2o               h470a237_1    conda-forge\r\n",
      "pandas                    0.23.4           py36hf8a1672_0    conda-forge\r\n",
      "pandoc                    2.2.2                         1    conda-forge\r\n",
      "pandocfilters             1.4.2                      py_1    conda-forge\r\n",
      "pango                     1.40.14              h9105a7a_2    conda-forge\r\n",
      "parallel                  20160622                      1    bioconda\r\n",
      "parallel-fastq-dump       0.6.3                    py36_1    bioconda\r\n",
      "paramiko                  2.4.2                 py36_1000    conda-forge\r\n",
      "parso                     0.3.1                      py_0    conda-forge\r\n",
      "patsy                     0.5.1                      py_0    conda-forge\r\n",
      "pcre                      8.41                 h470a237_2    conda-forge\r\n",
      "perl                      5.26.2               h470a237_0    conda-forge\r\n",
      "perl-app-cpanminus        1.7044                  pl526_1    bioconda\r\n",
      "perl-archive-tar          2.18                    pl526_3    bioconda\r\n",
      "perl-bioperl              1.6.924                       4    bioconda\r\n",
      "perl-carp                 1.38                    pl526_1    bioconda\r\n",
      "perl-compress-raw-bzip2   2.074           pl526hfc679d8_0    bioconda\r\n",
      "perl-compress-raw-zlib    2.081           pl526h2d50403_0    bioconda\r\n",
      "perl-constant             1.33                    pl526_1    bioconda\r\n",
      "perl-data-dumper          2.161                   pl526_2    bioconda\r\n",
      "perl-encode               2.88                    pl526_1    bioconda\r\n",
      "perl-exporter             5.72                    pl526_1    bioconda\r\n",
      "perl-exporter-tiny        1.000000                pl526_0    bioconda\r\n",
      "perl-extutils-makemaker   7.34                    pl526_2    bioconda\r\n",
      "perl-file-path            2.15                    pl526_0    bioconda\r\n",
      "perl-file-temp            0.2304                  pl526_2    bioconda\r\n",
      "perl-io-compress          2.069           pl526hfc679d8_5    bioconda\r\n",
      "perl-io-zlib              1.10                    pl526_2    bioconda\r\n",
      "perl-list-moreutils       0.428                   pl526_1    bioconda\r\n",
      "perl-list-moreutils-xs    0.428                   pl526_0    bioconda\r\n",
      "perl-parent               0.236                   pl526_1    bioconda\r\n",
      "perl-pathtools            3.73                 h470a237_2    bioconda\r\n",
      "perl-scalar-list-utils    1.45            pl526h470a237_3    bioconda\r\n",
      "perl-test-more            1.001002                pl526_1    bioconda\r\n",
      "perl-threaded             5.22.0                       13    bioconda\r\n",
      "perl-xml-namespacesupport 1.12                    pl526_0    bioconda\r\n",
      "perl-xml-parser           2.44            pl526h3a4f0e9_6    bioconda\r\n",
      "perl-xml-sax              1.00                    pl526_0    bioconda\r\n",
      "perl-xml-sax-base         1.09                    pl526_0    bioconda\r\n",
      "perl-xml-sax-expat        0.51                    pl526_2    bioconda\r\n",
      "perl-xml-simple           2.22                    pl526_3    bioconda\r\n",
      "perl-xsloader             0.24                    pl526_0    bioconda\r\n",
      "perl-yaml                 1.26                    pl526_1    bioconda\r\n",
      "pexpect                   4.6.0                    py36_0    conda-forge\r\n",
      "pickleshare               0.7.4                    py36_0    conda-forge\r\n",
      "pip                       18.0                     py36_1    conda-forge\r\n",
      "pixman                    0.34.0                        2    conda-forge\r\n",
      "prettytable               0.7.2                      py_3    conda-forge\r\n",
      "prodigal                  2.6.3                         1    bioconda\r\n",
      "prokka                    1.13                          2    bioconda\r\n",
      "prometheus_client         0.3.0                      py_0    conda-forge\r\n",
      "prompt_toolkit            1.0.15                   py36_0    conda-forge\r\n",
      "protobuf                  3.6.1            py36hfc679d8_1    conda-forge\r\n",
      "psutil                    5.4.8            py36h470a237_0    conda-forge\r\n",
      "pthread-stubs             0.4                  h470a237_1    conda-forge\r\n",
      "ptyprocess                0.6.0                    py36_0    conda-forge\r\n",
      "pyani                     0.2.7            py36h24bf2e0_1    bioconda\r\n",
      "pyasn1                    0.4.4                      py_0    conda-forge\r\n",
      "pyasn1-modules            0.2.1                      py_0    conda-forge\r\n",
      "pycparser                 2.18                       py_1    conda-forge\r\n",
      "pygments                  2.2.0                      py_1    conda-forge\r\n",
      "pyhamcrest                1.9.0                      py_2    conda-forge\r\n",
      "pynacl                    1.3.0            py36h470a237_0    conda-forge\r\n",
      "pyopenssl                 18.0.0                   py36_0    conda-forge\r\n",
      "pyparsing                 2.2.2                      py_0    conda-forge\r\n",
      "pyqt                      5.6.0            py36h8210e8a_7    conda-forge\r\n",
      "pysam                     0.14.1           py36hae42fb6_1    bioconda\r\n",
      "pysftp                    0.2.9                    py36_0    bioconda\r\n",
      "python                    3.6.6                h5001a0f_0    conda-forge\r\n",
      "python-dateutil           2.7.3                      py_0    conda-forge\r\n",
      "python-irodsclient        0.7.0                      py_0    conda-forge\r\n",
      "pytz                      2018.5                     py_0    conda-forge\r\n",
      "pyyaml                    3.13             py36h470a237_1    conda-forge\r\n",
      "pyzmq                     17.1.0           py36hae99301_0    conda-forge\r\n",
      "qt                        5.6.2                h50c60fd_8    conda-forge\r\n",
      "r-assertthat              0.2.0                  r3.4.1_0    conda-forge\r\n",
      "r-base                    3.4.1                h4fe35fd_8    conda-forge\r\n",
      "r-base64enc               0.1_3            r341h470a237_1    conda-forge\r\n",
      "r-bh                      1.66.0_1               r3.4.1_0    conda-forge\r\n",
      "r-bindr                   0.1.1                  r3.4.1_0    conda-forge\r\n",
      "r-bindrcpp                0.2.2            r341h9d2a408_0    conda-forge\r\n",
      "r-bit                     1.1_12           r341h470a237_1    conda-forge\r\n",
      "r-bit64                   0.9_5            r341h470a237_1    conda-forge\r\n",
      "r-blob                    1.1.1                  r3.4.1_0    conda-forge\r\n",
      "r-boot                    1.3_19                 r3.4.1_0    r\r\n",
      "r-cli                     1.0.0                  r3.4.1_0    conda-forge\r\n",
      "r-crayon                  1.3.4                  r3.4.1_0    conda-forge\r\n",
      "r-dbi                     1.0.0                    r341_0    conda-forge\r\n",
      "r-dbplyr                  1.2.2                    r341_0    conda-forge\r\n",
      "r-digest                  0.6.15           r341h470a237_1    conda-forge\r\n",
      "r-dplyr                   0.7.6            r341h9d2a408_0    conda-forge\r\n",
      "r-evaluate                0.10.1                 r3.4.1_0    conda-forge\r\n",
      "r-glue                    1.3.0            r341h470a237_1    conda-forge\r\n",
      "r-htmltools               0.3.6            r341hfc679d8_1    conda-forge\r\n",
      "r-irdisplay               0.4.4                  r3.4.1_0    conda-forge\r\n",
      "r-irkernel                0.8.12                   r341_0    conda-forge\r\n",
      "r-jsonlite                1.5              r341h470a237_1    conda-forge\r\n",
      "r-magrittr                1.5                    r3.4.1_0    conda-forge\r\n",
      "r-memoise                 1.1.0                  r3.4.1_0    conda-forge\r\n",
      "r-network                 1.13.0.1         r341hc070d10_0    conda-forge\r\n",
      "r-pbdzmq                  0.3_2                  r3.4.1_0    conda-forge\r\n",
      "r-pillar                  1.2.2            r341h6115d3f_1    conda-forge\r\n",
      "r-pkgconfig               2.0.1                  r3.4.1_0    conda-forge\r\n",
      "r-plogr                   0.2.0            r341h6115d3f_0    conda-forge\r\n",
      "r-prettyunits             1.0.2                  r3.4.1_0    conda-forge\r\n",
      "r-purrr                   0.2.4            r341h470a237_1    conda-forge\r\n",
      "r-r6                      2.2.2                  r3.4.1_0    conda-forge\r\n",
      "r-rcpp                    0.12.17          r341h9d2a408_1    conda-forge\r\n",
      "r-repr                    0.15                     r341_0    conda-forge\r\n",
      "r-rlang                   0.2.1            r341h470a237_1    conda-forge\r\n",
      "r-rsqlite                 2.0              r341hfc679d8_1    conda-forge\r\n",
      "r-stringi                 1.2.3            r341hfc679d8_1    conda-forge\r\n",
      "r-stringr                 1.3.1                    r341_0    conda-forge\r\n",
      "r-tibble                  1.4.2            r341h470a237_1    conda-forge\r\n",
      "r-tidyselect              0.2.4            r341hfc679d8_1    conda-forge\r\n",
      "r-utf8                    1.1.3            r341h470a237_1    conda-forge\r\n",
      "r-uuid                    0.1_2                    r341_0    conda-forge\r\n",
      "ratelimiter               1.2.0                 py36_1000    conda-forge\r\n",
      "readline                  7.0                  haf1bffa_1    conda-forge\r\n",
      "requests                  2.13.0                   py36_0    conda-forge\r\n",
      "rpy2                      2.9.4           py36r341h24bf2e0_0    conda-forge\r\n",
      "rsa                       3.1.4                    py36_0    bioconda\r\n",
      "s3transfer                0.1.13                py36_1001    conda-forge\r\n",
      "samtools                  1.9                  h8ee4bcc_1    bioconda\r\n",
      "scikit-learn              0.19.1          py36_blas_openblas_201  [blas_openblas]  conda-forge\r\n",
      "scipy                     1.1.0           py36_blas_openblash7943236_201  [blas_openblas]  conda-forge\r\n",
      "seaborn                   0.9.0                      py_0    conda-forge\r\n",
      "send2trash                1.5.0                      py_0    conda-forge\r\n",
      "service_identity          17.0.0                     py_0    conda-forge\r\n",
      "setuptools                40.0.0                   py36_1    conda-forge\r\n",
      "simplegeneric             0.8.1                      py_1    conda-forge\r\n",
      "sip                       4.18.1           py36hfc679d8_0    conda-forge\r\n",
      "six                       1.11.0                   py36_1    conda-forge\r\n",
      "snakemake                 4.7.0                    py36_0    bioconda\r\n",
      "sqlite                    3.24.0               h2f33b56_0    conda-forge\r\n",
      "sra-tools                 2.9.1_1              h470a237_0    bioconda\r\n",
      "statsmodels               0.9.0            py36h7eb728f_0    conda-forge\r\n",
      "taxonkit                  0.2.4                         1    bioconda\r\n",
      "tbl2asn                   25.6                          3    bioconda\r\n",
      "terminado                 0.8.1                    py36_0    conda-forge\r\n",
      "testpath                  0.3.1                    py36_0    conda-forge\r\n",
      "tk                        8.6.8                         0    conda-forge\r\n",
      "tornado                   5.1              py36h470a237_1    conda-forge\r\n",
      "traitlets                 4.3.2                    py36_0    conda-forge\r\n",
      "twisted                   18.7.0           py36h14c3975_1  \r\n",
      "tzlocal                   1.5.1                      py_0    conda-forge\r\n",
      "urllib3                   1.12                     py36_0    bioconda\r\n",
      "wcwidth                   0.1.7                      py_1    conda-forge\r\n",
      "webencodings              0.5                      py36_0    conda-forge\r\n",
      "wheel                     0.31.1                   py36_1    conda-forge\r\n",
      "wrapt                     1.10.11          py36h470a237_1    conda-forge\r\n",
      "xmlrunner                 1.7.7                      py_0    conda-forge\r\n",
      "xorg-kbproto              1.0.7                h470a237_2    conda-forge\r\n",
      "xorg-libice               1.0.9                h470a237_4    conda-forge\r\n",
      "xorg-libsm                1.2.2                h470a237_5    conda-forge\r\n",
      "xorg-libx11               1.6.5                h470a237_2    conda-forge\r\n",
      "xorg-libxau               1.0.8                h470a237_6    conda-forge\r\n",
      "xorg-libxdmcp             1.1.2                h470a237_7    conda-forge\r\n",
      "xorg-libxext              1.3.3                h470a237_4    conda-forge\r\n",
      "xorg-libxrender           0.9.10               h470a237_2    conda-forge\r\n",
      "xorg-libxt                1.1.5                h470a237_2    conda-forge\r\n",
      "xorg-renderproto          0.11.1               h470a237_2    conda-forge\r\n",
      "xorg-xextproto            7.3.0                h470a237_2    conda-forge\r\n",
      "xorg-xproto               7.0.31               h470a237_7    conda-forge\r\n",
      "xz                        5.2.4                h470a237_1    conda-forge\r\n",
      "yaml                      0.1.7                h470a237_1    conda-forge\r\n",
      "yarl                      1.3.0            py36h470a237_0    conda-forge\r\n",
      "zeromq                    4.2.5                hfc679d8_5    conda-forge\r\n",
      "zlib                      1.2.11               h470a237_3    conda-forge\r\n",
      "zope.interface            4.5.0            py36h470a237_0    conda-forge\r\n"
     ]
    }
   ],
   "source": [
    "!conda list -n metacompass"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:metacompass]",
   "language": "python",
   "name": "conda-env-metacompass-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": "block",
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
