{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "Jacobo de la Cuesta-Zuluaga, May 2019.\n",
    "\n",
    "Now that I have the gene clusters from `panX`, I can obtain a better annotation of them using `InterproScan` and `eggNOG mapper`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-08-21T12:12:04.112383Z",
     "start_time": "2019-08-21T12:12:02.887558Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/ebio/abt3_projects/software/miniconda3_gt4.4/envs/py3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88\n",
      "  return f(*args, **kwds)\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import subprocess\n",
    "import json\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-08-21T12:12:04.328211Z",
     "start_time": "2019-08-21T12:12:04.114998Z"
    }
   },
   "outputs": [],
   "source": [
    "# Work dir\n",
    "work_dir = \"/ebio/abt3_projects/vadinCA11/data/V11\"\n",
    "panX_dir = os.path.join(work_dir, \"panx_output/pan-genome-analysis/data/Methanomassilii/vis/geneCluster\")\n",
    "annotation_dir = os.path.join(work_dir, \"cluster_annot\")\n",
    "if not os.path.exists(annotation_dir):\n",
    "    os.makedirs(annotation_dir)\n",
    "# Misc\n",
    "metacompass_env = \"metacompass\"\n",
    "panx_env = \"py2_panX\"\n",
    "\n",
    "# eggNOG database\n",
    "eggnog_db = \"/ebio/abt3_projects/databases_no-backup/Eggnog\"\n",
    "\n",
    "# interpro executable_dir\n",
    "interpro_bin = \"/ebio/abt3_projects/vadinCA11/bin/interpro/interproscan_5\"\n",
    "interpro_sh = os.path.join(interpro_bin, \"interproscan.sh\")\n",
    "\n",
    "# ALPs\n",
    "ALP_dir = os.path.join(work_dir, \"ALPs/Mbrevi_alp/unaligned\")\n",
    "\n",
    "# PanX gene cluster table\n",
    "Gene_clust = pd.read_csv(\"/ebio/abt3_projects/vadinCA11/data/V11/panx_output/tables/gene_clust_summary.txt\", \n",
    "                         sep='\\t', header=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-24T12:58:12.428192Z",
     "start_time": "2019-05-24T12:58:12.422213Z"
    }
   },
   "source": [
    "# Annotate gene clusters"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Consensus sequence of gene clusters\n",
    "\n",
    "For `eggNOG` and `Interpro` I will use the consensus sequence of each gene cluster. I will first create a fasta file with the consensus of all clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-27T12:35:48.468259Z",
     "start_time": "2019-05-27T12:35:48.426269Z"
    }
   },
   "outputs": [],
   "source": [
    "# Create list of name of resulting clusters from PanX table\n",
    "# and add name of aa alignment file\n",
    "cluster_names = Gene_clust[\"MSA\"].tolist()\n",
    "cluster_files = []\n",
    "for name in cluster_names:\n",
    "    file = os.path.join(panX_dir, name + \"_aa_aln_reduced.fa.gz\")\n",
    "    cluster_files.append(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-27T12:49:29.464265Z",
     "start_time": "2019-05-27T12:49:29.457917Z"
    }
   },
   "outputs": [],
   "source": [
    "# panX_output = os.listdir(panX_dir)\n",
    "# cluster_aligns = [fasta for fasta in panX_output if \"_aa_aln_reduced.fa.gz\" in fasta]\n",
    "cluster_files = cluster_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-05-27T12:50:36.497Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "# Create a single file with all consensus sequences\n",
    "# File path\n",
    "consensus_gapped_file = os.path.join(annotation_dir, \"consensus_gapped.fasta\")\n",
    "!touch $consensus_gapped_file\n",
    "for fasta in cluster_aligns:\n",
    "    fasta_path = os.path.join(panX_dir, fasta)\n",
    "    # Change header of sequence to cluster name\n",
    "    cluster_name = fasta.replace(\"_aa_aln_reduced.fa.gz\", \"\")\n",
    "    header = \">{cluster}\".format(cluster = cluster_name)\n",
    "    # Extract first sequence from cluster align file\n",
    "    zcat_cmd = 'zcat {file} | awk \"/^>/ {{n++}} n>1 {{exit}} {{print}}\"'.format(file = fasta_path)\n",
    "    # Replace header\n",
    "    sed_cmd = \"{zcat} | sed 's/^>.*$/{header}/' >> {out_file}\".format(zcat = zcat_cmd, \n",
    "                                                                  header = header, \n",
    "                                                                  out_file =  consensus_gapped_file)\n",
    "\n",
    "    !$sed_cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-24T13:23:09.852220Z",
     "start_time": "2019-05-24T13:23:09.845640Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate metacompass; cat /ebio/abt3_projects/vadinCA11/data/V11/cluster_annot/consensus_gapped.fasta | seqtk seq -A | tr -d -  > /ebio/abt3_projects/vadinCA11/data/V11/cluster_annot/consensus.fasta\"\n"
     ]
    }
   ],
   "source": [
    "# Ungap consensus sequences and format\n",
    "consensus_file = os.path.join(annotation_dir, \"consensus.fasta\")\n",
    "ungap_cmd = \"cat {in_fasta} | seqtk seq -A | tr -d -  > {out_fasta}\".format(in_fasta = consensus_gapped_file, \n",
    "                                                                           out_fasta = consensus_file)\n",
    "\n",
    "ungap_job = 'bash -c \"source activate {0}; {1}\"'.format(metacompass_env, ungap_cmd)\n",
    "print(ungap_job)\n",
    "#!$ungap_job"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-24T12:58:12.376142Z",
     "start_time": "2019-05-24T12:58:12.371120Z"
    }
   },
   "source": [
    "## Execute InterproScan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-24T13:23:16.304178Z",
     "start_time": "2019-05-24T13:23:16.297235Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate metacompass; /ebio/abt3_projects/vadinCA11/bin/interpro/interproscan_5/interproscan.sh -i /ebio/abt3_projects/vadinCA11/data/V11/cluster_annot/consensus.fasta     --output-dir /ebio/abt3_projects/vadinCA11/data/V11/cluster_annot     --cpu 20     --tempdir  /tmp/global2/jdelacuesta     --formats tsv\"\n"
     ]
    }
   ],
   "source": [
    "interpro_cmd = \"{interpro} -i {in_fasta} \\\n",
    "    --output-dir {out_dir} \\\n",
    "    --cpu {cpu} \\\n",
    "    --tempdir {tmp} \\\n",
    "    --formats tsv\"\n",
    "interpro_cmd = interpro_cmd.format(interpro = interpro_sh, \n",
    "                                   in_fasta = consensus_file, \n",
    "                                   out_dir = annotation_dir, \n",
    "                                   tmp =\" /tmp/global2/jdelacuesta\",\n",
    "                                   cpu = 20)\n",
    "\n",
    "interpro_job = 'bash -c \"source activate {0}; {1}\"'.format(\"metacompass\", interpro_cmd)\n",
    "print(interpro_job)\n",
    "#!$interpro_job"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Execute eggNOG mapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-27T14:06:43.512236Z",
     "start_time": "2019-05-27T14:06:43.503284Z"
    }
   },
   "outputs": [],
   "source": [
    "eggnog_cluster = \"\"\"#!/bin/bash\n",
    "#$ -N {name}\n",
    "#$ -pe parallel {cpu}\n",
    "#$ -l h_vmem=10G\n",
    "#$ -l h_rt=200:0:0\n",
    "#$ -o /ebio/abt3_projects/vadinCA11/data/V11/eggnog/cluster_jobs/logs\n",
    "#$ -j y\n",
    "#$ -wd {workdir}\n",
    "#$ -m ea\n",
    "#$ -M jdelacuesta@tuebingen.mpg.de\n",
    "\n",
    "export PATH='/ebio/abt3_projects/software/miniconda3_gt4.4/envs/eggNOG/bin':$PATH\n",
    "\n",
    "emapper.py -i {input_file} \\\n",
    "    --database {DB_type} \\\n",
    "    --data_dir {eggnog_database}  \\\n",
    "    --target_orthologs all \\\n",
    "    --output_dir {output_dir} \\\n",
    "    --output {base_name} \\\n",
    "    --scratch_dir {tmp_folder} \\\n",
    "    --temp_dir {tmp_folder} \\\n",
    "    --go_evidence non-electronic \\\n",
    "    --cpu {cpu} \\\n",
    "    --usemem \\\n",
    "    --override \\\n",
    "    -m {method}\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-27T14:07:30.072191Z",
     "start_time": "2019-05-27T14:07:30.061985Z"
    }
   },
   "outputs": [],
   "source": [
    "job_name = \"eggnog_clusters.sh\"\n",
    "\n",
    "eggnog_output = os.path.join(annotation_dir, \"eggNOG\")\n",
    "script_eggnog = os.path.join(eggnog_output, job_name)\n",
    "\n",
    "\n",
    "if not os.path.exists(eggnog_output):\n",
    "    os.makedirs(eggnog_output)\n",
    "\n",
    "prot_eggnog = eggnog_cluster.format(name = job_name, \n",
    "                                    cpu = 20, \n",
    "                                    input_file = consensus_file, \n",
    "                                    DB_type = \"arch\", \n",
    "                                    workdir = eggnog_output, \n",
    "                                    eggnog_database = eggnog_db, \n",
    "                                    output_dir = eggnog_output, \n",
    "                                    base_name = \"Clusters\", \n",
    "                                    tmp_folder = \"/tmp/global/jdelacuesta/\", \n",
    "                                    method = \"diamond\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-27T14:07:31.212406Z",
     "start_time": "2019-05-27T14:07:31.023945Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 4633871 (\"eggnog_clusters.sh\") has been submitted\r\n"
     ]
    }
   ],
   "source": [
    "fh = open(script_eggnog, \"w\")\n",
    "fh.write(prot_eggnog)\n",
    "fh.close()\n",
    "!qsub $script_eggnog"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotate *M. smithii* ALPs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Consensus sequence of *M. smithii* ALPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-24T13:59:58.708161Z",
     "start_time": "2019-05-24T13:59:58.703730Z"
    }
   },
   "outputs": [],
   "source": [
    "ALP_files = os.listdir(ALP_dir)\n",
    "ALP_seqs = [fasta for fasta in ALP_files if \"faa\" in fasta]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-24T14:02:48.804311Z",
     "start_time": "2019-05-24T14:02:47.022876Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: write error: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "# Create a single file with all consensus sequences\n",
    "# File path\n",
    "ALPs_file = os.path.join(annotation_dir, \"Msmithii_ALPs.fasta\")\n",
    "!touch $ALPs_file\n",
    "for fasta in ALP_seqs:\n",
    "    fasta_path = os.path.join(ALP_dir, fasta)\n",
    "    # Change header of sequence to cluster name\n",
    "    cluster_name = fasta.replace(\".faa\", \"\")\n",
    "    header = \">{cluster}\".format(cluster = cluster_name)\n",
    "    # Extract first sequence from cluster align file\n",
    "    cat_cmd = 'cat {file} | awk \"/^>/ {{n++}} n>1 {{exit}} {{print}}\"'.format(file = fasta_path)\n",
    "    # Replace header\n",
    "    sed_cmd = \"{cat} | sed 's/^>.*$/{header}/' >> {out_file}\".format(cat = cat_cmd, \n",
    "                                                                  header = header, \n",
    "                                                                  out_file =  ALPs_file)\n",
    "#     print(sed_cmd)\n",
    "    !$sed_cmd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-24T12:58:12.436162Z",
     "start_time": "2019-05-24T12:58:12.430300Z"
    }
   },
   "source": [
    "## Execute InterproScan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-05-24T14:03:26.176219Z",
     "start_time": "2019-05-24T14:03:26.167733Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash -c \"source activate metacompass; /ebio/abt3_projects/vadinCA11/bin/interpro/interproscan_5/interproscan.sh -i /ebio/abt3_projects/vadinCA11/data/V11/cluster_annot/Msmithii_ALPs.fasta     --output-dir /ebio/abt3_projects/vadinCA11/data/V11/cluster_annot     --cpu 20     --tempdir  /tmp/global2/jdelacuesta     --formats tsv\"\n"
     ]
    }
   ],
   "source": [
    "interpro_cmd = \"{interpro} -i {in_fasta} \\\n",
    "    --output-dir {out_dir} \\\n",
    "    --cpu {cpu} \\\n",
    "    --tempdir {tmp} \\\n",
    "    --formats tsv\"\n",
    "interpro_cmd = interpro_cmd.format(interpro = interpro_sh, \n",
    "                                   in_fasta = ALPs_file, \n",
    "                                   out_dir = annotation_dir, \n",
    "                                   tmp =\" /tmp/global2/jdelacuesta\",\n",
    "                                   cpu = 20)\n",
    "\n",
    "interpro_job = 'bash -c \"source activate {0}; {1}\"'.format(\"metacompass\", interpro_cmd)\n",
    "print(interpro_job)\n",
    "#!$interpro_job"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Session Info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-11T07:34:08.579052Z",
     "start_time": "2020-09-11T07:34:01.614852Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# packages in environment at /ebio/abt3_projects/software/miniconda3_gt4.4/envs/metacompass:\r\n",
      "#\r\n",
      "# Name                    Version                   Build  Channel\r\n",
      "_libgcc_mutex             0.1                        main  \r\n",
      "_r-mutex                  1.0.0               anacondar_1  \r\n",
      "aioeasywebdav             2.2.0                    py36_0    conda-forge\r\n",
      "aiohttp                   3.4.4            py36h470a237_0    conda-forge\r\n",
      "anvio                     5.1.0            py36hcb787e7_1    bioconda\r\n",
      "appdirs                   1.4.3                      py_1    conda-forge\r\n",
      "aragorn                   1.2.38               h470a237_2    bioconda\r\n",
      "asn1crypto                0.24.0                     py_1    conda-forge\r\n",
      "async-timeout             3.0.1                   py_1000    conda-forge\r\n",
      "attrs                     18.1.0                     py_1    conda-forge\r\n",
      "automat                   0.7.0                    py36_0    conda-forge\r\n",
      "backcall                  0.1.0                      py_0    conda-forge\r\n",
      "barrnap                   0.9                           2    bioconda\r\n",
      "bcftools                  1.9                  h4da6232_0    bioconda\r\n",
      "bcrypt                    3.1.4            py36h470a237_0    conda-forge\r\n",
      "bedtools                  2.27.1               he941832_2    bioconda\r\n",
      "biopython                 1.72                     py36_0    conda-forge\r\n",
      "blas                      1.1                    openblas    conda-forge\r\n",
      "blast                     2.7.1                h4422958_6    bioconda\r\n",
      "blast-legacy              2.2.26                        1    bioconda\r\n",
      "bleach                    2.1.3                      py_0    conda-forge\r\n",
      "boost                     1.67.0           py36h3e44d54_0    conda-forge\r\n",
      "boost-cpp                 1.67.0               h3a22d5f_0    conda-forge\r\n",
      "boto3                     1.7.84                     py_0    conda-forge\r\n",
      "botocore                  1.10.84                    py_0    conda-forge\r\n",
      "bottle                    0.12.13                    py_1    conda-forge\r\n",
      "bzip2                     1.0.6                h470a237_2    conda-forge\r\n",
      "ca-certificates           2019.6.16            hecc5488_0    conda-forge\r\n",
      "cachetools                2.1.0                      py_0    conda-forge\r\n",
      "cairo                     1.14.12              he6fea26_5    conda-forge\r\n",
      "capnproto                 0.6.1                hfc679d8_1    conda-forge\r\n",
      "cd-hit                    4.6.8                hfc679d8_2    bioconda\r\n",
      "centrifuge                1.0.4_beta      py36pl526he941832_2    bioconda\r\n",
      "certifi                   2019.6.16                py36_1    conda-forge\r\n",
      "cffi                      1.11.5           py36h5e8e0c9_1    conda-forge\r\n",
      "chardet                   3.0.4                 py36_1003    conda-forge\r\n",
      "checkm-genome             1.0.7                    py36_0    bioconda\r\n",
      "cherrypy                  8.0.0                    py36_0    conda-forge\r\n",
      "clustalw                  2.1                  h2d50403_2    bioconda\r\n",
      "configargparse            0.13.0                     py_1    conda-forge\r\n",
      "constantly                15.1.0                     py_0    conda-forge\r\n",
      "cryptography              2.3              py36hdffb7b8_0    conda-forge\r\n",
      "cryptography-vectors      2.3                      py36_1    conda-forge\r\n",
      "curl                      7.61.0               h93b3f91_1    conda-forge\r\n",
      "cycler                    0.10.0                     py_1    conda-forge\r\n",
      "datrie                    0.7.1                    py36_0    https://repo.anaconda.com/pkgs/free\r\n",
      "dbus                      1.13.0               h3a4f0e9_0    conda-forge\r\n",
      "decorator                 4.3.0                      py_0    conda-forge\r\n",
      "dendropy                  4.4.0                      py_1    bioconda\r\n",
      "diamond                   0.9.24               ha87ae23_0    bioconda\r\n",
      "django                    2.0.2                    py36_0    conda-forge\r\n",
      "docutils                  0.14                  py36_1001    conda-forge\r\n",
      "drep                      2.3.2                      py_0    bioconda\r\n",
      "dropbox                   7.3.1                    py36_0    conda-forge\r\n",
      "entrypoints               0.2.3                    py36_1    conda-forge\r\n",
      "ete3                      3.1.1                    py36_0    bioconda\r\n",
      "expat                     2.2.5                hfc679d8_2    conda-forge\r\n",
      "fasttree                  2.1.10               h470a237_2    bioconda\r\n",
      "filechunkio               1.6                      py36_0    bioconda\r\n",
      "fontconfig                2.13.0               h9420a91_0  \r\n",
      "freetype                  2.9.1                h6debe1e_4    conda-forge\r\n",
      "ftputil                   3.2                      py36_0    bioconda\r\n",
      "gawk                      4.2.1                h470a237_0    conda-forge\r\n",
      "gettext                   0.19.8.1                      0    conda-forge\r\n",
      "giflib                    5.1.4                h470a237_1    conda-forge\r\n",
      "glib                      2.55.0               h464dc38_2    conda-forge\r\n",
      "gmp                       6.1.2                hfc679d8_0    conda-forge\r\n",
      "gnutls                    3.5.19               h2a4e5f8_1    conda-forge\r\n",
      "google-auth               1.6.1                      py_0    conda-forge\r\n",
      "google-auth-httplib2      0.0.3                      py_2    conda-forge\r\n",
      "google-cloud-core         0.24.1                   py36_0    conda-forge\r\n",
      "google-cloud-storage      1.1.1                    py36_0    conda-forge\r\n",
      "google-resumable-media    0.0.2                    py36_0    conda-forge\r\n",
      "googleapis-common-protos  1.5.5                      py_0    conda-forge\r\n",
      "graphite2                 1.3.11               hfc679d8_0    conda-forge\r\n",
      "graphviz                  2.38.0               h08bfae6_9    conda-forge\r\n",
      "gsl                       2.2.1           blas_openblas_4  [blas_openblas]  conda-forge\r\n",
      "gst-plugins-base          1.12.5               hde13a9d_0    conda-forge\r\n",
      "gstreamer                 1.12.5               h61a6719_0    conda-forge\r\n",
      "h5py                      2.8.0            py36he5c79e1_5    conda-forge\r\n",
      "harfbuzz                  1.9.0                h08d66d9_0    conda-forge\r\n",
      "hdf5                      1.10.4          nompi_h5598ddc_1003    conda-forge\r\n",
      "hmmer                     3.2.1                hfc679d8_0    bioconda\r\n",
      "html5lib                  1.0.1                      py_0    conda-forge\r\n",
      "htslib                    1.7                           0    bioconda\r\n",
      "httplib2                  0.12.0                py36_1000    conda-forge\r\n",
      "hyperlink                 17.3.1                     py_0    conda-forge\r\n",
      "icu                       58.2                 hfc679d8_0    conda-forge\r\n",
      "idna                      2.7                      py36_2    conda-forge\r\n",
      "idna_ssl                  1.0.0                         0    conda-forge\r\n",
      "incremental               17.5.0                     py_0    conda-forge\r\n",
      "infernal                  1.1.2                h470a237_1    bioconda\r\n",
      "ipykernel                 4.8.2                    py36_0    conda-forge\r\n",
      "ipython                   6.5.0                    py36_0    conda-forge\r\n",
      "ipython_genutils          0.2.0                      py_1    conda-forge\r\n",
      "java-jdk                  8.0.92                        1    bioconda\r\n",
      "jedi                      0.12.1                   py36_0    conda-forge\r\n",
      "jellyfish                 1.1.12               h2d50403_0    bioconda\r\n",
      "jinja2                    2.10                       py_1    conda-forge\r\n",
      "jmespath                  0.9.3                      py_1    conda-forge\r\n",
      "jpeg                      9c                   h470a237_0    conda-forge\r\n",
      "jsonschema                2.6.0                    py36_1    conda-forge\r\n",
      "jupyter_client            5.2.3                      py_1    conda-forge\r\n",
      "jupyter_core              4.4.0                      py_0    conda-forge\r\n",
      "jupyterlab                0.35.4                   py36_0    conda-forge\r\n",
      "jupyterlab_server         0.2.0                      py_0    conda-forge\r\n",
      "kiwisolver                1.0.1            py36h2d50403_2    conda-forge\r\n",
      "kraken2                   2.0.7_beta      pl526h2d50403_0    bioconda\r\n",
      "krb5                      1.14.6                        0    conda-forge\r\n",
      "libdb                     6.1.26            hfc679d8_1000    conda-forge\r\n",
      "libdeflate                1.0                  h470a237_0    bioconda\r\n",
      "libffi                    3.2.1                         3    conda-forge\r\n",
      "libgcc                    7.2.0                h69d50b8_2  \r\n",
      "libgcc-ng                 7.2.0                hdf63c60_3  \r\n",
      "libgd                     2.2.5                h5400f36_4    conda-forge\r\n",
      "libgfortran               3.0.0                         1    https://repo.anaconda.com/pkgs/free\r\n",
      "libgfortran-ng            7.2.0                hdf63c60_3  \r\n",
      "libiconv                  1.15                 h470a237_1    conda-forge\r\n",
      "libidn11                  1.33                          0    conda-forge\r\n",
      "libopenblas               0.2.20               h9ac9557_7  \r\n",
      "libpng                    1.6.35               ha92aebf_0    conda-forge\r\n",
      "libprotobuf               3.6.1                hd28b015_0    conda-forge\r\n",
      "libsodium                 1.0.16                        0    conda-forge\r\n",
      "libssh2                   1.8.0                h5b517e9_2    conda-forge\r\n",
      "libstdcxx-ng              7.2.0                hdf63c60_3  \r\n",
      "libtiff                   4.0.9                he6b73bb_1    conda-forge\r\n",
      "libtool                   2.4.6                h470a237_2    conda-forge\r\n",
      "libuuid                   1.0.3                         1    conda-forge\r\n",
      "libwebp                   0.5.2                         7    conda-forge\r\n",
      "libxcb                    1.13                 h470a237_2    conda-forge\r\n",
      "libxml2                   2.9.8                h422b904_2    conda-forge\r\n",
      "libxslt                   1.1.32               h88dbc4e_2    conda-forge\r\n",
      "llvm-meta                 7.0.0                         0    conda-forge\r\n",
      "lxml                      4.2.5            py36hc9114bc_0    conda-forge\r\n",
      "mafft                     7.310                         1    bioconda\r\n",
      "markupsafe                1.0                      py36_0    conda-forge\r\n",
      "mash                      2.1                  h9dd4a16_0    bioconda\r\n",
      "matplotlib                2.2.2            py36hb69df0a_2  \r\n",
      "mcl                       14.137          pl526h470a237_4    bioconda\r\n",
      "minced                    0.2.0                         0    bioconda\r\n",
      "mistune                   0.7.4                    py36_0    conda-forge\r\n",
      "multidict                 4.5.1            py36h470a237_0    conda-forge\r\n",
      "mummer                    3.23                    pl526_6    bioconda\r\n",
      "muscle                    3.8.1551             h2d50403_3    bioconda\r\n",
      "nb_conda                  2.2.1                    py36_0    conda-forge\r\n",
      "nb_conda_kernels          2.1.1                    py36_1    conda-forge\r\n",
      "nbconvert                 5.3.1                      py_1    conda-forge\r\n",
      "nbformat                  4.4.0                      py_1    conda-forge\r\n",
      "ncbi-genome-download      0.2.9                    py36_0    bioconda\r\n",
      "ncurses                   6.1                  hfc679d8_1    conda-forge\r\n",
      "nettle                    3.3                           0    conda-forge\r\n",
      "notebook                  5.6.0                    py36_0    conda-forge\r\n",
      "numpy                     1.15.2          py36_blas_openblashd3ea46f_1  [blas_openblas]  conda-forge\r\n",
      "openblas                  0.2.20                        8    conda-forge\r\n",
      "openmp                    7.0.0                h2d50403_0    conda-forge\r\n",
      "openssl                   1.0.2p               h470a237_2    conda-forge\r\n",
      "paml                      4.9                  h470a237_3    bioconda\r\n",
      "pandas                    0.23.4           py36h04863e7_0    anaconda\r\n",
      "pandoc                    2.2.2                         1    conda-forge\r\n",
      "pandocfilters             1.4.2                      py_1    conda-forge\r\n",
      "pango                     1.40.14              he752989_2    conda-forge\r\n",
      "parallel                  20190322                      0    conda-forge\r\n",
      "parallel-fastq-dump       0.6.3                    py36_1    bioconda\r\n",
      "paramiko                  2.4.2                 py36_1000    conda-forge\r\n",
      "parso                     0.3.1                      py_0    conda-forge\r\n",
      "patsy                     0.5.1                      py_0    conda-forge\r\n",
      "pcre                      8.41                 h470a237_2    conda-forge\r\n",
      "perl                      5.26.2               h470a237_0    conda-forge\r\n",
      "perl-aceperl              1.92                    pl526_2    bioconda\r\n",
      "perl-algorithm-diff       1.1903                  pl526_2    bioconda\r\n",
      "perl-algorithm-munkres    0.08                    pl526_1    bioconda\r\n",
      "perl-apache-test          1.40                    pl526_1    bioconda\r\n",
      "perl-app-cpanminus        1.7044                  pl526_1    bioconda\r\n",
      "perl-appconfig            1.71                    pl526_1    bioconda\r\n",
      "perl-archive-tar          2.18                    pl526_3    bioconda\r\n",
      "perl-array-compare        3.0.1                   pl526_1    bioconda\r\n",
      "perl-autoloader           5.74                    pl526_2    bioconda\r\n",
      "perl-base                 2.23                    pl526_1    bioconda\r\n",
      "perl-bio-asn1-entrezgene  1.73                    pl526_0    bioconda\r\n",
      "perl-bio-coordinate       1.007001                pl526_0    bioconda\r\n",
      "perl-bio-featureio        1.6.905                 pl526_1    bioconda\r\n",
      "perl-bio-phylo            0.58                    pl526_1    bioconda\r\n",
      "perl-bio-samtools         1.43            pl526h1341992_1    bioconda\r\n",
      "perl-bio-tools-phylo-paml 1.7.3                   pl526_0    bioconda\r\n",
      "perl-bio-tools-run-alignment-clustalw 1.7.4                   pl526_0    bioconda\r\n",
      "perl-bio-tools-run-alignment-tcoffee 1.7.4                   pl526_1    bioconda\r\n",
      "perl-bioperl              1.7.2                   pl526_9    bioconda\r\n",
      "perl-bioperl-core         1.7.2                   pl526_3    bioconda\r\n",
      "perl-bioperl-run          1.007002                pl526_3    bioconda\r\n",
      "perl-business-isbn        3.004                   pl526_0    bioconda\r\n",
      "perl-business-isbn-data   20140910.003            pl526_0    bioconda\r\n",
      "perl-cache-cache          1.08                    pl526_0    bioconda\r\n",
      "perl-capture-tiny         0.48                    pl526_0    bioconda\r\n",
      "perl-carp                 1.38                    pl526_1    bioconda\r\n",
      "perl-cgi                  4.40            pl526h470a237_0    bioconda\r\n",
      "perl-class-data-inheritable 0.08                    pl526_1    bioconda\r\n",
      "perl-class-inspector      1.34                    pl526_0    bioconda\r\n",
      "perl-class-load           0.25                    pl526_0    bioconda\r\n",
      "perl-class-load-xs        0.10            pl526h2d50403_0    bioconda\r\n",
      "perl-class-method-modifiers 2.12                    pl526_0    bioconda\r\n",
      "perl-clone                0.41            pl526h470a237_0    bioconda\r\n",
      "perl-common-sense         3.74                    pl526_2    bioconda\r\n",
      "perl-compress-raw-bzip2   2.074           pl526hfc679d8_0    bioconda\r\n",
      "perl-compress-raw-zlib    2.081           pl526h2d50403_0    bioconda\r\n",
      "perl-constant             1.33                    pl526_1    bioconda\r\n",
      "perl-convert-binary-c     0.78            pl526h2d50403_2    bioconda\r\n",
      "perl-convert-binhex       1.125                   pl526_1    bioconda\r\n",
      "perl-crypt-rc4            2.02                    pl526_1    bioconda\r\n",
      "perl-data-dumper          2.161                   pl526_2    bioconda\r\n",
      "perl-data-optlist         0.110                   pl526_2    bioconda\r\n",
      "perl-data-stag            0.14                    pl526_1    bioconda\r\n",
      "perl-date-format          2.30                    pl526_2    bioconda\r\n",
      "perl-db-file              1.843           pl526h470a237_0    bioconda\r\n",
      "perl-dbd-sqlite           1.60            pl526h470a237_0    bioconda\r\n",
      "perl-dbi                  1.642                   pl526_0    bioconda\r\n",
      "perl-devel-globaldestruction 0.14                    pl526_0    bioconda\r\n",
      "perl-devel-overloadinfo   0.005                   pl526_0    bioconda\r\n",
      "perl-devel-stacktrace     2.03                    pl526_1    bioconda\r\n",
      "perl-digest-hmac          1.03                    pl526_3    bioconda\r\n",
      "perl-digest-md5           2.55                    pl526_0    bioconda\r\n",
      "perl-digest-perl-md5      1.9                     pl526_1    bioconda\r\n",
      "perl-digest-sha1          2.13            pl526h2d50403_0    bioconda\r\n",
      "perl-dist-checkconflicts  0.11                    pl526_2    bioconda\r\n",
      "perl-dynaloader           1.25                    pl526_1    bioconda\r\n",
      "perl-email-date-format    1.005                   pl526_2    bioconda\r\n",
      "perl-encode               2.88                    pl526_1    bioconda\r\n",
      "perl-encode-locale        1.05                    pl526_6    bioconda\r\n",
      "perl-error                0.17027                 pl526_1    bioconda\r\n",
      "perl-eval-closure         0.14            pl526h2d50403_3    bioconda\r\n",
      "perl-exception-class      1.44                    pl526_0    bioconda\r\n",
      "perl-exporter             5.72                    pl526_1    bioconda\r\n",
      "perl-exporter-tiny        1.000000                pl526_0    bioconda\r\n",
      "perl-extutils-makemaker   7.34                    pl526_2    bioconda\r\n",
      "perl-file-listing         6.04                    pl526_1    bioconda\r\n",
      "perl-file-path            2.15                    pl526_0    bioconda\r\n",
      "perl-file-slurp-tiny      0.004                   pl526_1    bioconda\r\n",
      "perl-file-sort            1.01                    pl526_2    bioconda\r\n",
      "perl-file-temp            0.2304                  pl526_2    bioconda\r\n",
      "perl-file-which           1.23                    pl526_0    bioconda\r\n",
      "perl-font-afm             1.20                    pl526_2    bioconda\r\n",
      "perl-font-ttf             1.06                    pl526_0    bioconda\r\n",
      "perl-gd                   2.70            pl526he941832_0    bioconda\r\n",
      "perl-getopt-long          2.50                    pl526_1    bioconda\r\n",
      "perl-graph                0.9704                  pl526_1    bioconda\r\n",
      "perl-graphviz             2.24            pl526h734ff71_0    bioconda\r\n",
      "perl-html-element-extended 1.18                    pl526_1    bioconda\r\n",
      "perl-html-entities-numbered 0.04                    pl526_1    bioconda\r\n",
      "perl-html-formatter       2.16                    pl526_0    bioconda\r\n",
      "perl-html-parser          3.72            pl526h2d50403_4    bioconda\r\n",
      "perl-html-tableextract    2.13                    pl526_2    bioconda\r\n",
      "perl-html-tagset          3.20                    pl526_3    bioconda\r\n",
      "perl-html-tidy            1.60                    pl526_0    bioconda\r\n",
      "perl-html-tree            5.07                    pl526_1    bioconda\r\n",
      "perl-html-treebuilder-xpath 0.14                    pl526_1    bioconda\r\n",
      "perl-http-cookies         6.04                    pl526_0    bioconda\r\n",
      "perl-http-daemon          6.01                    pl526_1    bioconda\r\n",
      "perl-http-date            6.02                    pl526_3    bioconda\r\n",
      "perl-http-message         6.18                    pl526_0    bioconda\r\n",
      "perl-http-negotiate       6.01                    pl526_3    bioconda\r\n",
      "perl-image-info           1.38                    pl526_1    bioconda\r\n",
      "perl-image-size           3.300                   pl526_2    bioconda\r\n",
      "perl-importer             0.025                   pl526_1    bioconda\r\n",
      "perl-io-compress          2.069           pl526hfc679d8_5    bioconda\r\n",
      "perl-io-html              1.001                   pl526_2    bioconda\r\n",
      "perl-io-sessiondata       1.03                    pl526_1    bioconda\r\n",
      "perl-io-socket-ssl        2.066                   pl526_0    bioconda\r\n",
      "perl-io-string            1.08                    pl526_3    bioconda\r\n",
      "perl-io-stringy           2.111                   pl526_1    bioconda\r\n",
      "perl-io-tty               1.12                    pl526_1    bioconda\r\n",
      "perl-io-zlib              1.10                    pl526_2    bioconda\r\n",
      "perl-ipc-run              20180523.0              pl526_0    bioconda\r\n",
      "perl-ipc-sharelite        0.17            pl526h2d50403_0    bioconda\r\n",
      "perl-jcode                2.07                    pl526_2    bioconda\r\n",
      "perl-json                 4.00                    pl526_0    bioconda\r\n",
      "perl-json-xs              2.34            pl526h2d50403_2    bioconda\r\n",
      "perl-lib                  0.63                    pl526_1    bioconda\r\n",
      "perl-libwww-perl          6.36                    pl526_1    bioconda\r\n",
      "perl-libxml-perl          0.08                    pl526_2    bioconda\r\n",
      "perl-list-moreutils       0.428                   pl526_1    bioconda\r\n",
      "perl-list-moreutils-xs    0.428                   pl526_0    bioconda\r\n",
      "perl-lwp-mediatypes       6.04                    pl526_0    bioconda\r\n",
      "perl-lwp-protocol-https   6.07                    pl526_4    bioconda\r\n",
      "perl-lwp-simple           6.15            pl526h470a237_4    bioconda\r\n",
      "perl-mailtools            2.20                    pl526_0    bioconda\r\n",
      "perl-math-cdf             0.1             pl526h470a237_4    bioconda\r\n",
      "perl-math-derivative      1.01                    pl526_0    bioconda\r\n",
      "perl-math-random          0.72                    pl526_1    bioconda\r\n",
      "perl-math-spline          0.02                    pl526_2    bioconda\r\n",
      "perl-mime-base64          3.15                    pl526_1    bioconda\r\n",
      "perl-mime-lite            3.030                   pl526_1    bioconda\r\n",
      "perl-mime-tools           5.508                   pl526_1    bioconda\r\n",
      "perl-mime-types           2.17                    pl526_0    bioconda\r\n",
      "perl-mldbm                2.05                    pl526_1    bioconda\r\n",
      "perl-module-implementation 0.09                    pl526_2    bioconda\r\n",
      "perl-module-pluggable     5.2                     pl526_1    bioconda\r\n",
      "perl-module-runtime       0.016                   pl526_1    bioconda\r\n",
      "perl-module-runtime-conflicts 0.003                   pl526_0    bioconda\r\n",
      "perl-moo                  2.003004                pl526_0    bioconda\r\n",
      "perl-moose                2.2011          pl526h2d50403_0    bioconda\r\n",
      "perl-mozilla-ca           20180117                pl526_1    bioconda\r\n",
      "perl-mro-compat           0.13                    pl526_0    bioconda\r\n",
      "perl-net-http             6.18                    pl526_0    bioconda\r\n",
      "perl-net-ssleay           1.85            pl526h9dad42d_0    bioconda\r\n",
      "perl-ntlm                 1.09                    pl526_4    bioconda\r\n",
      "perl-ole-storage_lite     0.19                    pl526_3    bioconda\r\n",
      "perl-package-deprecationmanager 0.17                    pl526_0    bioconda\r\n",
      "perl-package-stash        0.38            pl526hfc679d8_0    bioconda\r\n",
      "perl-package-stash-xs     0.28            pl526hfc679d8_0    bioconda\r\n",
      "perl-params-util          1.07            pl526h2d50403_3    bioconda\r\n",
      "perl-parent               0.236                   pl526_1    bioconda\r\n",
      "perl-parse-recdescent     1.967015                pl526_0    bioconda\r\n",
      "perl-pathtools            3.73                 h470a237_2    bioconda\r\n",
      "perl-pdf-api2             2.033                   pl526_0    bioconda\r\n",
      "perl-pod-escapes          1.07                    pl526_1    bioconda\r\n",
      "perl-pod-usage            1.69                    pl526_1    bioconda\r\n",
      "perl-postscript           0.06                    pl526_2    bioconda\r\n",
      "perl-role-tiny            2.000006                pl526_0    bioconda\r\n",
      "perl-scalar-list-utils    1.45            pl526h470a237_3    bioconda\r\n",
      "perl-scope-guard          0.21                    pl526_2    bioconda\r\n",
      "perl-set-scalar           1.29                    pl526_2    bioconda\r\n",
      "perl-soap-lite            1.19                    pl526_1    bioconda\r\n",
      "perl-socket               2.027                   pl526_1    bioconda\r\n",
      "perl-sort-naturally       1.03                    pl526_2    bioconda\r\n",
      "perl-spreadsheet-parseexcel 0.65                    pl526_2    bioconda\r\n",
      "perl-spreadsheet-writeexcel 2.40                    pl526_2    bioconda\r\n",
      "perl-statistics-descriptive 3.0702                  pl526_0    bioconda\r\n",
      "perl-storable             3.11                    pl526_0    bioconda\r\n",
      "perl-sub-exporter         0.987                   pl526_2    bioconda\r\n",
      "perl-sub-exporter-progressive 0.001013                pl526_0    bioconda\r\n",
      "perl-sub-identify         0.12                    pl526_1    bioconda\r\n",
      "perl-sub-info             0.002                   pl526_0    bioconda\r\n",
      "perl-sub-install          0.928                   pl526_2    bioconda\r\n",
      "perl-sub-name             0.21                    pl526_1    bioconda\r\n",
      "perl-sub-quote            2.006003                pl526_1    bioconda\r\n",
      "perl-sub-uplevel          0.2800          pl526h470a237_0    bioconda\r\n",
      "perl-svg                  2.84                    pl526_0    bioconda\r\n",
      "perl-svg-graph            0.02                    pl526_3    bioconda\r\n",
      "perl-task-weaken          1.06                    pl526_0    bioconda\r\n",
      "perl-template-toolkit     2.26                    pl526_1    bioconda\r\n",
      "perl-term-table           0.013                   pl526_0    bioconda\r\n",
      "perl-test                 1.26                    pl526_1    bioconda\r\n",
      "perl-test-deep            1.128                   pl526_1    bioconda\r\n",
      "perl-test-differences     0.64                    pl526_2    bioconda\r\n",
      "perl-test-exception       0.43                    pl526_2    bioconda\r\n",
      "perl-test-harness         3.42                    pl526_0    bioconda\r\n",
      "perl-test-leaktrace       0.16                    pl526_1    bioconda\r\n",
      "perl-test-more            1.001002                pl526_1    bioconda\r\n",
      "perl-test-most            0.35                    pl526_0    bioconda\r\n",
      "perl-test-requiresinternet 0.05                    pl526_0    bioconda\r\n",
      "perl-test-warn            0.36                    pl526_1    bioconda\r\n",
      "perl-test2-suite          0.000117                pl526_0    bioconda\r\n",
      "perl-text-diff            1.45                    pl526_0    bioconda\r\n",
      "perl-threaded             5.22.0                       13    bioconda\r\n",
      "perl-tie-ixhash           1.23                    pl526_2    bioconda\r\n",
      "perl-time-hires           1.9758                  pl526_0    bioconda\r\n",
      "perl-time-local           1.28                    pl526_1    bioconda\r\n",
      "perl-timedate             2.30                    pl526_1    bioconda\r\n",
      "perl-tree-dag_node        1.31                    pl526_0    bioconda\r\n",
      "perl-try-tiny             0.30                    pl526_1    bioconda\r\n",
      "perl-type-tiny            1.004004                pl526_0    bioconda\r\n",
      "perl-types-serialiser     1.0                     pl526_2    bioconda\r\n",
      "perl-unicode-map          0.112           pl526h2d50403_2    bioconda\r\n",
      "perl-uri                  1.76                    pl526_0    bioconda\r\n",
      "perl-www-robotrules       6.02                    pl526_3    bioconda\r\n",
      "perl-xml-dom              1.46                    pl526_0    bioconda\r\n",
      "perl-xml-dom-xpath        0.14                    pl526_1    bioconda\r\n",
      "perl-xml-filter-buffertext 1.01                    pl526_2    bioconda\r\n",
      "perl-xml-libxml           2.0132          pl526hbc14f71_0    bioconda\r\n",
      "perl-xml-libxslt          1.94                    pl526_1    bioconda\r\n",
      "perl-xml-namespacesupport 1.12                    pl526_0    bioconda\r\n",
      "perl-xml-parser           2.44            pl526h3a4f0e9_6    bioconda\r\n",
      "perl-xml-regexp           0.04                    pl526_2    bioconda\r\n",
      "perl-xml-sax              1.00                    pl526_0    bioconda\r\n",
      "perl-xml-sax-base         1.09                    pl526_0    bioconda\r\n",
      "perl-xml-sax-expat        0.51                    pl526_2    bioconda\r\n",
      "perl-xml-sax-writer       0.57                    pl526_0    bioconda\r\n",
      "perl-xml-simple           2.22                    pl526_3    bioconda\r\n",
      "perl-xml-twig             3.52                    pl526_2    bioconda\r\n",
      "perl-xml-writer           0.625                   pl526_2    bioconda\r\n",
      "perl-xml-xpath            1.44                    pl526_0    bioconda\r\n",
      "perl-xml-xpathengine      0.14                    pl526_2    bioconda\r\n",
      "perl-xsloader             0.24                    pl526_0    bioconda\r\n",
      "perl-yaml                 1.26                    pl526_1    bioconda\r\n",
      "pexpect                   4.6.0                    py36_0    conda-forge\r\n",
      "pfam_scan                 1.6                     pl526_2    bioconda\r\n",
      "pickleshare               0.7.4                    py36_0    conda-forge\r\n",
      "pigz                      2.4                  h84994c4_0    anaconda\r\n",
      "pip                       18.0                     py36_1    conda-forge\r\n",
      "pirate                    prerelease                    1    sionbayliss\r\n",
      "pixman                    0.34.0                        2    conda-forge\r\n",
      "plass                     2.c7e35              h21aa3a5_1    bioconda\r\n",
      "prettytable               0.7.2                      py_2    conda-forge\r\n",
      "prodigal                  2.6.3                         1    bioconda\r\n",
      "prokka                    1.13                          3    bioconda\r\n",
      "prometheus_client         0.3.0                      py_0    conda-forge\r\n",
      "prompt_toolkit            1.0.15                   py36_0    conda-forge\r\n",
      "protobuf                  3.6.1            py36hfc679d8_1    conda-forge\r\n",
      "psutil                    5.4.8            py36h470a237_0    conda-forge\r\n",
      "pthread-stubs             0.4                  h470a237_1    conda-forge\r\n",
      "ptyprocess                0.6.0                    py36_0    conda-forge\r\n",
      "pyani                     0.2.7            py36h24bf2e0_1    bioconda\r\n",
      "pyasn1                    0.4.4                      py_0    conda-forge\r\n",
      "pyasn1-modules            0.2.1                      py_0    conda-forge\r\n",
      "pycparser                 2.18                       py_1    conda-forge\r\n",
      "pygments                  2.2.0                      py_1    conda-forge\r\n",
      "pyhamcrest                1.9.0                      py_2    conda-forge\r\n",
      "pynacl                    1.3.0            py36h470a237_0    conda-forge\r\n",
      "pyopenssl                 18.0.0                   py36_0    conda-forge\r\n",
      "pyparsing                 2.2.2                      py_0    conda-forge\r\n",
      "pyqt                      5.6.0            py36h8210e8a_7    conda-forge\r\n",
      "pysam                     0.14.1           py36hae42fb6_1    bioconda\r\n",
      "pysftp                    0.2.9                    py36_0    bioconda\r\n",
      "python                    3.6.6                h5001a0f_0    conda-forge\r\n",
      "python-dateutil           2.7.3                      py_0    conda-forge\r\n",
      "python-irodsclient        0.7.0                      py_0    conda-forge\r\n",
      "pytz                      2018.5                     py_0    conda-forge\r\n",
      "pyyaml                    3.13             py36h470a237_1    conda-forge\r\n",
      "pyzmq                     17.1.0           py36hae99301_0    conda-forge\r\n",
      "qt                        5.6.2                hf70d934_9    conda-forge\r\n",
      "r-assertthat              0.2.0                  r3.4.1_0    conda-forge\r\n",
      "r-base                    3.4.1                h4fe35fd_8    conda-forge\r\n",
      "r-base64enc               0.1_3            r341h470a237_1    conda-forge\r\n",
      "r-bh                      1.66.0_1               r3.4.1_0    conda-forge\r\n",
      "r-bindr                   0.1.1                  r3.4.1_0    conda-forge\r\n",
      "r-bindrcpp                0.2.2            r341h9d2a408_0    conda-forge\r\n",
      "r-bit                     1.1_12           r341h470a237_1    conda-forge\r\n",
      "r-bit64                   0.9_5            r341h470a237_1    conda-forge\r\n",
      "r-blob                    1.1.1                  r3.4.1_0    conda-forge\r\n",
      "r-boot                    1.3_19                 r3.4.1_0    r\r\n",
      "r-cli                     1.0.0                  r3.4.1_0    conda-forge\r\n",
      "r-colorspace              1.3_2                  r3.4.1_0    r\r\n",
      "r-crayon                  1.3.4                  r3.4.1_0    conda-forge\r\n",
      "r-dbi                     1.0.0                    r341_0    conda-forge\r\n",
      "r-dbplyr                  1.2.2                    r341_0    conda-forge\r\n",
      "r-dichromat               2.0_0                  r3.4.1_2    r\r\n",
      "r-digest                  0.6.15           r341h470a237_1    conda-forge\r\n",
      "r-dplyr                   0.7.6            r341h9d2a408_0    conda-forge\r\n",
      "r-evaluate                0.10.1                 r3.4.1_0    conda-forge\r\n",
      "r-ggplot2                 2.2.1                  r3.4.1_0    r\r\n",
      "r-glue                    1.3.0            r341h470a237_1    conda-forge\r\n",
      "r-gtable                  0.2.0                  r3.4.1_0    r\r\n",
      "r-htmltools               0.3.6            r341hfc679d8_1    conda-forge\r\n",
      "r-irdisplay               0.4.4                  r3.4.1_0    conda-forge\r\n",
      "r-irkernel                0.8.12                   r341_0    conda-forge\r\n",
      "r-jsonlite                1.5              r341h470a237_1    conda-forge\r\n",
      "r-labeling                0.3                    r3.4.1_2    r\r\n",
      "r-lazyeval                0.2.0                  r3.4.1_0    r\r\n",
      "r-magrittr                1.5                    r3.4.1_0    conda-forge\r\n",
      "r-mass                    7.3_47                 r3.4.1_0    r\r\n",
      "r-memoise                 1.1.0                  r3.4.1_0    conda-forge\r\n",
      "r-munsell                 0.4.3                  r3.4.1_0    r\r\n",
      "r-network                 1.13.0.1         r341hc070d10_0    conda-forge\r\n",
      "r-pbdzmq                  0.3_2                  r3.4.1_0    conda-forge\r\n",
      "r-pillar                  1.2.2            r341h6115d3f_1    conda-forge\r\n",
      "r-pkgconfig               2.0.1                  r3.4.1_0    conda-forge\r\n",
      "r-plogr                   0.2.0            r341h6115d3f_0    conda-forge\r\n",
      "r-plyr                    1.8.4                  r3.4.1_0    r\r\n",
      "r-prettyunits             1.0.2                  r3.4.1_0    conda-forge\r\n",
      "r-purrr                   0.2.4            r341h470a237_1    conda-forge\r\n",
      "r-r6                      2.2.2                  r3.4.1_0    conda-forge\r\n",
      "r-rcolorbrewer            1.1_2                  r3.4.1_3    r\r\n",
      "r-rcpp                    0.12.17          r341h9d2a408_1    conda-forge\r\n",
      "r-repr                    0.15                     r341_0    conda-forge\r\n",
      "r-reshape2                1.4.2                  r3.4.1_0    r\r\n",
      "r-rlang                   0.2.1            r341h470a237_1    conda-forge\r\n",
      "r-rsqlite                 2.0              r341hfc679d8_1    conda-forge\r\n",
      "r-scales                  0.4.1                  r3.4.1_0    r\r\n",
      "r-stringi                 1.2.3            r341hfc679d8_1    conda-forge\r\n",
      "r-stringr                 1.3.1                    r341_0    conda-forge\r\n",
      "r-tibble                  1.4.2            r341h470a237_1    conda-forge\r\n",
      "r-tidyselect              0.2.4            r341hfc679d8_1    conda-forge\r\n",
      "r-utf8                    1.1.3            r341h470a237_1    conda-forge\r\n",
      "r-uuid                    0.1_2                    r341_0    conda-forge\r\n",
      "ratelimiter               1.2.0                 py36_1000    conda-forge\r\n",
      "readline                  7.0                  haf1bffa_1    conda-forge\r\n",
      "requests                  2.13.0                   py36_0    conda-forge\r\n",
      "rpy2                      2.9.4           py36r341h24bf2e0_0    conda-forge\r\n",
      "rsa                       3.1.4                    py36_0    bioconda\r\n",
      "s3transfer                0.1.13                py36_1001    conda-forge\r\n",
      "samtools                  1.9                  h8ee4bcc_1    bioconda\r\n",
      "scikit-learn              0.19.1          py36_blas_openblas_201  [blas_openblas]  conda-forge\r\n",
      "scipy                     1.1.0           py36_blas_openblash7943236_201  [blas_openblas]  conda-forge\r\n",
      "screamingbackpack         0.2.333                    py_1    bioconda\r\n",
      "seaborn                   0.9.0                      py_0    conda-forge\r\n",
      "send2trash                1.5.0                      py_0    conda-forge\r\n",
      "seqtk                     1.3                  ha92aebf_0    bioconda\r\n",
      "service_identity          17.0.0                     py_0    conda-forge\r\n",
      "setuptools                40.0.0                   py36_1    conda-forge\r\n",
      "simplegeneric             0.8.1                      py_1    conda-forge\r\n",
      "sip                       4.18.1           py36hfc679d8_0    conda-forge\r\n",
      "six                       1.11.0                   py36_1    conda-forge\r\n",
      "snakemake                 4.7.0                    py36_0    bioconda\r\n",
      "sqlite                    3.24.0               h2f33b56_0    conda-forge\r\n",
      "sra-tools                 2.9.1_1              h470a237_0    bioconda\r\n",
      "statsmodels               0.9.0            py36h7eb728f_0    conda-forge\r\n",
      "t_coffee                  11.0.8           py36h075ee95_7    bioconda\r\n",
      "taxonkit                  0.2.4                         1    bioconda\r\n",
      "tbl2asn                   25.6                          3    bioconda\r\n",
      "terminado                 0.8.1                    py36_0    conda-forge\r\n",
      "testpath                  0.3.1                    py36_0    conda-forge\r\n",
      "tidyp                     1.04                          1    bioconda\r\n",
      "tk                        8.6.8                         0    conda-forge\r\n",
      "tornado                   5.1              py36h470a237_1    conda-forge\r\n",
      "traitlets                 4.3.2                    py36_0    conda-forge\r\n",
      "twisted                   18.7.0           py36h14c3975_1  \r\n",
      "tzlocal                   1.5.1                      py_0    conda-forge\r\n",
      "urllib3                   1.12                     py36_0    bioconda\r\n",
      "wcwidth                   0.1.7                      py_1    conda-forge\r\n",
      "webencodings              0.5                      py36_0    conda-forge\r\n",
      "wheel                     0.31.1                   py36_1    conda-forge\r\n",
      "wrapt                     1.10.11          py36h470a237_1    conda-forge\r\n",
      "xmlrunner                 1.7.7                      py_0    conda-forge\r\n",
      "xorg-kbproto              1.0.7                h470a237_2    conda-forge\r\n",
      "xorg-libice               1.0.9                h470a237_4    conda-forge\r\n",
      "xorg-libsm                1.2.2                h470a237_5    conda-forge\r\n",
      "xorg-libx11               1.6.5                h470a237_2    conda-forge\r\n",
      "xorg-libxau               1.0.8                h470a237_6    conda-forge\r\n",
      "xorg-libxdmcp             1.1.2                h470a237_7    conda-forge\r\n",
      "xorg-libxext              1.3.3                h470a237_4    conda-forge\r\n",
      "xorg-libxpm               3.5.12               h470a237_2    conda-forge\r\n",
      "xorg-libxrender           0.9.10               h470a237_2    conda-forge\r\n",
      "xorg-libxt                1.1.5                h470a237_2    conda-forge\r\n",
      "xorg-renderproto          0.11.1               h470a237_2    conda-forge\r\n",
      "xorg-xextproto            7.3.0                h470a237_2    conda-forge\r\n",
      "xorg-xproto               7.0.31               h470a237_7    conda-forge\r\n",
      "xz                        5.2.4                h470a237_1    conda-forge\r\n",
      "yaml                      0.1.7                h470a237_1    conda-forge\r\n",
      "yarl                      1.3.0            py36h470a237_0    conda-forge\r\n",
      "zeromq                    4.2.5                hfc679d8_5    conda-forge\r\n",
      "zlib                      1.2.11               h470a237_3    conda-forge\r\n",
      "zope.interface            4.5.0            py36h470a237_0    conda-forge\r\n"
     ]
    }
   ],
   "source": [
    "!conda list -n metacompass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:metacompass]",
   "language": "python",
   "name": "conda-env-metacompass-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
